---
title: "combined_figures"
author: "Kathryn Atherton"
date: "2024-10-14"
output: html_document
---

```{r libraries, include=FALSE}
# setwd("/projectnb/talbot-lab-data/Katies_data/Street_Trees/")
library(ggplot2)
library(ggrepel)
library(vegan)
library(lmerTest)
library(tidyr)
library(MuMIn)
library(dplyr)
library(multcompView)
library(plotrix)
library(ggpubr)
library(Hmisc)
library(reshape2)
library(wesanderson)
library(patchwork)
library(indicspecies)
library(stringr)
library(indicspecies)
library(stringr)
library(Biostrings)
library(seqinr)
library(ggtree)
library(ape)
library(ggnewscale)
library(emmeans)
library(plotly)
library(nlme)
library(fields)
library(ggtext)
library(tibble)
```

```{r read in data}
shapes <- c(16, 17, 15, 8)
plot_pal <- c("#669BBC", "#003049")
pal <- c("#FF8080","#FF0000","#C00000","#800000","#000000")
# group_pal_fungi <- c("#840000", "#AF0000", "#FF0000", "#FF4040", "#FF7070", 
#                      "#FFB8B8", "#000040", "#000080", "#0000C0", "#0000FF", 
#                      "#4040FF", "#8080FF", "#C0C0FF", "#C08D02", "#E8AB02", 
#                      "#FFDE01", "#FFFF00", "#FFFF80", "#FFFFC0", "#808080")
# names(group_pal_fungi) <- c("Algal Parasite", "Human/Animal Pathogen", "Lichen Parasite", "Mycoparasite",
#                             "Plant Pathogen", "Sooty Mold", "Dung Saprotroph", 
#                             "Litter Saprotroph", "Nectar/Tap Saprotroph", 
#                             "Pollen Saprotroph", "Soil Saprotroph", "Wood Saprotroph",
#                             "Other Saprotroph", "Arbuscular Mycorrhizal",
#                             "Ectomycorrhizal", "Epiphyte", "Foliar Endophyte",
#                             "Lichenized", "Root Endophyte", "Other Fungi")

mean_se <- function(x, mult = 1) {  
  x <- na.omit(x)
  se <- mult * sqrt(var(x) / length(x))
  mean <- mean(x)
  data.frame(y = mean, ymin = mean - se, ymax = mean + se)
}

# read in metadata
metadata_its <- read.csv("/projectnb/talbot-lab-data/Katies_data/M-BUDS/Data/metadata_its_new.csv")
metadata_its <- distinct(metadata_its)
# metadata_16s_100 <- read.csv("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Data/metadata_16s_100.csv")
metadata_16s_200 <- read.csv("/projectnb/talbot-lab-data/M-BUDS/Street_Trees/Data/metadata_16s_200_new.csv")
metadata_16s_200 <- distinct(metadata_16s_200)
# metadata_16s_300 <- read.csv("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Data/metadata_16s_300.csv")
metadata_its$tree_distance_from_edge <- as.numeric(metadata_its$tree_distance_from_edge)
metadata_16s_200$tree_distance_from_edge <- as.numeric(metadata_16s_200$tree_distance_from_edge)

 # read in ASV data
all_aitch_its <- read.csv("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Analysis_dysbiosis/Data_Cleaning/Aitchison_Distance/ITS/all_its_sample_aitch.csv", 
                          row.names = 1)

leaf_rare_its <- read.csv("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Analysis_dysbiosis/Data_Cleaning/Rarefaction/ITS/leaf_its_asv_rare_wtax_guild_wasvid.csv", 
                          row.names = 1)
m_rare_its <- read.csv("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Analysis_dysbiosis/Data_Cleaning/Rarefaction/ITS/m_its_asv_rare_wtax_guild_wasvid.csv", 
                       row.names = 1)
o_rare_its <- read.csv("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Analysis_dysbiosis/Data_Cleaning/Rarefaction/ITS/o_its_asv_rare_wtax_guild_wasvid.csv", 
                       row.names = 1)
root_rare_its <- read.csv("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Analysis_dysbiosis/Data_Cleaning/Rarefaction/ITS/root_its_asv_rare_wtax_guild_wasvid.csv", 
                          row.names = 1)


bg_aitch_16s <- read.csv("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Analysis_dysbiosis/Data_Cleaning/Aitchison_Distance/16S/all_16s_sample_aitch.csv", 
                         row.names = 1)
leaf_aitch_16s <- read.csv("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Analysis_dysbiosis/Data_Cleaning/Aitchison_Distance/16S/leaf_16s_sample_aitch.csv", 
                           row.names = 1)

leaf_rare_16s_200 <- read.csv("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Analysis_dysbiosis/Data_Cleaning/Rarefaction/16S/leaf_16s_asv_rare_wtax_guild_200.csv", 
                              row.names = 1)
m_rare_16s <- read.csv("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Analysis_dysbiosis/Data_Cleaning/Rarefaction/16S/m_16s_asv_rare_wtax_guild.csv", 
                       row.names = 1)
o_rare_16s <- read.csv("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Analysis_dysbiosis/Data_Cleaning/Rarefaction/16S/o_16s_asv_rare_wtax_guild.csv", 
                       row.names = 1)
root_rare_16s <- read.csv("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Analysis_dysbiosis/Data_Cleaning/Rarefaction/16S/root_16s_asv_rare_wtax_guild.csv", 
                          row.names = 1)


metadata_its$sample_type<- paste0(metadata_its$soil_horizon, " ", 
                                  metadata_its$sample_type)
metadata_its$sample_type <- gsub("NA ", "", 
                                 metadata_its$sample_type)

metadata_its$sample_type <- gsub("O Soil", "Soil, 0-15 cm", 
                                 metadata_its$sample_type)
metadata_its$sample_type <- gsub("M Soil", "Soil, 15-30 cm", 
                                 metadata_its$sample_type)

metadata_its$sample_type <- gsub("O Root", "Roots, 0-15 cm", 
                                 metadata_its$sample_type)
metadata_its$sample_type <- gsub("M Root", "Roots, 15-30 cm", 
                                 metadata_its$sample_type)

metadata_its$sample_type <- factor(metadata_its$sample_type, 
                                   levels = c("Leaf", "Soil, 0-15 cm", 
                                              "Soil, 15-30 cm", "Roots, 0-15 cm", 
                                              "Roots, 15-30 cm"))

metadata_its$tree_pit_type <- gsub("Pit", "Street Tree",
                                   metadata_its$tree_pit_type)
metadata_its$tree_pit_type <- gsub("Grass", "Street Tree",
                                   metadata_its$tree_pit_type)

metadata_its <- metadata_its[which(metadata_its$sample_name %in% 
                                     colnames(all_aitch_its)),]
metadata_its$tree_pit_type <- factor(metadata_its$tree_pit_type,
                                 levels = c("Street Tree", "Urban Edge",
                                            "Urban Interior", "Rural Edge",
                                            "Rural Interior"))
# metadata_its$tree_pit_type <- factor(metadata_its$tree_pit_type, 
#                                  levels = c("Pit", "Grass", "Urban Edge", 
#                                             "Urban Interior", "Rural Edge", 
#                                             "Rural Interior"))

fungal_trait_names <- c("perc_algal_parasite", "perc_animal_endosymbiont", "perc_animal_parasite", 
                        "perc_amf", "perc_arthropod_associated", 
                        "perc_dung_saprotroph", "perc_ecm", "perc_epiphyte",
                        "perc_foliar_endophyte", "perc_lichen_parasite",
                        "perc_lichenized", "perc_litter_saprotroph",
                        "perc_moss_symbiont", "perc_mycoparasite",
                        "perc_nectar_tap_saprotroph", 
                        "perc_plant_pathogen", 
                        "perc_pollen_saprotroph", "perc_root_endophyte",
                        "perc_soil_saprotroph", "perc_sooty_mold",
                        "perc_wood_saprotroph", 
                        "perc_animal_pathogen", 
                        "perc_human_pathogen", "perc_opportunistic_pathogen",
                        "perc_root_associated_pathogen", 
                        "perc_leaf_fruit_seed_pathogen", "perc_wood_pathogen",
                        "shannon", "shannon_nopath", "perc_all_saprotroph", "perc_all_pathotroph", "perc_all_symbiont")

metadata_its$soil_percent_moisture <- gsub("%", "", 
                                           metadata_its$soil_percent_moisture)
metadata_its$soil_percent_organic_matter <- gsub("%", "",
                                                 metadata_its$soil_percent_organic_matter)
metadata_its$soil_percent_moisture <- as.numeric(metadata_its$soil_percent_moisture)
metadata_its$soil_percent_organic_matter <- as.numeric(metadata_its$soil_percent_organic_matter)

metadata_16s_200$sample_type<- paste0(metadata_16s_200$soil_horizon, " ",
                                      metadata_16s_200$sample_type)
metadata_16s_200$sample_type <- gsub("NA ", "", 
                                     metadata_16s_200$sample_type)

metadata_16s_200$sample_type <- gsub("O Soil", "Soil, 0-15 cm",
                                     metadata_16s_200$sample_type)
metadata_16s_200$sample_type <- gsub("M Soil", "Soil, 15-30 cm",
                                     metadata_16s_200$sample_type)

metadata_16s_200$sample_type <- gsub("O Root", "Roots, 0-15 cm",
                                     metadata_16s_200$sample_type)
metadata_16s_200$sample_type <- gsub("M Root", "Roots, 15-30 cm",
                                     metadata_16s_200$sample_type)

metadata_16s_200$sample_type <- factor(metadata_16s_200$sample_type, 
                                       levels = c("Leaf", "Soil, 0-15 cm", 
                                                  "Soil, 15-30 cm", 
                                                  "Roots, 0-15 cm", 
                                                  "Roots, 15-30 cm"))

metadata_16s_200$tree_pit_type <- gsub("Pit", "Street Tree",
                                       metadata_16s_200$tree_pit_type)
metadata_16s_200$tree_pit_type <- gsub("Grass", "Street Tree",
                                       metadata_16s_200$tree_pit_type)

metadata_16s_200 <- metadata_16s_200[which(metadata_16s_200$sample_name %in%
                                             c(colnames(bg_aitch_16s), colnames(leaf_aitch_16s))),]
metadata_16s_200 <- metadata_16s_200[which(metadata_16s_200$sequences_dropped == "No"),]
metadata_16s_200$tree_pit_type <- factor(metadata_16s_200$tree_pit_type,
                                 levels = c("Street Tree", "Urban Edge",
                                            "Urban Interior", "Rural Edge",
                                            "Rural Interior"))
# metadata_16s_200$tree_pit_type <- factor(metadata_16s_200$tree_pit_type,
#                                  levels = c("Pit", "Grass", "Urban Edge", 
#                                             "Urban Interior", "Rural Edge", 
#                                             "Rural Interior"))

bacterial_trait_names <- c("perc_c_fixation", "perc_cellulolytic", 
                           "perc_chitinolytic", "perc_lignolytic",
                           "perc_carbon_moNO<sub>x</sub> Concentrationide_oxidation", "perc_n_fixation",
                           "perc_copiotroph", "perc_denitrification",
                           "perc_dissim_nitrate_reduction", 
                           "perc_hydrocarbon_degradation",
                           "perc_sulfonate_desulfurization",
                           "perc_other_p_cycling",
                           "perc_oxidize_reduced_sulfur", "perc_methanotroph",
                           "perc_oligotroph", "perc_partial_nitrification", 
                           "perc_archaea", "perc_plant_pathogen", 
                           "perc_animal_pathogen", "perc_zoonotic_pathogen", 
                           "shannon", "shannon.x", "shannon_nopath")

metadata_16s_200$soil_percent_moisture <- gsub("%", "",
                                               metadata_16s_200$soil_percent_moisture)
metadata_16s_200$soil_percent_organic_matter <- gsub("%", "",
                                                     metadata_16s_200$soil_percent_organic_matter)
metadata_16s_200$soil_percent_moisture <- as.numeric(metadata_16s_200$soil_percent_moisture)
metadata_16s_200$soil_percent_organic_matter <- as.numeric(metadata_16s_200$soil_percent_organic_matter)

metadata_its$perc_algal_parasite <- log(metadata_its$perc_algal_parasite+1)
# metadata_its$perc_animal_parasite <- log(metadata_its$perc_animal_parasite+1)
metadata_its$perc_amf <- log(metadata_its$perc_amf+1)
metadata_its$perc_arthropod_associated <- log(metadata_its$perc_arthropod_associated+1)
metadata_its$perc_dung_saprotroph <- log(metadata_its$perc_dung_saprotroph+1)
# metadata_its$perc_ecm <- log(metadata_its$perc_ecm+1)
metadata_its$perc_epiphyte <- log(metadata_its$perc_epiphyte+1)
metadata_its$perc_foliar_endophyte <- log(metadata_its$perc_foliar_endophyte+1)
metadata_its$perc_lichen_parasite <- log(metadata_its$perc_lichen_parasite+1)
metadata_its$perc_lichenized <- log(metadata_its$perc_lichenized+1)
metadata_its$perc_litter_saprotroph <- log(metadata_its$perc_litter_saprotroph+1)
metadata_its$perc_moss_symbiont <- log(metadata_its$perc_moss_symbiont+1)
metadata_its$perc_mycoparasite <- log(metadata_its$perc_mycoparasite+1)
metadata_its$perc_nectar_tap_saprotroph <- log(metadata_its$perc_nectar_tap_saprotroph+1)
metadata_its$perc_plant_pathogen <- log(metadata_its$perc_plant_pathogen+1)
metadata_its$perc_pollen_saprotroph <- log(metadata_its$perc_pollen_saprotroph+1)
metadata_its$perc_root_endophyte <- log(metadata_its$perc_root_endophyte+1)
metadata_its$perc_soil_saprotroph <- log(metadata_its$perc_soil_saprotroph+1)
metadata_its$perc_sooty_mold <- log(metadata_its$perc_sooty_mold+1)
metadata_its$perc_wood_saprotroph <- log(metadata_its$perc_wood_saprotroph+1)
metadata_its$perc_all_saprotroph <- log(metadata_its$perc_all_saprotroph+1)
metadata_its$perc_all_pathotroph <- log(metadata_its$perc_all_pathotroph+1)
metadata_its$perc_all_symbiont <- log(metadata_its$perc_all_symbiont+1)
# metadata_its$perc_animal_pathogen <- log(metadata_its$perc_animal_pathogen+1)
# metadata_its$perc_human_pathogen <- log(metadata_its$perc_human_pathogen+1)
metadata_its$perc_opportunistic_pathogen <- log(metadata_its$perc_opportunistic_pathogen+1)
metadata_its$perc_root_associated_pathogen <- log(metadata_its$perc_root_associated_pathogen+1)
metadata_its$perc_leaf_fruit_seed_pathogen <- log(metadata_its$perc_leaf_fruit_seed_pathogen+1)
metadata_its$perc_wood_pathogen <- log(metadata_its$perc_wood_pathogen+1)

metadata_16s_200$perc_c_fixation <- log(metadata_16s_200$perc_c_fixation+1)
metadata_16s_200$perc_cellulolytic <- log(metadata_16s_200$perc_cellulolytic+1)
metadata_16s_200$perc_chitinolytic <- log(metadata_16s_200$perc_chitinolytic+1)
metadata_16s_200$perc_lignolytic <- log(metadata_16s_200$perc_lignolytic+1)
metadata_16s_200$perc_carbon_monoxide_oxidation <- log(metadata_16s_200$perc_carbon_monoxide_oxidation+1)
metadata_16s_200$perc_n_fixation <- log(metadata_16s_200$perc_n_fixation+1)
metadata_16s_200$perc_copiotroph <- log(metadata_16s_200$perc_copiotroph+1)
metadata_16s_200$perc_denitrification <- log(metadata_16s_200$perc_denitrification+1)
metadata_16s_200$perc_dissim_nitrate_reduction <- log(metadata_16s_200$perc_dissim_nitrate_reduction+1)
metadata_16s_200$perc_hydrocarbon_degradation <- log(metadata_16s_200$perc_hydrocarbon_degradation+1)
metadata_16s_200$perc_sulfonate_desulfurization <- log(metadata_16s_200$perc_sulfonate_desulfurization+1)
metadata_16s_200$perc_other_p_cycling <- log(metadata_16s_200$perc_other_p_cycling+1)
metadata_16s_200$perc_oxidize_reduced_sulfur <- log(metadata_16s_200$perc_oxidize_reduced_sulfur+1)
metadata_16s_200$perc_methanotroph <- log(metadata_16s_200$perc_methanotroph+1)
metadata_16s_200$perc_oligotroph <- log(metadata_16s_200$perc_oligotroph+1)
metadata_16s_200$perc_partial_nitrification <- log(metadata_16s_200$perc_partial_nitrification+1)
metadata_16s_200$perc_archaea <- log(metadata_16s_200$perc_archaea+1)
metadata_16s_200$perc_plant_pathogen <- log(metadata_16s_200$perc_plant_pathogen+1)
# metadata_16s_200$perc_animal_pathogen <- log(metadata_16s_200$perc_animal_pathogen+1)
# metadata_16s_200$perc_zoonotic_pathogen <- log(metadata_16s_200$perc_zoonotic_pathogen+1)

all_aitch_its <- all_aitch_its[which(row.names(all_aitch_its) %in%
                              metadata_its$sample_name),
                              which(colnames(all_aitch_its) %in%
                                       metadata_its$sample_name)]
bg_aitch_16s <- bg_aitch_16s[which(row.names(bg_aitch_16s) %in%
                                       metadata_16s_200$sample_name),
                             which(colnames(bg_aitch_16s) %in%
                                       metadata_16s_200$sample_name)]
leaf_aitch_16s <- leaf_aitch_16s[which(row.names(leaf_aitch_16s) %in%
                                       metadata_16s_200$sample_name),
                             which(colnames(leaf_aitch_16s) %in%
                                       metadata_16s_200$sample_name)]
fungal_trait_names <- c(fungal_trait_names, bacterial_trait_names)

network_metrics <- read.csv("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Analysis_dysbiosis/Networks/Results/network_metrics_plot.csv")
network_metrics$Sample_Type <- factor(network_metrics$Sample_Type, levels = c("Leaf", "Soil, 0-15 cm", "Soil, 15-30 cm", "Root"))
network_metrics$Tree_Type <- factor(network_metrics$Tree_Type, levels = c("Street Tree", "Urban Forest", "Rural Forest"))

leaf_indic_16s <- read.csv("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Analysis_dysbiosis/Indicator_Species/16S/leaf_indic_phylogenetic_info_sp.csv")
bg_indic_16s <- read.csv("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Analysis_dysbiosis/Indicator_Species/16S/bg_indic_phylogenetic_info_sp.csv")

leaf_indic_its <- read.csv("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Analysis_dysbiosis/Indicator_Species/ITS/leaf_indic_phylogenetic_info_sp.csv")
bg_indic_its <- read.csv("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Analysis_dysbiosis/Indicator_Species/ITS/bg_indic_phylogenetic_info_sp.csv")


tax_w_paths <- read.csv("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Analysis_dysbiosis/Data_Cleaning/Raw_Data/16S/tax_16s_allruns_wpathogen.csv")
tax_w_paths <- tax_w_paths[,c(2,10)]
tax_w_paths$pathogen_type <- gsub("t__", "", tax_w_paths$pathogen_type)
tax_w_paths$pathogen_type <- gsub(" ", ", ", tax_w_paths$pathogen_type)
tax_w_paths$pathogen_type <- gsub(" ", ", ", tax_w_paths$pathogen_type)
tax_w_paths$pathogen_type <- gsub("Animal, Zoonotic, Plant", "Plant, Animal, Zoonotic", tax_w_paths$pathogen_type)
tax_w_paths$pathogen_type <- gsub("Zoonotic, Animal", "Animal, Zoonotic", tax_w_paths$pathogen_type)
tax_w_paths$pathogen_type <- gsub("Animal, Plant", "Plant, Animal", tax_w_paths$pathogen_type)
tax_w_paths$pathogen_type <- gsub("Animal, Zoonotic, Plant", "Plant, Animal, Zoonotic", tax_w_paths$pathogen_type)
tax_w_paths$pathogen_type <- gsub("Zoonotic, Plant, Animal", "Plant, Animal, Zoonotic", tax_w_paths$pathogen_type)
tax_w_paths$pathogen_type <- gsub("Zoonotic, Plant", "Plant, Zoonotic", tax_w_paths$pathogen_type)

pathogens <- vroom::vroom("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Analysis_dysbiosis/MBPD/street_trees_pathogens_data.txt")
pathogens$row <- rownames(pathogens)

asv_ids <- unique(pathogens$asv_id)
keep <- c()
for(i in 1:length(asv_ids)){
  asv <- asv_ids[i]
  asv_matches <- pathogens[which(pathogens$asv_id == asv),]
  perc_id <- max(asv_matches$perc_id)
  keep <- c(keep, asv_matches$row[which(asv_matches$perc_id == perc_id)])
}

pathogens <- pathogens[keep,c(1,3)]

its_pathogens <- read.table("/projectnb/talbot-lab-data/Katies_data/Databases/UNITE/blast_unite.txt")
colnames(its_pathogens) <- c("asv_id", "ref_id", "perc_id", "perc_cov", "length", "mismatch", "gapopen", "qstart", "qend", "sstart", "send", "eval", "bitscore")
its_pathogens <- its_pathogens[,c(1,3)]
```

```{r Table S1}
hf069_dfb <- metadata_its[which(metadata_its$study_system == "Harvard Biomass"),]
hf069_dfb <- hf069_dfb[,which(colnames(hf069_dfb) %in% c("tree_id", "tree_dist_to_boston_km"))]
hf069_dfb <- distinct(hf069_dfb)
mean(hf069_dfb$tree_dist_to_boston_km)

ab_dfb <- metadata_its[which(metadata_its$plot_name == "AB"),]
ab_dfb <- ab_dfb[,which(colnames(ab_dfb) %in% c("tree_id", "tree_dist_to_boston_km"))]
ab_dfb <- distinct(ab_dfb)
mean(ab_dfb$tree_dist_to_boston_km)

bb_dfb <- metadata_its[which(metadata_its$plot_name == "BB"),]
bb_dfb <- bb_dfb[,which(colnames(bb_dfb) %in% c("tree_id", "tree_dist_to_boston_km"))]
bb_dfb <- distinct(bb_dfb)
mean(bb_dfb$tree_dist_to_boston_km)

do_dfb <- metadata_its[which(metadata_its$plot_name == "DO"),]
do_dfb <- do_dfb[,which(colnames(do_dfb) %in% c("tree_id", "tree_dist_to_boston_km"))]
do_dfb <- distinct(do_dfb)
mean(do_dfb$tree_dist_to_boston_km)

jp_dfb <- metadata_its[which(metadata_its$plot_name == "JP"),]
jp_dfb <- jp_dfb[,which(colnames(jp_dfb) %in% c("tree_id", "tree_dist_to_boston_km"))]
jp_dfb <- distinct(jp_dfb)
mean(jp_dfb$tree_dist_to_boston_km)

mc_dfb <- metadata_its[which(metadata_its$plot_name == "MC"),]
mc_dfb <- mc_dfb[,which(colnames(mc_dfb) %in% c("tree_id", "tree_dist_to_boston_km"))]
mc_dfb <- distinct(mc_dfb)
mean(mc_dfb$tree_dist_to_boston_km)

mp_dfb <- metadata_its[which(metadata_its$plot_name == "MP"),]
mp_dfb <- mp_dfb[,which(colnames(mp_dfb) %in% c("tree_id", "tree_dist_to_boston_km"))]
mp_dfb <- distinct(mp_dfb)
mean(mp_dfb$tree_dist_to_boston_km)

sb_dfb <- metadata_its[which(metadata_its$plot_name == "SB"),]
sb_dfb <- sb_dfb[,which(colnames(sb_dfb) %in% c("tree_id", "tree_dist_to_boston_km"))]
sb_dfb <- distinct(sb_dfb)
mean(sb_dfb$tree_dist_to_boston_km)

se_dfb <- metadata_its[which(metadata_its$plot_name == "SE"),]
se_dfb <- se_dfb[,which(colnames(se_dfb) %in% c("tree_id", "tree_dist_to_boston_km"))]
se_dfb <- distinct(se_dfb)
mean(se_dfb$tree_dist_to_boston_km)

wr_dfb <- metadata_its[which(metadata_its$plot_name == "WR"),]
wr_dfb <- wr_dfb[,which(colnames(wr_dfb) %in% c("tree_id", "tree_dist_to_boston_km"))]
wr_dfb <- distinct(wr_dfb)
mean(wr_dfb$tree_dist_to_boston_km)
```

```{r Table S2}
metadata_its <- read.csv("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Data/metadata_its.csv")
metadata_its <- distinct(metadata_its)
metadata_16s_200 <- read.csv("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Data/metadata_16s_200.csv")
metadata_16s_200 <- distinct(metadata_16s_200)

metadata_its <- metadata_its[which(metadata_its$sequences_dropped != "No sequence data"),]
metadata_16s <- metadata_16s_200[which(metadata_16s_200$sequences_dropped != "No sequence data"),]

sum(metadata_its$raw_seq_count)
sum(metadata_16s$raw_seq_count)

mean(metadata_its$raw_seq_count)
mean(metadata_16s$raw_seq_count)

sd(metadata_its$raw_seq_count)
sd(metadata_16s$raw_seq_count)

metadata_its_leaf <- metadata_its[which(metadata_its$sample_type == "Leaf"),]
metadata_its_o <- metadata_its[which(metadata_its$sample_type == "Soil" & metadata_its$soil_horizon == "O"),]
metadata_its_m <- metadata_its[which(metadata_its$sample_type == "Soil" & metadata_its$soil_horizon == "M"),]
metadata_its_root <- metadata_its[which(metadata_its$sample_type == "Root"),]

paste0(round(mean(metadata_its_leaf$raw_seq_count),2),  " +/- ", round(sd(metadata_its_leaf$raw_seq_count),2))
paste0(round(mean(metadata_its_o$raw_seq_count),2),  " +/- ", round(sd(metadata_its_o$raw_seq_count),2))
paste0(round(mean(metadata_its_m$raw_seq_count),2),  " +/- ", round(sd(metadata_its_m$raw_seq_count),2))
paste0(round(mean(metadata_its_root$raw_seq_count),2),  " +/- ", round(sd(metadata_its_root$raw_seq_count),2))

paste0(round(mean(metadata_its_leaf$seq_count_dada2),2),  " +/- ", round(sd(metadata_its_leaf$seq_count_dada2),2))
paste0(round(mean(metadata_its_o$seq_count_dada2),2),  " +/- ", round(sd(metadata_its_o$seq_count_dada2),2))
paste0(round(mean(metadata_its_m$seq_count_dada2),2),  " +/- ", round(sd(metadata_its_m$seq_count_dada2),2))
paste0(round(mean(metadata_its_root$seq_count_dada2),2),  " +/- ", round(sd(metadata_its_root$seq_count_dada2),2))

metadata_its_leaf <- metadata_its_leaf[which(metadata_its_leaf$sequences_dropped == "No"),]
metadata_its_o <- metadata_its_o[which(metadata_its_o$sequences_dropped == "No"),]
metadata_its_m <- metadata_its_m[which(metadata_its_m$sequences_dropped == "No"),]
metadata_its_root <- metadata_its_root[which(metadata_its_root$sequences_dropped == "No"),]

paste0(round(mean(metadata_its_leaf$seq_count_dada2),2),  " +/- ", round(sd(metadata_its_leaf$seq_count_dada2),2))
paste0(round(mean(metadata_its_o$seq_count_dada2),2),  " +/- ", round(sd(metadata_its_o$seq_count_dada2),2))
paste0(round(mean(metadata_its_m$seq_count_dada2),2),  " +/- ", round(sd(metadata_its_m$seq_count_dada2),2))
paste0(round(mean(metadata_its_root$seq_count_dada2),2),  " +/- ", round(sd(metadata_its_root$seq_count_dada2),2))

paste0(round(mean(metadata_its_leaf$seq_count_decontam),2),  " +/- ", round(sd(metadata_its_leaf$seq_count_decontam),2))
paste0(round(mean(metadata_its_o$seq_count_decontam),2),  " +/- ", round(sd(metadata_its_o$seq_count_decontam),2))
paste0(round(mean(metadata_its_m$seq_count_decontam),2),  " +/- ", round(sd(metadata_its_m$seq_count_decontam),2))
paste0(round(mean(metadata_its_root$seq_count_decontam),2),  " +/- ", round(sd(metadata_its_root$seq_count_decontam),2))






metadata_16s_leaf <- metadata_16s[which(metadata_16s$sample_type == "Leaf"),]
metadata_16s_o <- metadata_16s[which(metadata_16s$sample_type == "Soil" & metadata_16s$soil_horizon == "O"),]
metadata_16s_m <- metadata_16s[which(metadata_16s$sample_type == "Soil" & metadata_16s$soil_horizon == "M"),]
metadata_16s_root <- metadata_16s[which(metadata_16s$sample_type == "Root"),]

paste0(round(mean(metadata_16s_leaf$raw_seq_count),2),  " +/- ", round(sd(metadata_16s_leaf$raw_seq_count),2))
paste0(round(mean(metadata_16s_o$raw_seq_count),2),  " +/- ", round(sd(metadata_16s_o$raw_seq_count),2))
paste0(round(mean(metadata_16s_m$raw_seq_count),2),  " +/- ", round(sd(metadata_16s_m$raw_seq_count),2))
paste0(round(mean(metadata_16s_root$raw_seq_count),2),  " +/- ", round(sd(metadata_16s_root$raw_seq_count),2))

paste0(round(mean(metadata_16s_leaf$seq_count_dada2),2),  " +/- ", round(sd(metadata_16s_leaf$seq_count_dada2),2))
paste0(round(mean(metadata_16s_o$seq_count_dada2, na.rm = TRUE),2),  " +/- ", round(sd(metadata_16s_o$seq_count_dada2, na.rm = TRUE),2))
paste0(round(mean(metadata_16s_m$seq_count_dada2),2),  " +/- ", round(sd(metadata_16s_m$seq_count_dada2),2))
paste0(round(mean(metadata_16s_root$seq_count_dada2),2),  " +/- ", round(sd(metadata_16s_root$seq_count_dada2),2))

metadata_16s_leaf <- metadata_16s_leaf[which(metadata_16s_leaf$sequences_dropped == "No"),]
metadata_16s_o <- metadata_16s_o[which(metadata_16s_o$sequences_dropped == "No"),]
metadata_16s_m <- metadata_16s_m[which(metadata_16s_m$sequences_dropped == "No"),]
metadata_16s_root <- metadata_16s_root[which(metadata_16s_root$sequences_dropped == "No"),]

paste0(round(mean(metadata_16s_leaf$seq_count_dada2),2),  " +/- ", round(sd(metadata_16s_leaf$seq_count_dada2),2))
paste0(round(mean(metadata_16s_o$seq_count_dada2, na.rm = TRUE),2),  " +/- ", round(sd(metadata_16s_o$seq_count_dada2, na.rm = TRUE),2))
paste0(round(mean(metadata_16s_m$seq_count_dada2),2),  " +/- ", round(sd(metadata_16s_m$seq_count_dada2),2))
paste0(round(mean(metadata_16s_root$seq_count_dada2),2),  " +/- ", round(sd(metadata_16s_root$seq_count_dada2),2))

paste0(round(mean(metadata_16s_leaf$seq_count_decontam),2),  " +/- ", round(sd(metadata_16s_leaf$seq_count_decontam),2))
paste0(round(mean(metadata_16s_o$seq_count_decontam, na.rm = TRUE),2),  " +/- ", round(sd(metadata_16s_o$seq_count_decontam, na.rm = TRUE),2))
paste0(round(mean(metadata_16s_m$seq_count_decontam),2),  " +/- ", round(sd(metadata_16s_m$seq_count_decontam),2))
paste0(round(mean(metadata_16s_root$seq_count_decontam),2),  " +/- ", round(sd(metadata_16s_root$seq_count_decontam),2))




colSums(leaf_rare_its[11:ncol(leaf_rare_its)])
colSums(o_rare_its[11:ncol(o_rare_its)])
colSums(m_rare_its[11:ncol(m_rare_its)])
colSums(root_rare_its[11:ncol(root_rare_its)])

colSums(leaf_rare_16s_200[9:(ncol(leaf_rare_16s_200)-1)])
colSums(o_rare_16s[9:(ncol(o_rare_16s)-1)])
colSums(m_rare_16s[9:(ncol(m_rare_16s)-1)])
colSums(root_rare_16s[9:(ncol(root_rare_16s)-1)])

length(unique(metadata_its_leaf$tree_id))
length(unique(metadata_16s_leaf$tree_id))

metadata_its_bg <- rbind(metadata_its_m, metadata_its_o)
metadata_its_bg <- rbind(metadata_its_bg, metadata_its_root)

metadata_16s_bg <- rbind(metadata_16s_m, metadata_16s_o)
metadata_16s_bg <- rbind(metadata_16s_bg, metadata_16s_root)

length(unique(metadata_its_bg$tree_id))
length(unique(metadata_16s_bg$tree_id))
```

Figure 1a
```{r fungi symbionts vs fragmentation}
metadata_soil_its <- metadata_its[which(metadata_its$sample_type %in% c("Soil, 0-15 cm")),]
metadata_soil_its$tree_distance_from_edge <- as.numeric(metadata_soil_its$tree_distance_from_edge)
length(unique(metadata_soil_its$tree_id))

metadata_leaf_its <- metadata_its[which(metadata_its$sample_type %in% c("Leaf")),]
metadata_leaf_its$tree_distance_from_edge <- as.numeric(metadata_leaf_its$tree_distance_from_edge)
length(unique(metadata_leaf_its$tree_id))

fungal_trait_soil <- metadata_soil_its[, colnames(metadata_soil_its) %in% 
                                    fungal_trait_names]
fungal_trait_leaf <- metadata_leaf_its[, colnames(metadata_leaf_its) %in% 
                                    fungal_trait_names]
standard_error <- function(x) sd(x) / sqrt(length(x)) 

mean_soil <- aggregate(fungal_trait_soil,
                  by=list(metadata_soil_its$tree_urban_type,
                          metadata_soil_its$tree_edge_type), FUN= "mean")
mean_leaf <- aggregate(fungal_trait_leaf,
                  by=list(metadata_leaf_its$tree_urban_type,
                          metadata_leaf_its$tree_edge_type), FUN= "mean")
se_soil <- aggregate(fungal_trait_soil,
                by=list(metadata_soil_its$tree_urban_type,
                        metadata_soil_its$tree_edge_type), FUN= standard_error)
se_leaf <- aggregate(fungal_trait_leaf,
                by=list(metadata_leaf_its$tree_urban_type,
                        metadata_leaf_its$tree_edge_type), FUN= standard_error)

vdata <- cbind(mean_soil[,1:2], mean_soil$perc_ecm, se_soil$perc_ecm, mean_leaf$perc_epiphyte, se_leaf$perc_epiphyte, mean_leaf$perc_foliar_endophyte, se_leaf$perc_foliar_endophyte)

colnames(vdata) <- c("Site", "Edge", "ecm", "ecm_se", "epiphyte", "epiphtye_se", "endophyte", "endophyte_se")
vdata$Edge <- factor(vdata$Edge, 
                     levels = c("Urban Street Tree", "Forest Edge", 
                                "Forest Interior"))

vdata_long <- as.data.frame(matrix(data = NA, nrow = 10, ncol = 5))
colnames(vdata_long) <- c("Site", "Edge", "Guild", "Abundance", "SE")
vdata_long$Site <- rep(vdata$Site, 2)
vdata_long$Edge <- rep(vdata$Edge, 2)
vdata_long$Guild <- c(rep("ECM in soil", 5), 
                      rep("Epiphytes in leaves", 5))
vdata_long$Abundance <- c(vdata$ecm, 20*vdata$epiphyte)
vdata_long$SE <- c(vdata$ecm_se, vdata$epiphyte_se)

vdata_long$site_edge <- paste0(vdata_long$Site, " ", vdata_long$Edge)
vdata_long$site_edge <- gsub("Urban Street Tree Urban Street Tree", 
                             "Urban Street Tree",
                             vdata_long$site_edge)
vdata_long$site_edge <- factor(vdata_long$site_edge, 
                               levels = c("Urban Street Tree", 
                                          "Urban Forest Edge",
                                          "Urban Forest Interior",
                                          "Rural Forest Edge", 
                                          "Rural Forest Interior"))


metadata_soil_its$site_edge <- paste0(metadata_soil_its$tree_urban_type, " ", metadata_soil_its$tree_edge_type)
metadata_soil_its$site_edge[which(metadata_soil_its$site_edge == "Urban Street Tree Urban Street Tree")] <- "Urban Street Tree"
metadata_leaf_its$site_edge <- paste0(metadata_leaf_its$tree_urban_type, " ", metadata_leaf_its$tree_edge_type)
metadata_leaf_its$site_edge[which(metadata_leaf_its$site_edge == "Urban Street Tree Urban Street Tree")] <- "Urban Street Tree"

vdata_long$Guild <- factor(vdata_long$Guild, levels = c("ECM in soil", "Epiphytes in leaves"))
vdata_long$tukey <- c(
  "ab[\"\\u25CF\"]", 
  "a[\"\\u25CF\"]", 
  "a[\"\\u25CF\"]", 
  "a[\"\\u25CF\"]", 
  "b[\"\\u25CF\"]", 
  "a[\"\\u25B2\"]", 
  "a[\"\\u25B2\"]", 
  "a[\"\\u25B2\"]", 
  "a[\"\\u25B2\"]", 
  "b[\"\\u25B2\"]"
)
vdata_long$site_edge <- gsub(" ", "\n", vdata_long$site_edge)
vdata_long$site_edge <- factor(vdata_long$site_edge, 
                               levels = c("Urban\nStreet\nTree", 
                                          "Urban\nForest\nEdge",
                                          "Urban\nForest\nInterior",
                                          "Rural\nForest\nEdge",
                                          "Rural\nForest\nInterior"))

fungi_ecm_v_frag_bg_v4_woutliers <- ggplot() +
   geom_line(data = vdata_long, 
             aes(x=site_edge, y=Abundance, shape = Guild, group = Guild), 
            linetype = 2) +
  geom_point(data = vdata_long, 
             aes(x=site_edge, y=Abundance, color=site_edge, shape = Guild, 
                 group = Guild), size=5) + 
  geom_errorbar(data = vdata_long, 
                aes(x=site_edge, ymin=Abundance-SE, ymax=Abundance+SE, 
                    color=site_edge), width = 0.2) +
  geom_text(data=vdata_long, family = "Arial",
            aes(x = site_edge, y = Abundance + SE + 3, label = tukey), size = 3, parse = TRUE) +
  # geom_point(data = metadata_soil_its, 
  #            aes(x = site_edge, y = perc_ecm, color = site_edge), 
  #            position = position_jitter(width = 0.2), size = 2, alpha = 0.2, stroke = NA) +
  theme_classic() + labs (y="ECM abundance (%)", x="", shape = "") + scale_y_continuous(sec.axis = sec_axis(~./20, name = "log(Epiphyte abundance (%))")) +
  theme(text = element_text(family = "Arial"),
        axis.title.x = element_text(size = 10, color="black"), 
        axis.title.y = element_text(size = 10), 
        axis.text.x = element_text(size = 10, color = "black"), 
        axis.text.y = element_text(size = 10, color = "black"), 
        legend.text = element_text(size = 7), 
        legend.title = element_text(size = 1), 
        plot.title = element_text(size = 10), legend.position = c(0.5, 0.2)) +
  scale_color_manual(values = pal, guide = "none") + 
  scale_fill_manual(values = pal, guide = "none") + 
  ggtitle("Tree-associated fungal symbionts")

fungi_ecm_v_frag_bg_v4_woutliers
ggsave("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Figures/figure1_fungal_symbionts_fragmentation.pdf", plot = fungi_ecm_v_frag_bg_v4_woutliers, dpi = 300, width = 6, height = 3.5, units = "in", device = cairo_pdf)

# model1 <- lmer(formula = perc_ecm ~ site_edge + (1|plot_name/tree_id),
#               data = metadata_soil_its)
# summary(emmeans(model1, pairwise ~ site_edge))
model1 <- lme(formula(perc_ecm ~ site_edge), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML", data = metadata_soil_its)
summary(model1)
summary(emmeans(model1, pairwise ~ site_edge))

# model1 <- lmer(formula = perc_epiphyte ~ site_edge + (1|plot_name/tree_id),
#               data = metadata_leaf_its)
# summary(emmeans(model1, pairwise ~ site_edge))
model1 <- lme(formula(perc_epiphyte ~ site_edge), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML", data = metadata_leaf_its)
summary(model1)
summary(emmeans(model1, pairwise ~ site_edge))


# model1 <- lmer(formula = perc_foliar_endophyte ~ site_edge + (1|plot_name/tree_id),
#               data = metadata_leaf_its)
# summary(emmeans(model1, pairwise ~ site_edge))
```

Figure 1b
```{r fungi pathogens vs. fragmentation}
metadata_soil_its <- metadata_its[which(metadata_its$sample_type == "Soil, 0-15 cm"),]
metadata_soil_16s <- metadata_16s_200[which(metadata_16s_200$sample_type == "Soil, 0-15 cm"),]
length(unique(metadata_soil_16s$tree_id))

fungal_trait <- metadata_soil_its[, colnames(metadata_soil_its) %in% 
                                    fungal_trait_names]
bacterial_trait <- metadata_soil_16s[, colnames(metadata_soil_16s) %in% 
                                    bacterial_trait_names]
standard_error <- function(x) sd(x) / sqrt(length(x)) 

meanits <- aggregate(fungal_trait,
                  by=list(metadata_soil_its$tree_urban_type,
                          metadata_soil_its$tree_edge_type), FUN= "mean")
seits <- aggregate(fungal_trait,
                by=list(metadata_soil_its$tree_urban_type,
                        metadata_soil_its$tree_edge_type), FUN= standard_error)
mean16s <- aggregate(bacterial_trait,
                  by=list(metadata_soil_16s$tree_urban_type,
                          metadata_soil_16s$tree_edge_type), FUN= "mean")
se16s <- aggregate(bacterial_trait,
                by=list(metadata_soil_16s$tree_urban_type,
                        metadata_soil_16s$tree_edge_type), FUN= standard_error)

vdata <- cbind(meanits[,1:2], meanits$perc_wood_saprotroph, seits$perc_wood_saprotroph, 
               mean16s$perc_cellulolytic, se16s$perc_cellulolytic,
               mean16s$perc_lignolytic, se16s$perc_lignolytic)
colnames(vdata) <- c("Site", "Edge", "woodsap", "woodsap_se", "cell", "cell_se", 
                     "lig", "lig_se")
vdata$Edge <- factor(vdata$Edge, 
                     levels = c("Urban Street Tree", "Forest Edge", 
                                "Forest Interior"))

vdata_long <- as.data.frame(matrix(data = NA, nrow = 15, ncol = 5))
colnames(vdata_long) <- c("Site", "Edge", "Guild", "Abundance", "SE")
vdata_long$Site <- rep(vdata$Site, 3)
vdata_long$Edge <- rep(vdata$Edge, 3)
vdata_long$Guild <- c(rep("Fungal wood decomposers", 5), 
                      rep("Cellulolytic bacteria", 5), 
                      rep("Lignolytic bacteria", 5))
vdata_long$Abundance <- c(vdata$woodsap, vdata$cell, vdata$lig)
vdata_long$SE <- c(vdata$woodsap_se, vdata$cell_se, vdata$lig_se)

vdata_long$site_edge <- paste0(vdata_long$Site, " ", vdata_long$Edge)
vdata_long$site_edge <- gsub("Urban Street Tree Urban Street Tree", "Urban Street Tree",
                             vdata_long$site_edge)
vdata_long$site_edge <- factor(vdata_long$site_edge, levels = c("Urban Street Tree", 
                                                      "Urban Forest Edge", 
                                                      "Urban Forest Interior", 
                                                      "Rural Forest Edge", 
                                                      "Rural Forest Interior"))

vdata_long$Guild <- factor(vdata_long$Guild, levels = c("Cellulolytic bacteria", "Lignolytic bacteria", "Fungal wood decomposers"))
vdata_long$tukey <- c("b[\"\\u25A0\"]", 
  "b[\"\\u25A0\"]", 
  "b[\"\\u25A0\"]", 
  "b[\"\\u25A0\"]", 
  "a[\"\\u25A0\"]",
  "b[\"\\u25CF\"]", 
  "b[\"\\u25CF\"]", 
  "b[\"\\u25CF\"]", 
  "b[\"\\u25CF\"]", 
  "a[\"\\u25CF\"]", 
  "b[\"\\u25B2\"]", 
  "b[\"\\u25B2\"]", 
  "b[\"\\u25B2\"]", 
  "b[\"\\u25B2\"]", 
  "a[\"\\u25B2\"]"
)
vdata_long$site_edge <- gsub(" ", "\n", vdata_long$site_edge)
vdata_long$site_edge <- factor(vdata_long$site_edge, 
                               levels = c("Urban\nStreet\nTree", 
                                          "Urban\nForest\nEdge",
                                          "Urban\nForest\nInterior",
                                          "Rural\nForest\nEdge",
                                          "Rural\nForest\nInterior"))

woodsap_v_frag_bg <- ggplot() +
   geom_line(data = vdata_long, 
             aes(x=site_edge, y=Abundance, shape = Guild, group = Guild), 
            linetype = 2) +
  geom_point(data = vdata_long, 
             aes(x=site_edge, y=Abundance, color=site_edge, shape = Guild, 
                 group = Guild), size=5) + 
  geom_errorbar(data = vdata_long, 
                aes(x=site_edge, ymin=Abundance-SE, ymax=Abundance+SE, 
                    color=site_edge), width = 0.2) +
  geom_text(data=vdata_long, family = "Arial",
            aes(x = site_edge, y = Abundance, label = tukey, hjust = 2.2, vjust = 0.5,       # Adjust text to the left of the points
            nudge_x = -0.5), size = 3, parse = TRUE) +
  # geom_text_repel(data=vdata_long, family = "Arial", direction = "y", segment.color = NA,
  #           aes(x = site_edge, y = Abundance, label = tukey, hjust = 2.1, vjust = 1        # Adjust text to the left of the points
  #           ), size = 3, parse = TRUE) +
  # geom_point(data = metadata_soil_its, 
  #            aes(x = site_edge, y = perc_ecm, color = site_edge), 
  #            position = position_jitter(width = 0.2), size = 2, alpha = 0.2, stroke = NA) +
  theme_classic() + labs (y="log(Functional group abundance (%))", x="", shape = "") +
  theme(text = element_text(family = "Arial"),
        axis.title.x = element_text(size = 10, color="black"), 
        axis.title.y = element_text(size = 10), 
        axis.text.x = element_text(size = 10, color = "black"), 
        axis.text.y = element_text(size = 10, color = "black"), 
        legend.text = element_text(size = 7), 
        legend.title = element_text(size = 1), 
        plot.title = element_text(size = 10), legend.position = c(0.5, 0.8)) +
  scale_color_manual(values = pal, guide = "none") + 
  scale_fill_manual(values = pal, guide = "none") + 
  ggtitle("Wood decomposers in soil")

woodsap_v_frag_bg
ggsave("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Figures/figure2_soil_woodrots_fragmentation.pdf", device = cairo_pdf, plot = woodsap_v_frag_bg, dpi = 300, width = 5.75, height = 3.5, units = "in")

metadata_soil_its$site_edge <- paste0(metadata_soil_its$tree_urban_type, " ", metadata_soil_its$tree_edge_type)
metadata_soil_its$site_edge[which(metadata_soil_its$site_edge == "Urban Street Tree Urban Street Tree")] <- "Urban Street Tree"

metadata_soil_16s$site_edge <- paste0(metadata_soil_16s$tree_urban_type, " ", metadata_soil_16s$tree_edge_type)
metadata_soil_16s$site_edge[which(metadata_soil_16s$site_edge == "Urban Street Tree Urban Street Tree")] <- "Urban Street Tree"

# model1 <- lmer(formula = perc_wood_saprotroph ~ site_edge + (1|plot_name/tree_id),
#               data = metadata_soil_its)
# summary(emmeans(model1, pairwise ~ site_edge))
model1 <- lme(formula(perc_wood_saprotroph ~ site_edge), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML", data = metadata_soil_its)
summary(model1)
summary(emmeans(model1, pairwise ~ site_edge))

# model2 <- lmer(formula = perc_cellulolytic ~ site_edge + (1|plot_name/tree_id),
#               data = metadata_soil_16s)
# summary(emmeans(model2, pairwise ~ site_edge))
model1 <- lme(formula(perc_cellulolytic ~ site_edge), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML", data = metadata_soil_16s)
summary(model1)
summary(emmeans(model1, pairwise ~ site_edge))


# model3 <- lmer(formula = perc_lignolytic ~ site_edge + (1|plot_name/tree_id),
#               data = metadata_soil_16s)
# summary(emmeans(model3, pairwise ~ site_edge))
model1 <- lme(formula(perc_lignolytic ~ site_edge), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML", data = metadata_soil_16s)
summary(model1)
summary(emmeans(model1, pairwise ~ site_edge))
```

Figure 1c
```{r fungi wood saprotroph leaf dfb}
metadata_leaf_its <- metadata_its[which(metadata_its$sample_type == "Leaf"),]
metadata_leaf_16s <- metadata_16s_200[which(metadata_16s_200$sample_type == "Leaf"),]
length(unique(metadata_leaf_16s$tree_id))

sooty_mold <- metadata_leaf_its[,which(colnames(metadata_leaf_its) %in% c("tree_dist_to_boston_km", "perc_sooty_mold"))]
sooty_mold$Guild <- "Sooty Molds"
colnames(sooty_mold)[2] <- "abundance"
plant_path <- metadata_leaf_16s[,which(colnames(metadata_leaf_16s) %in% c("tree_dist_to_boston_km", "perc_plant_pathogen"))]
plant_path$Guild <- "Bacterial Plant Pathogens"
colnames(plant_path)[2] <- "abundance"

plot_data <- rbind(sooty_mold, plant_path)
plot_data$Guild <- factor(plot_data$Guild, 
                          levels = c("Bacterial Plant Pathogens", 
                                     "Sooty Molds"))

epiphyte_vs_dfb_ag <- ggplot(plot_data) + 
  xlab("log(Distance from Boston (km))") + 
  ylab("log(Functional group abundance (%))") + 
  geom_smooth(method = "lm", alpha = 0.12, aes(x = log(tree_dist_to_boston_km+1),
                 y = abundance,
                 col = Guild, fill = Guild,
                 ymax = after_stat(y + se * sqrt(length(y))),
                  ymin = after_stat(y - se * sqrt(length(y))))) +
  geom_smooth(method = "lm", se = FALSE, aes(x = log(tree_dist_to_boston_km+1),
                 y = abundance,
                 col = Guild, fill = Guild)) +
  geom_point(aes(x = log(tree_dist_to_boston_km+1),
                 y = abundance,
                 col = Guild), 
             alpha = 0.5, stroke = NA, size = 2) +
  theme_classic() + 
  theme(text = element_text(family = "Arial"),
        axis.title.x = element_text(size = 10, color="black"), 
        axis.title.y = element_text(size = 10), 
        axis.text.x = element_text(size = 10, color = "black"), 
        axis.text.y = element_text(size = 10, color = "black"), 
        legend.text = element_text(size = 7), 
        legend.title = element_text(size = 1), 
        plot.title = element_text(size = 10), legend.position = "none") +
  scale_fill_manual(values = rev(plot_pal)) + 
  scale_color_manual(values = rev(plot_pal)) + 
  labs(fill = "Microbial Guild", color = "Microbial Guild") +
  ggtitle("Plant pathogens in leaves")
epiphyte_vs_dfb_ag

ggsave("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Figures/figure3_leaf_plantpaths_dfb.pdf", device = cairo_pdf, plot = epiphyte_vs_dfb_ag, dpi = 300, width = 5.75, height = 3.5, units = "in")

 # model1 <- lmer(formula = perc_sooty_mold ~ log(tree_dist_to_boston_km+1) * as.numeric(tree_distance_from_edge) + (1|plot_name/tree_id),
#               data = metadata_leaf_its)
# summary(model1)
# r.squaredGLMM(model1)

model1 <- lme(formula(perc_sooty_mold ~ log(tree_dist_to_boston_km+1) * as.numeric(tree_distance_from_edge)), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML", data = metadata_leaf_its)
anova(model1)
r.squaredGLMM(model1)

metadata_leaf_16s$tree_distance_from_edge <- as.numeric(metadata_leaf_16s$tree_distance_from_edge)
# model1 <- lmer(formula = perc_plant_pathogen ~ log(tree_dist_to_boston_km+1) * as.numeric(tree_distance_from_edge) + (1|plot_name/tree_id),
#               data = metadata_leaf_16s)
# summary(model1)
# r.squaredGLMM(model1)
model1 <- lme(formula(perc_plant_pathogen ~ log(tree_dist_to_boston_km+1) * as.numeric(tree_distance_from_edge)), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML", data = metadata_leaf_16s)
anova(model1)
r.squaredGLMM(model1)
```

Figure 1d
```{r soil 0-15cm ectos and wood rots heatmap}
metadata_soil_its <- metadata_its[which(metadata_its$sample_type == "Soil, 0-15 cm"),]
metadata_soil_16s <- metadata_16s_200[which(metadata_16s_200$sample_type == "Soil, 0-15 cm"),]

metadata_leaf_its <- metadata_its[which(metadata_its$sample_type == "Leaf"),]
metadata_leaf_16s <- metadata_16s_200[which(metadata_16s_200$sample_type == "Leaf"),]

fungal_trait_soil_its <- metadata_soil_its[,colnames(metadata_soil_its) %in%
                                        c(fungal_trait_names)]
bacterial_trait_soil_16s <- metadata_soil_16s[,colnames(metadata_soil_16s) %in% c(bacterial_trait_names)]

fungal_trait_leaf_its <- metadata_leaf_its[,colnames(metadata_leaf_its) %in%
                                        c(fungal_trait_names)]
bacterial_trait_leaf_16s <- metadata_leaf_16s[,colnames(metadata_leaf_16s) %in% c(bacterial_trait_names)]
func_soil_its <- fungal_trait_soil_its
func_leaf_its <- fungal_trait_leaf_its

func_soil_its <- func_soil_its[,which(colnames(func_soil_its) %in% c("perc_ecm", "perc_wood_saprotroph"))]
func_leaf_its <- func_leaf_its[,which(colnames(func_leaf_its) %in% c("perc_epiphyte", "perc_sooty_mold"))]

colnames(func_soil_its) <- c("ECM","Wood decomposers")
colnames(func_soil_its) <- paste0(colnames(func_soil_its), " in soil")
colnames(func_leaf_its) <- c("Fungal epiphytes", "Sooty molds")
colnames(func_leaf_its) <- paste0(colnames(func_leaf_its), " in leaves")

func_soil_16s <- bacterial_trait_soil_16s
func_leaf_16s <- bacterial_trait_leaf_16s

func_soil_16s <- func_soil_16s[,which(colnames(func_soil_16s) %in% c("perc_cellulolytic", "perc_lignolytic"))]
func_leaf_16s <- as.data.frame(func_leaf_16s[,which(colnames(func_leaf_16s) %in% c("perc_plant_pathogen"))])

colnames(func_soil_16s) <- c("Cellulolytic bacteria", "Lignolytic bacteria")
colnames(func_soil_16s) <- paste0(colnames(func_soil_16s), " in soil")
colnames(func_leaf_16s) <- c("Bacterial plant pathogens")
colnames(func_leaf_16s) <- paste0(colnames(func_leaf_16s), " in leaves")


metadata_soil_heatmap_its <- metadata_soil_its[,c(17,18,20,23,30:36,38,49,53:55,59:61,63:65,67:76)]
metadata_soil_heatmap_16s <- metadata_soil_16s[,c(17,18,20,23,30:36,38,49,53:55,59:61,63:65,67:76)]
metadata_leaf_heatmap_its <- metadata_leaf_its[,c(17,18,20,23,30:36,38,49,53:55,59:61,63:65,67:76)]
metadata_leaf_heatmap_16s <- metadata_leaf_16s[,c(17,18,20,23,30:36,38,49,53:55,59:61,63:65,67:76)]
colnames(metadata_soil_heatmap_its) <- c("Tree distance from edge", 
                                         "Tree distance from Boston", 
                                         "Tree DBH", "Tree growth rate",
                                         "Soil temperature", 
                                         "Soil bulk density", 
                                         "Soil pH", "Soil moisture", 
                                         "Soil organic matter", 
                                         "Soil available NH<sub>4</sub><sup>+</sup>",
                                         "Soil available NO<sub>3</sub><sup>-</sup>", 
                                         "Normalized root biomass", 
                                         "Soil soluble salt",
                                         "NO concentration",
                                         "NO<sub>2</sub> concentration",
                                         "NO<sub>x</sub> concentration",
                                         "NH<sub>4</sub><sup>+</sup> annual throughfall",
                                         "NO<sub>3</sub><sup>-</sup> annual throughfall",
                                         "Total inorganic N annual throughfall",
                                         "O<sub>3</sub> concentration",
                                         "Foliar N %",
                                         "Foliar C:N",
                                         "Ca<sup>2+</sup> annual wet deposition",
                                         "Cl<sup>-</sup> annual wet deposition",
                                         "H<sup>+</sup> annual wet deposition",
                                         "K<sup>+</sup> annual wet deposition",
                                         "Mg<sup>2+</sup> annual wet deposition",
                                         "Na<sup>+</sup> annual wet deposition",
                                         "NH<sub>4</sub><sup>+</sup> annual wet deposition",
                                         "NO<sub>3</sub><sup>-</sup> annual wet deposition",
                                         "SO<sub>4</sub><sup>2-</sup> annual wet deposition",
                                         "Total inorganic N annual wet deposition")
colnames(metadata_soil_heatmap_16s) <- c("Tree distance from edge", 
                                         "Tree distance from Boston", 
                                         "Tree DBH", "Tree growth rate",
                                         "Soil temperature", 
                                         "Soil bulk density", 
                                         "Soil pH", "Soil moisture", 
                                         "Soil organic matter", 
                                         "Soil available NH<sub>4</sub><sup>+</sup>",
                                         "Soil available NO<sub>3</sub><sup>-</sup>", 
                                         "Normalized root biomass", 
                                         "Soil soluble salt",
                                         "NO concentration",
                                         "NO<sub>2</sub> concentration",
                                         "NO<sub>x</sub> concentration",
                                         "NH<sub>4</sub><sup>+</sup> annual throughfall",
                                         "NO<sub>3</sub><sup>-</sup> annual throughfall",
                                         "Total inorganic N annual throughfall",
                                         "O<sub>3</sub> concentration",
                                         "Foliar N %",
                                         "Foliar C:N",
                                         "Ca<sup>2+</sup> annual wet deposition",
                                         "Cl<sup>-</sup> annual wet deposition",
                                         "H<sup>+</sup> annual wet deposition",
                                         "K<sup>+</sup> annual wet deposition",
                                         "Mg<sup>2+</sup> annual wet deposition",
                                         "Na<sup>+</sup> annual wet deposition",
                                         "NH<sub>4</sub><sup>+</sup> annual wet deposition",
                                         "NO<sub>3</sub><sup>-</sup> annual wet deposition",
                                         "SO<sub>4</sub><sup>2-</sup> annual wet deposition",
                                         "Total inorganic N annual wet deposition")
colnames(metadata_leaf_heatmap_its) <- c("Tree distance from edge", 
                                         "Tree distance from Boston", 
                                         "Tree DBH", "Tree growth rate",
                                         "Soil temperature", 
                                         "Soil bulk density", 
                                         "Soil pH", "Soil moisture", 
                                         "Soil organic matter", 
                                         "Soil available NH<sub>4</sub><sup>+</sup>",
                                         "Soil available NO<sub>3</sub><sup>-</sup>", 
                                         "Normalized root biomass", 
                                         "Soil soluble salt",
                                         "NO concentration",
                                         "NO<sub>2</sub> concentration",
                                         "NO<sub>x</sub> concentration",
                                         "NH<sub>4</sub><sup>+</sup> annual throughfall",
                                         "NO<sub>3</sub><sup>-</sup> annual throughfall",
                                         "Total inorganic N annual throughfall",
                                         "O<sub>3</sub> concentration",
                                         "Foliar N %",
                                         "Foliar C:N",
                                         "Ca<sup>2+</sup> annual wet deposition",
                                         "Cl<sup>-</sup> annual wet deposition",
                                         "H<sup>+</sup> annual wet deposition",
                                         "K<sup>+</sup> annual wet deposition",
                                         "Mg<sup>2+</sup> annual wet deposition",
                                         "Na<sup>+</sup> annual wet deposition",
                                         "NH<sub>4</sub><sup>+</sup> annual wet deposition",
                                         "NO<sub>3</sub><sup>-</sup> annual wet deposition",
                                         "SO<sub>4</sub><sup>2-</sup> annual wet deposition",
                                         "Total inorganic N annual wet deposition")
colnames(metadata_leaf_heatmap_16s) <- c("Tree distance from edge", 
                                         "Tree distance from Boston", 
                                         "Tree DBH", "Tree growth rate",
                                         "Soil temperature", 
                                         "Soil bulk density", 
                                         "Soil pH", "Soil moisture", 
                                         "Soil organic matter", 
                                         "Soil available NH<sub>4</sub><sup>+</sup>",
                                         "Soil available NO<sub>3</sub><sup>-</sup>", 
                                         "Normalized root biomass", 
                                         "Soil soluble salt",
                                         "NO concentration",
                                         "NO<sub>2</sub> concentration",
                                         "NO<sub>x</sub> concentration",
                                         "NH<sub>4</sub><sup>+</sup> annual throughfall",
                                         "NO<sub>3</sub><sup>-</sup> annual throughfall",
                                         "Total inorganic N annual throughfall",
                                         "O<sub>3</sub> concentration",
                                         "Foliar N %",
                                         "Foliar C:N",
                                         "Ca<sup>2+</sup> annual wet deposition",
                                         "Cl<sup>-</sup> annual wet deposition",
                                         "H<sup>+</sup> annual wet deposition",
                                         "K<sup>+</sup> annual wet deposition",
                                         "Mg<sup>2+</sup> annual wet deposition",
                                         "Na<sup>+</sup> annual wet deposition",
                                         "NH<sub>4</sub><sup>+</sup> annual wet deposition",
                                         "NO<sub>3</sub><sup>-</sup> annual wet deposition",
                                         "SO<sub>4</sub><sup>2-</sup> annual wet deposition",
                                         "Total inorganic N annual wet deposition")
# Make coefficient matrix
Result_soil_its <- NULL
for (i in 1:ncol(func_soil_its)){
  estimate_soil_its <- NULL
  for (j in 1:ncol(metadata_soil_heatmap_its)){
    cor_soil_its <- cor.test(func_soil_its[,i],metadata_soil_heatmap_its[,j], method="p")
    estimate_soil_its <- c(estimate_soil_its, cor_soil_its$estimate)
  }
  Result_soil_its <- rbind(Result_soil_its, estimate_soil_its)
}
colnames(Result_soil_its) <- colnames(metadata_soil_heatmap_its)
rownames(Result_soil_its) <- colnames(func_soil_its) 

Result_leaf_its <- NULL
for (i in 1:ncol(func_leaf_its)){
  estimate_leaf_its <- NULL
  for (j in 1:ncol(metadata_leaf_heatmap_its)){
    cor_leaf_its <- cor.test(func_leaf_its[,i],metadata_leaf_heatmap_its[,j], method="p")
    estimate_leaf_its <- c(estimate_leaf_its, cor_leaf_its$estimate)
  }
  Result_leaf_its <- rbind(Result_leaf_its, estimate_leaf_its)
}
colnames(Result_leaf_its) <- colnames(metadata_leaf_heatmap_its)
rownames(Result_leaf_its) <- colnames(func_leaf_its) 

Result_soil_16s <- NULL
for (i in 1:ncol(func_soil_16s)){
  estimate_soil_16s <- NULL
  for (j in 1:ncol(metadata_soil_heatmap_16s)){
    cor_soil_16s <- cor.test(func_soil_16s[,i],metadata_soil_heatmap_16s[,j], method="p")
    estimate_soil_16s <- c(estimate_soil_16s, cor_soil_16s$estimate)
  }
  Result_soil_16s <- rbind(Result_soil_16s, estimate_soil_16s)
}
colnames(Result_soil_16s) <- colnames(metadata_soil_heatmap_16s)
rownames(Result_soil_16s) <- colnames(func_soil_16s) 

Result_leaf_16s <- NULL
  estimate_leaf_16s <- NULL
  for (j in 1:ncol(metadata_leaf_heatmap_16s)){
    cor_leaf_16s <- cor.test(func_leaf_16s[,1],metadata_leaf_heatmap_16s[,j], method="p")
    estimate_leaf_16s <- c(estimate_leaf_16s, cor_leaf_16s$estimate)
  }
  Result_leaf_16s <- rbind(Result_leaf_16s, estimate_leaf_16s)
colnames(Result_leaf_16s) <- colnames(metadata_leaf_heatmap_16s)
rownames(Result_leaf_16s) <- "Bacterial plant pathogens in leaves" 

# Make Pval matrix
Result.Pval_soil_its <- NULL
for (i in 1:ncol(func_soil_its)){
  Pval_soil_its <- NULL
  for (j in 1:ncol(metadata_soil_heatmap_its)){
    cor_soil_its <- cor.test(func_soil_its[,i],metadata_soil_heatmap_its[,j], method="p")
    Pval_soil_its <- c(Pval_soil_its, cor_soil_its$p.value)
  }
  Result.Pval_soil_its <- rbind (Result.Pval_soil_its, Pval_soil_its)
}
colnames(Result.Pval_soil_its) <- colnames(metadata_soil_heatmap_its)
rownames(Result.Pval_soil_its) <- colnames(func_soil_its) 

Result.Pval_leaf_its <- NULL
for (i in 1:ncol(func_leaf_its)){
  Pval_leaf_its <- NULL
  for (j in 1:ncol(metadata_leaf_heatmap_its)){
    cor_leaf_its <- cor.test(func_leaf_its[,i],metadata_leaf_heatmap_its[,j], method="p")
    Pval_leaf_its <- c(Pval_leaf_its, cor_leaf_its$p.value)
  }
  Result.Pval_leaf_its <- rbind (Result.Pval_leaf_its, Pval_leaf_its)
}
colnames(Result.Pval_leaf_its) <- colnames(metadata_leaf_heatmap_its)
rownames(Result.Pval_leaf_its) <- colnames(func_leaf_its) 

Result.Pval_soil_16s <- NULL
for (i in 1:ncol(func_soil_16s)){
  Pval_soil_16s <- NULL
  for (j in 1:ncol(metadata_soil_heatmap_16s)){
    cor_soil_16s <- cor.test(func_soil_16s[,i],metadata_soil_heatmap_16s[,j], method="p")
    Pval_soil_16s <- c(Pval_soil_16s, cor_soil_16s$p.value)
  }
  Result.Pval_soil_16s <- rbind (Result.Pval_soil_16s, Pval_soil_16s)
}
colnames(Result.Pval_soil_16s) <- colnames(metadata_soil_heatmap_16s)
rownames(Result.Pval_soil_16s) <- colnames(func_soil_16s) 

Result.Pval_leaf_16s <- NULL
  Pval_leaf_16s <- NULL
  for (j in 1:ncol(metadata_leaf_heatmap_16s)){
    cor_leaf_16s <- cor.test(func_leaf_16s[,1],metadata_leaf_heatmap_16s[,j], method="p")
    Pval_leaf_16s <- c(Pval_leaf_16s, cor_leaf_16s$p.value)
  }
  Result.Pval_leaf_16s <- rbind (Result.Pval_leaf_16s, Pval_leaf_16s)
colnames(Result.Pval_leaf_16s) <- colnames(metadata_leaf_heatmap_16s)
rownames(Result.Pval_leaf_16s) <- "Bacterial plant pathogens in leaves"

# Make Asterisk matrix
Result.Pval.combined_soil_its <- Result.Pval_soil_its
Result.asterisk_soil_its <- Result.Pval.combined_soil_its
for (k in 1:ncol(Result.Pval.combined_soil_its)){
  for (i in 1:nrow(Result.Pval.combined_soil_its)){
    if(!is.na(Result.Pval.combined_soil_its[i,k])){
      if (Result.Pval.combined_soil_its[i,k] < 0.001){
        Result.asterisk_soil_its[i,k] <- "***"
      } else if (Result.Pval.combined_soil_its[i,k] < 0.01){
        Result.asterisk_soil_its[i,k] <- "**"
      } else if (Result.Pval.combined_soil_its[i,k] < 0.05){
        Result.asterisk_soil_its[i,k] <- "*"
      } else {
        Result.asterisk_soil_its[i,k]<- " "
      }
    } 
  }
}

Result.Pval.combined_leaf_its <- Result.Pval_leaf_its
Result.asterisk_leaf_its <- Result.Pval.combined_leaf_its
for (k in 1:ncol(Result.Pval.combined_leaf_its)){
  for (i in 1:nrow(Result.Pval.combined_leaf_its)){
    if(!is.na(Result.Pval.combined_leaf_its[i,k])){
      if (Result.Pval.combined_leaf_its[i,k] < 0.001){
        Result.asterisk_leaf_its[i,k] <- "***"
      } else if (Result.Pval.combined_leaf_its[i,k] < 0.01){
        Result.asterisk_leaf_its[i,k] <- "**"
      } else if (Result.Pval.combined_leaf_its[i,k] < 0.05){
        Result.asterisk_leaf_its[i,k] <- "*"
      } else {
        Result.asterisk_leaf_its[i,k]<- " "
      }
    } 
  }
}

Result.Pval.combined_soil_16s <- Result.Pval_soil_16s
Result.asterisk_soil_16s <- Result.Pval.combined_soil_16s
for (k in 1:ncol(Result.Pval.combined_soil_16s)){
  for (i in 1:nrow(Result.Pval.combined_soil_16s)){
    if(!is.na(Result.Pval.combined_soil_16s[i,k])){
      if (Result.Pval.combined_soil_16s[i,k] < 0.001){
        Result.asterisk_soil_16s[i,k] <- "***"
      } else if (Result.Pval.combined_soil_16s[i,k] < 0.01){
        Result.asterisk_soil_16s[i,k] <- "**"
      } else if (Result.Pval.combined_soil_16s[i,k] < 0.05){
        Result.asterisk_soil_16s[i,k] <- "*"
      } else {
        Result.asterisk_soil_16s[i,k]<- " "
      }
    } 
  }
}

Result.Pval.combined_leaf_16s <- Result.Pval_leaf_16s
Result.asterisk_leaf_16s <- Result.Pval.combined_leaf_16s
for (k in 1:ncol(Result.Pval.combined_leaf_16s)){
  for (i in 1:nrow(Result.Pval.combined_leaf_16s)){
    if(!is.na(Result.Pval.combined_leaf_16s[i,k])){
      if (Result.Pval.combined_leaf_16s[i,k] < 0.001){
        Result.asterisk_leaf_16s[i,k] <- "***"
      } else if (Result.Pval.combined_leaf_16s[i,k] < 0.01){
        Result.asterisk_leaf_16s[i,k] <- "**"
      } else if (Result.Pval.combined_leaf_16s[i,k] < 0.05){
        Result.asterisk_leaf_16s[i,k] <- "*"
      } else {
        Result.asterisk_leaf_16s[i,k]<- " "
      }
    } 
  }
}
# Combine netowrk propeties + others
Result.combined_soil_its <- Result_soil_its
melt_soil_its <- melt(Result.combined_soil_its)
melt.asterisk_soil_its <- melt(Result.asterisk_soil_its)

Result.combined_leaf_its <- Result_leaf_its
melt_leaf_its <- melt(Result.combined_leaf_its)
melt.asterisk_leaf_its <- melt(Result.asterisk_leaf_its)

Result.combined_soil_16s <- Result_soil_16s
melt_soil_16s <- melt(Result.combined_soil_16s)
melt.asterisk_soil_16s <- melt(Result.asterisk_soil_16s)

Result.combined_leaf_16s <- Result_leaf_16s
melt_leaf_16s <- melt(Result.combined_leaf_16s)
melt.asterisk_leaf_16s <- melt(Result.asterisk_leaf_16s)

melt_all <- rbind(melt_soil_its, melt_soil_16s, melt_leaf_its, melt_leaf_16s)
melt_asterisk_all <- rbind(melt.asterisk_soil_its, melt.asterisk_soil_16s, 
                           melt.asterisk_leaf_its, melt.asterisk_leaf_16s)

# Visualize
vdata_its <- cbind(melt_all, melt_asterisk_all$value)
colnames(vdata_its) <- c("microbe", "environment", "coef", "Pval")

vdata_its$microbe <- factor(vdata_its$microbe, 
                            levels = c("ECM in soil",
                                       "Fungal epiphytes in leaves",
                                       "Wood decomposers in soil", 
                                       "Cellulolytic bacteria in soil", 
                                       "Lignolytic bacteria in soil",
                                       "Sooty molds in leaves",
                                       "Bacterial plant pathogens in leaves"))

order_of_env_factors <- vdata_its[which(vdata_its$microbe == "ECM in soil"),]

vdata_its$environment <- factor(vdata_its$environment, 
                            levels = order_of_env_factors$environment[order(order_of_env_factors$coef)])
library(ggtext)
heatmap_soilO <- ggplot(vdata_its) +
  geom_tile(aes(x=microbe, y=environment, fill=coef)) +
  geom_text(aes(x=microbe, y=environment, label = Pval), size = 3) +
  labs(x="",y="") + theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, hjust=1,vjust=0.3)) +
  theme(axis.text.x = element_text(angle = 90, hjust=1,vjust=0.3)) + 
  scale_fill_gradient2(low = "#003049", mid = "#FDF0D5", high = "#C00000", 
                       name = "Correlation\ncoefficient") +
  theme(text = element_text(family = "Arial"),
        axis.title.x = element_text(size = 8), 
        axis.title.y = element_markdown(size = 8), 
        axis.text.y = element_markdown(size = 8, color = "black"),
        axis.text.x = element_text(size = 8, color = "black"),
        legend.text = element_text(size = 8), 
        legend.title = element_text(size = 8), 
        plot.title = element_text(size = 8)) + 
  ggtitle("") + coord_fixed()
heatmap_soilO

ggsave("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Figures/heatmap_manuscript_fig1d_new.pdf", device = cairo_pdf, width = 5, height = 10, units = "in", dpi = 300)
```


Figure 2a
```{r pathogen leaf fragmentation}
metadata_leaf_its <- metadata_its[which(metadata_its$sample_type == "Leaf"),]
metadata_leaf_16s <- metadata_16s_200[which(metadata_16s_200$sample_type == "Leaf"),]

metadata_soil_its <- metadata_its[which(metadata_its$sample_type == "Soil, 0-15 cm"),]
metadata_soil_16s <- metadata_16s_200[which(metadata_16s_200$sample_type == "Soil, 0-15 cm"),]

fungal_trait_leaf <- metadata_leaf_its[, colnames(metadata_leaf_its) %in% 
                                    fungal_trait_names]
bacterial_trait_leaf <- metadata_leaf_16s[, colnames(metadata_leaf_16s) %in% 
                                    bacterial_trait_names]
fungal_trait_soil <- metadata_soil_its[, colnames(metadata_soil_its) %in% 
                                    fungal_trait_names]
bacterial_trait_soil <- metadata_soil_16s[, colnames(metadata_soil_16s) %in% 
                                    bacterial_trait_names]
standard_error <- function(x) sd(x) / sqrt(length(x)) 

meanits_leaf <- aggregate(fungal_trait_leaf,
                  by=list(metadata_leaf_its$tree_urban_type,
                          metadata_leaf_its$tree_edge_type), FUN= "mean")
seits_leaf <- aggregate(fungal_trait_leaf,
                by=list(metadata_leaf_its$tree_urban_type,
                        metadata_leaf_its$tree_edge_type), FUN= standard_error)
mean16s_leaf <- aggregate(bacterial_trait_leaf,
                  by=list(metadata_leaf_16s$tree_urban_type,
                          metadata_leaf_16s$tree_edge_type), FUN= "mean")
se16s_leaf <- aggregate(bacterial_trait_leaf,
                by=list(metadata_leaf_16s$tree_urban_type,
                        metadata_leaf_16s$tree_edge_type), FUN= standard_error)

meanits_soil <- aggregate(fungal_trait_soil,
                  by=list(metadata_soil_its$tree_urban_type,
                          metadata_soil_its$tree_edge_type), FUN= "mean")
seits_soil <- aggregate(fungal_trait_soil,
                by=list(metadata_soil_its$tree_urban_type,
                        metadata_soil_its$tree_edge_type), FUN= standard_error)
mean16s_soil <- aggregate(bacterial_trait_soil,
                  by=list(metadata_soil_16s$tree_urban_type,
                          metadata_soil_16s$tree_edge_type), FUN= "mean")
se16s_soil <- aggregate(bacterial_trait_soil,
                by=list(metadata_soil_16s$tree_urban_type,
                        metadata_soil_16s$tree_edge_type), FUN= standard_error)

vdata <- cbind(meanits_leaf[,1:2], meanits_leaf$perc_animal_pathogen, 
               seits_leaf$perc_animal_pathogen, meanits_soil$perc_animal_pathogen, 
               seits_soil$perc_animal_pathogen, mean16s_soil$perc_zoonotic_pathogen,
               se16s_soil$perc_zoonotic_pathogen)
colnames(vdata) <- c("Site", "Edge", "animal_leaf", "animal_leaf_se", 
                     "animal_soil", "animal_soil_se", "zoo_soil", "zoo_soil_se")
vdata$Edge <- factor(vdata$Edge, 
                     levels = c("Urban Street Tree", "Forest Edge", 
                                "Forest Interior"))

vdata_long <- as.data.frame(matrix(data = NA, nrow = 15, ncol = 5))
colnames(vdata_long) <- c("Site", "Edge", "Guild", "Abundance", "SE")
vdata_long$Site <- rep(vdata$Site, 3)
vdata_long$Edge <- rep(vdata$Edge, 3)
vdata_long$Guild <- c(rep("Fungal animal pathogens in leaves", 5),
                      rep("Fungal animal pathogens in soil", 5),
                      rep("Bacterial zoonotic pathogens in soil", 5))
vdata_long$Guild <- factor(vdata_long$Guild, 
                           levels = c("Fungal animal pathogens in leaves", 
                                      "Bacterial zoonotic pathogens in soil",
                                      "Fungal animal pathogens in soil"))
vdata_long$Abundance <- c(vdata$animal_leaf, vdata$animal_soil, vdata$zoo_soil)
vdata_long$SE <- c(vdata$animal_leaf_se, vdata$animal_soil_se, vdata$zoo_soil_se)

vdata_long$site_edge <- paste0(vdata_long$Site, " ", vdata_long$Edge)
vdata_long$site_edge <- gsub("Urban Street Tree Urban Street Tree", "Urban Street Tree",
                             vdata_long$site_edge)
vdata_long$site_edge <- factor(vdata_long$site_edge, levels = c("Urban Street Tree", 
                                                      "Urban Forest Edge", 
                                                      "Urban Forest Interior", 
                                                      "Rural Forest Edge", 
                                                      "Rural Forest Interior"))

vdata_long$tukey <- c(
  "ab[\"\\u25CF\"]", 
  "b[\"\\u25CF\"]", 
  "b[\"\\u25CF\"]", 
  "ab[\"\\u25CF\"]", 
  "a[\"\\u25CF\"]", 
  "ab[\"\\u25A0\"]", 
  "ab[\"\\u25A0\"]", 
  "b[\"\\u25A0\"]", 
  "ab[\"\\u25A0\"]", 
  "a[\"\\u25A0\"]", 
  "b[\"\\u25B2\"]", 
  "b[\"\\u25B2\"]", 
  "b[\"\\u25B2\"]", 
  "b[\"\\u25B2\"]", 
  "a[\"\\u25B2\"]"
)
vdata_long$site_edge <- gsub(" ", "\n", vdata_long$site_edge)
vdata_long$site_edge <- factor(vdata_long$site_edge, 
                               levels = c("Urban\nStreet\nTree", 
                                          "Urban\nForest\nEdge",
                                          "Urban\nForest\nInterior",
                                          "Rural\nForest\nEdge",
                                          "Rural\nForest\nInterior"))

woodsap_v_frag_bg <- ggplot() +
   geom_line(data = vdata_long, 
             aes(x=site_edge, y=Abundance, shape = Guild, group = Guild), 
            linetype = 2) +
  geom_point(data = vdata_long, 
             aes(x=site_edge, y=Abundance, color=site_edge, shape = Guild, 
                 group = Guild), size=5) + 
  geom_errorbar(data = vdata_long, 
                aes(x=site_edge, ymin=Abundance-SE, ymax=Abundance+SE, 
                    color=site_edge), width = 0.2) +
  geom_text_repel(data = vdata_long, family = "Arial", direction = "y",
             aes(x=site_edge, y=Abundance, label = tukey), 
            hjust = 1, nudge_x = -0.1, nudge_y = 0.5,   # Align left of points
            size = 3, parse = TRUE, segment.color = NA) +
  # geom_point(data = metadata_leaf_its, 
  #            aes(x = site_edge, y = perc_ecm, color = site_edge), 
  #            position = position_jitter(width = 0.2), size = 2, alpha = 0.2, stroke = NA) +
  theme_classic() + labs (y="Functional group abundance (%)", x="", shape = "") +
  theme(text = element_text(family = "Arial"),
        axis.title.x = element_text(size = 9.5, color="black"), 
        axis.title.y = element_text(size = 9.5), 
        axis.text.x = element_text(size = 9.5, color = "black"), 
        axis.text.y = element_text(size = 9.5, color = "black"), 
        legend.text = element_text(size = 7), 
        legend.title = element_text(size = 1), 
        plot.title = element_text(size = 9.5), legend.position = c(0.5, 0.8)) +
  scale_color_manual(values = pal, guide = "none") + 
  scale_fill_manual(values = pal, guide = "none") + 
  ggtitle("Fungal animal pathogens in soil and leaves and bacterial zoonotic pathogens in soil")

woodsap_v_frag_bg
ggsave("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Figures/figure2_pathogens_fragmentation.pdf", device = cairo_pdf, plot = woodsap_v_frag_bg, dpi = 300, width = 6, height = 3.5, units = "in")


metadata_leaf_16s$site_edge <- paste0(metadata_leaf_16s$tree_urban_type, " ", metadata_leaf_16s$tree_edge_type)
metadata_leaf_16s$site_edge[which(metadata_leaf_16s$site_edge == "Urban Street Tree Urban Street Tree")] <- "Urban Street Tree"

metadata_leaf_its$site_edge <- paste0(metadata_leaf_its$tree_urban_type, " ", metadata_leaf_its$tree_edge_type)
metadata_leaf_its$site_edge[which(metadata_leaf_its$site_edge == "Urban Street Tree Urban Street Tree")] <- "Urban Street Tree"

metadata_soil_16s$site_edge <- paste0(metadata_soil_16s$tree_urban_type, " ", metadata_soil_16s$tree_edge_type)
metadata_soil_16s$site_edge[which(metadata_soil_16s$site_edge == "Urban Street Tree Urban Street Tree")] <- "Urban Street Tree"

metadata_soil_its$site_edge <- paste0(metadata_soil_its$tree_urban_type, " ", metadata_soil_its$tree_edge_type)
metadata_soil_its$site_edge[which(metadata_soil_its$site_edge == "Urban Street Tree Urban Street Tree")] <- "Urban Street Tree"

# model1 <- lmer(formula = perc_animal_pathogen ~ site_edge + (1|plot_name/tree_id),
#               data = metadata_leaf_its, REML = FALSE)
# summary(emmeans(model1, pairwise ~ site_edge))

model1 <- lme(formula(perc_animal_pathogen ~ site_edge), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML", data = metadata_leaf_its)
summary(model1)
summary(emmeans(model1, pairwise ~ site_edge))

# model1 <- lmer(formula = perc_animal_pathogen ~ site_edge + (1|plot_name/tree_id),
#               data = metadata_soil_its)
# summary(model1)
# summary(emmeans(model1, pairwise ~ site_edge))
model1 <- lme(formula(perc_animal_pathogen ~ site_edge), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML", data = metadata_soil_its)
summary(model1)
summary(emmeans(model1, pairwise ~ site_edge))

# model2 <- lmer(formula = perc_zoonotic_pathogen ~ site_edge + (1|plot_name/tree_id),
#               data = metadata_soil_16s)
# summary(emmeans(model2, pairwise ~ site_edge))
model2 <- lme(formula(perc_zoonotic_pathogen ~ site_edge), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML", data = metadata_soil_16s)
summary(model2)
summary(emmeans(model2, pairwise ~ site_edge))
```

Figure 2b
```{r pathogens leaf dfb}
metadata_leaf_16s <- metadata_16s_200[which(metadata_16s_200$sample_type == "Leaf"),]
# metadata_leaf_16s <- metadata_leaf_16s[which(metadata_leaf_16s$plot_name != "MP"),]
# metadata_leaf_16s <- metadata_leaf_16s[which(metadata_leaf_16s$plot_name != "WR"),]

plot_pal <- c("#003049")
epiphyte_vs_dfb_ag <- ggplot(metadata_leaf_16s) + 
  geom_smooth(method = "lm", alpha = 0.12, aes(x = log(tree_dist_to_boston_km+1),
                 y = perc_zoonotic_pathogen,
                 col = plot_pal, fill = plot_pal,
                 ymax = after_stat(y + se * sqrt(length(y))),
                  ymin = after_stat(y - se * sqrt(length(y))))) +
  geom_smooth(method = "lm", se = FALSE, aes(x = log(tree_dist_to_boston_km+1),
                 y = perc_zoonotic_pathogen,
                 col = plot_pal, fill = plot_pal)) +
  geom_point(aes(x = log(tree_dist_to_boston_km+1),
                 y = perc_zoonotic_pathogen,
                 col = plot_pal), 
             alpha = 0.5, stroke = NA, size = 2) +
  xlab("log(Distance from Boston (km))") + 
  ylab("Bacterial zoonotic pathogen abundance (%)") + 
  theme_classic() + 
  theme(text = element_text(family = "Arial"),
        axis.title.x = element_text(size = 9.5, color="black"), 
        axis.title.y = element_text(size = 9.5), 
        axis.text.x = element_text(size = 9.5, color = "black"), 
        axis.text.y = element_text(size = 9.5, color = "black"), 
        legend.text = element_text(size = 7), 
        legend.title = element_text(size = 1), 
        plot.title = element_text(size = 9.5), legend.position = "none") +
  scale_fill_manual(values = plot_pal) + 
  scale_color_manual(values = plot_pal) + 
  labs(fill = "Microbial Guild", color = "Microbial Guild") +
  ggtitle("Bacterial zoonotic pathogens in leaves")
epiphyte_vs_dfb_ag

ggsave("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Figures/figure2_pathogens_dfb.pdf", device = cairo_pdf, plot = epiphyte_vs_dfb_ag, dpi = 300, width = 6, height = 3.5, units = "in")



metadata_leaf_16s$tree_distance_from_edge <- as.numeric(metadata_leaf_16s$tree_distance_from_edge)
# model1 <- lmer(formula = perc_zoonotic_pathogen ~ log(tree_dist_to_boston_km+1) * as.numeric(tree_distance_from_edge) + (1|plot_name/tree_id),
#               data = metadata_leaf_16s)
# summary(model1)
# r.squaredGLMM(model1)

model2 <- lme(formula(perc_zoonotic_pathogen ~ log(tree_dist_to_boston_km+1) * as.numeric(tree_distance_from_edge)), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML", data = metadata_leaf_16s)
anova(model2)
r.squaredGLMM(model2)
```

Figure 2c
```{r soil 0-15cm ectos and wood rots heatmap}
metadata_soil_its <- metadata_its[which(metadata_its$sample_type == "Soil, 0-15 cm"),]
metadata_soil_16s <- metadata_16s_200[which(metadata_16s_200$sample_type == "Soil, 0-15 cm"),]

metadata_leaf_its <- metadata_its[which(metadata_its$sample_type == "Leaf"),]
metadata_leaf_16s <- metadata_16s_200[which(metadata_16s_200$sample_type == "Leaf"),]

fungal_trait_soil_its <- metadata_soil_its[,colnames(metadata_soil_its) %in%
                                        c(fungal_trait_names)]
bacterial_trait_soil_16s <- metadata_soil_16s[,colnames(metadata_soil_16s) %in% c(bacterial_trait_names)]

fungal_trait_leaf_its <- metadata_leaf_its[,colnames(metadata_leaf_its) %in%
                                        c(fungal_trait_names)]
bacterial_trait_leaf_16s <- metadata_leaf_16s[,colnames(metadata_leaf_16s) %in% c(bacterial_trait_names)]
func_soil_its <- fungal_trait_soil_its
func_leaf_its <- fungal_trait_leaf_its

func_soil_its <- as.data.frame(func_soil_its[,which(colnames(func_soil_its) %in% c("perc_animal_pathogen"))])
func_leaf_its <- as.data.frame(func_leaf_its[,which(colnames(func_leaf_its) %in% c("perc_animal_pathogen"))])

colnames(func_soil_its) <- c("Animal fungal pathogens")
colnames(func_soil_its) <- paste0(colnames(func_soil_its), " in soil")
colnames(func_leaf_its) <- c("Animal fungal pathogens")
colnames(func_leaf_its) <- paste0(colnames(func_leaf_its), "in leaves")

func_soil_16s <- bacterial_trait_soil_16s
func_leaf_16s <- bacterial_trait_leaf_16s

func_soil_16s <- as.data.frame(func_soil_16s[,which(colnames(func_soil_16s) %in% c("perc_zoonotic_pathogen"))])
func_leaf_16s <- as.data.frame(func_leaf_16s[,which(colnames(func_leaf_16s) %in% c("perc_zoonotic_pathogen"))])
colnames(func_soil_16s) <- c("Bacterial zoonotic pathogens")
colnames(func_soil_16s) <- paste0(colnames(func_soil_16s), " in soil")
colnames(func_leaf_16s) <- c("Bacterial hoonotic pathogens")
colnames(func_leaf_16s) <- paste0(colnames(func_leaf_16s), " in leaves")

metadata_soil_heatmap_its <- metadata_soil_its[,c(17,18,20,23,30:36,38,49,53:55,59:61,63:65,67:76)]
metadata_soil_heatmap_16s <- metadata_soil_16s[,c(17,18,20,23,30:36,38,49,53:55,59:61,63:65,67:76)]
metadata_leaf_heatmap_its <- metadata_leaf_its[,c(17,18,20,23,30:36,38,49,53:55,59:61,63:65,67:76)]
metadata_leaf_heatmap_16s <- metadata_leaf_16s[,c(17,18,20,23,30:36,38,49,53:55,59:61,63:65,67:76)]
colnames(metadata_soil_heatmap_its) <- c("Tree distance from edge", 
                                         "Tree distance from Boston", 
                                         "Tree DBH", "Tree growth rate",
                                         "Soil temperature", 
                                         "Soil bulk density", 
                                         "Soil pH", "Soil moisture", 
                                         "Soil organic matter", 
                                         "Soil available NH<sub>4</sub><sup>+</sup>",
                                         "Soil available NO<sub>3</sub><sup>-</sup>", 
                                         "Normalized root biomass", 
                                         "Soil soluble salt",
                                         "NO concentration",
                                         "NO<sub>2</sub> concentration",
                                         "NO<sub>x</sub> concentration",
                                         "NH<sub>4</sub><sup>+</sup> annual throughfall",
                                         "NO<sub>3</sub><sup>-</sup> annual throughfall",
                                         "Total inorganic N annual throughfall",
                                         "O<sub>3</sub> concentration",
                                         "Foliar N %",
                                         "Foliar C:N",
                                         "Ca<sup>2+</sup> annual wet deposition",
                                         "Cl<sup>-</sup> annual wet deposition",
                                         "H<sup>+</sup> annual wet deposition",
                                         "K<sup>+</sup> annual wet deposition",
                                         "Mg<sup>2+</sup> annual wet deposition",
                                         "Na<sup>+</sup> annual wet deposition",
                                         "NH<sub>4</sub><sup>+</sup> annual wet deposition",
                                         "NO<sub>3</sub><sup>-</sup> annual wet deposition",
                                         "SO<sub>4</sub><sup>2-</sup> annual wet deposition",
                                         "Total inorganic N annual wet deposition")
colnames(metadata_soil_heatmap_16s) <- c("Tree distance from edge", 
                                         "Tree distance from Boston", 
                                         "Tree DBH", "Tree growth rate",
                                         "Soil temperature", 
                                         "Soil bulk density", 
                                         "Soil pH", "Soil moisture", 
                                         "Soil organic matter", 
                                         "Soil available NH<sub>4</sub><sup>+</sup>",
                                         "Soil available NO<sub>3</sub><sup>-</sup>", 
                                         "Normalized root biomass", 
                                         "Soil soluble salt",
                                         "NO concentration",
                                         "NO<sub>2</sub> concentration",
                                         "NO<sub>x</sub> concentration",
                                         "NH<sub>4</sub><sup>+</sup> annual throughfall",
                                         "NO<sub>3</sub><sup>-</sup> annual throughfall",
                                         "Total inorganic N annual throughfall",
                                         "O<sub>3</sub> concentration",
                                         "Foliar N %",
                                         "Foliar C:N",
                                         "Ca<sup>2+</sup> annual wet deposition",
                                         "Cl<sup>-</sup> annual wet deposition",
                                         "H<sup>+</sup> annual wet deposition",
                                         "K<sup>+</sup> annual wet deposition",
                                         "Mg<sup>2+</sup> annual wet deposition",
                                         "Na<sup>+</sup> annual wet deposition",
                                         "NH<sub>4</sub><sup>+</sup> annual wet deposition",
                                         "NO<sub>3</sub><sup>-</sup> annual wet deposition",
                                         "SO<sub>4</sub><sup>2-</sup> annual wet deposition",
                                         "Total inorganic N annual wet deposition")
colnames(metadata_leaf_heatmap_its) <- c("Tree distance from edge", 
                                         "Tree distance from Boston", 
                                         "Tree DBH", "Tree growth rate",
                                         "Soil temperature", 
                                         "Soil bulk density", 
                                         "Soil pH", "Soil moisture", 
                                         "Soil organic matter", 
                                         "Soil available NH<sub>4</sub><sup>+</sup>",
                                         "Soil available NO<sub>3</sub><sup>-</sup>", 
                                         "Normalized root biomass", 
                                         "Soil soluble salt",
                                         "NO concentration",
                                         "NO<sub>2</sub> concentration",
                                         "NO<sub>x</sub> concentration",
                                         "NH<sub>4</sub><sup>+</sup> annual throughfall",
                                         "NO<sub>3</sub><sup>-</sup> annual throughfall",
                                         "Total inorganic N annual throughfall",
                                         "O<sub>3</sub> concentration",
                                         "Foliar N %",
                                         "Foliar C:N",
                                         "Ca<sup>2+</sup> annual wet deposition",
                                         "Cl<sup>-</sup> annual wet deposition",
                                         "H<sup>+</sup> annual wet deposition",
                                         "K<sup>+</sup> annual wet deposition",
                                         "Mg<sup>2+</sup> annual wet deposition",
                                         "Na<sup>+</sup> annual wet deposition",
                                         "NH<sub>4</sub><sup>+</sup> annual wet deposition",
                                         "NO<sub>3</sub><sup>-</sup> annual wet deposition",
                                         "SO<sub>4</sub><sup>2-</sup> annual wet deposition",
                                         "Total inorganic N annual wet deposition")
colnames(metadata_leaf_heatmap_16s) <- c("Tree distance from edge", 
                                         "Tree distance from Boston", 
                                         "Tree DBH", "Tree growth rate",
                                         "Soil temperature", 
                                         "Soil bulk density", 
                                         "Soil pH", "Soil moisture", 
                                         "Soil organic matter", 
                                         "Soil available NH<sub>4</sub><sup>+</sup>",
                                         "Soil available NO<sub>3</sub><sup>-</sup>", 
                                         "Normalized root biomass", 
                                         "Soil soluble salt",
                                         "NO concentration",
                                         "NO<sub>2</sub> concentration",
                                         "NO<sub>x</sub> concentration",
                                         "NH<sub>4</sub><sup>+</sup> annual throughfall",
                                         "NO<sub>3</sub><sup>-</sup> annual throughfall",
                                         "Total inorganic N annual throughfall",
                                         "O<sub>3</sub> concentration",
                                         "Foliar N %",
                                         "Foliar C:N",
                                         "Ca<sup>2+</sup> annual wet deposition",
                                         "Cl<sup>-</sup> annual wet deposition",
                                         "H<sup>+</sup> annual wet deposition",
                                         "K<sup>+</sup> annual wet deposition",
                                         "Mg<sup>2+</sup> annual wet deposition",
                                         "Na<sup>+</sup> annual wet deposition",
                                         "NH<sub>4</sub><sup>+</sup> annual wet deposition",
                                         "NO<sub>3</sub><sup>-</sup> annual wet deposition",
                                         "SO<sub>4</sub><sup>2-</sup> annual wet deposition",
                                         "Total inorganic N annual wet deposition")
# Make coefficient matrix
Result_soil_its <- NULL
  estimate_soil_its <- NULL
  for (j in 1:ncol(metadata_soil_heatmap_its)){
    cor_soil_its <- cor.test(func_soil_its[,1],metadata_soil_heatmap_its[,j], method="p")
    estimate_soil_its <- c(estimate_soil_its, cor_soil_its$estimate)
  }
  Result_soil_its <- rbind(Result_soil_its, estimate_soil_its)
colnames(Result_soil_its) <- colnames(metadata_soil_heatmap_its)
rownames(Result_soil_its) <- "Fungal animal pathogens in soil"

Result_leaf_its <- NULL
  estimate_leaf_its <- NULL
  for (j in 1:ncol(metadata_leaf_heatmap_its)){
    cor_leaf_its <- cor.test(func_leaf_its[,1],metadata_leaf_heatmap_its[,j], method="p")
    estimate_leaf_its <- c(estimate_leaf_its, cor_leaf_its$estimate)
  }
  Result_leaf_its <- rbind(Result_leaf_its, estimate_leaf_its)
colnames(Result_leaf_its) <- colnames(metadata_leaf_heatmap_its)
rownames(Result_leaf_its) <- "Fungal animal pathogens in leaves"

Result_soil_16s <- NULL
  estimate_soil_16s <- NULL
  for (j in 1:ncol(metadata_soil_heatmap_16s)){
    cor_soil_16s <- cor.test(func_soil_16s[,1],metadata_soil_heatmap_16s[,j], method="p")
    estimate_soil_16s <- c(estimate_soil_16s, cor_soil_16s$estimate)
  }
  Result_soil_16s <- rbind(Result_soil_16s, estimate_soil_16s)
colnames(Result_soil_16s) <- colnames(metadata_soil_heatmap_16s)
rownames(Result_soil_16s) <- "Bacterial zoonotic pathogens in soil"

Result_leaf_16s <- NULL
  estimate_leaf_16s <- NULL
  for (j in 1:ncol(metadata_leaf_heatmap_16s)){
    cor_leaf_16s <- cor.test(func_leaf_16s[,1],metadata_leaf_heatmap_16s[,j], method="p")
    estimate_leaf_16s <- c(estimate_leaf_16s, cor_leaf_16s$estimate)
  }
  Result_leaf_16s <- rbind(Result_leaf_16s, estimate_leaf_16s)
colnames(Result_leaf_16s) <- colnames(metadata_leaf_heatmap_16s)
rownames(Result_leaf_16s) <- "Bacterial zoonotic pathogens in leaves" 

# Make Pval matrix
Result.Pval_soil_its <- NULL
  Pval_soil_its <- NULL
  for (j in 1:ncol(metadata_soil_heatmap_its)){
    cor_soil_its <- cor.test(func_soil_its[,1],metadata_soil_heatmap_its[,j], method="p")
    Pval_soil_its <- c(Pval_soil_its, cor_soil_its$p.value)
  }
  Result.Pval_soil_its <- rbind (Result.Pval_soil_its, Pval_soil_its)
colnames(Result.Pval_soil_its) <- colnames(metadata_soil_heatmap_its)
rownames(Result.Pval_soil_its) <- "Fungal animal pathogens in soil"

Result.Pval_leaf_its <- NULL
  Pval_leaf_its <- NULL
  for (j in 1:ncol(metadata_leaf_heatmap_its)){
    cor_leaf_its <- cor.test(func_leaf_its[,1],metadata_leaf_heatmap_its[,j], method="p")
    Pval_leaf_its <- c(Pval_leaf_its, cor_leaf_its$p.value)
  }
  Result.Pval_leaf_its <- rbind (Result.Pval_leaf_its, Pval_leaf_its)
colnames(Result.Pval_leaf_its) <- colnames(metadata_leaf_heatmap_its)
rownames(Result.Pval_leaf_its) <- "Fungal animal pathogens in leaves"

Result.Pval_soil_16s <- NULL
  Pval_soil_16s <- NULL
  for (j in 1:ncol(metadata_soil_heatmap_16s)){
    cor_soil_16s <- cor.test(func_soil_16s[,1],metadata_soil_heatmap_16s[,j], method="p")
    Pval_soil_16s <- c(Pval_soil_16s, cor_soil_16s$p.value)
  }
  Result.Pval_soil_16s <- rbind (Result.Pval_soil_16s, Pval_soil_16s)
colnames(Result.Pval_soil_16s) <- colnames(metadata_soil_heatmap_16s)
rownames(Result.Pval_soil_16s) <- "Bacterial zoonotic pathogens in soil"

Result.Pval_leaf_16s <- NULL
  Pval_leaf_16s <- NULL
  for (j in 1:ncol(metadata_leaf_heatmap_16s)){
    cor_leaf_16s <- cor.test(func_leaf_16s[,1],metadata_leaf_heatmap_16s[,j], method="p")
    Pval_leaf_16s <- c(Pval_leaf_16s, cor_leaf_16s$p.value)
  }
  Result.Pval_leaf_16s <- rbind (Result.Pval_leaf_16s, Pval_leaf_16s)
colnames(Result.Pval_leaf_16s) <- colnames(metadata_leaf_heatmap_16s)
rownames(Result.Pval_leaf_16s) <- "Bacterial zoonotic pathogens in leaves"

# Make Asterisk matrix
Result.Pval.combined_soil_its <- Result.Pval_soil_its
Result.asterisk_soil_its <- Result.Pval.combined_soil_its
for (k in 1:ncol(Result.Pval.combined_soil_its)){
  for (i in 1:nrow(Result.Pval.combined_soil_its)){
    if(!is.na(Result.Pval.combined_soil_its[i,k])){
      if (Result.Pval.combined_soil_its[i,k] < 0.001){
        Result.asterisk_soil_its[i,k] <- "***"
      } else if (Result.Pval.combined_soil_its[i,k] < 0.01){
        Result.asterisk_soil_its[i,k] <- "**"
      } else if (Result.Pval.combined_soil_its[i,k] < 0.05){
        Result.asterisk_soil_its[i,k] <- "*"
      } else {
        Result.asterisk_soil_its[i,k]<- " "
      }
    } 
  }
}

Result.Pval.combined_leaf_its <- Result.Pval_leaf_its
Result.asterisk_leaf_its <- Result.Pval.combined_leaf_its
for (k in 1:ncol(Result.Pval.combined_leaf_its)){
  for (i in 1:nrow(Result.Pval.combined_leaf_its)){
    if(!is.na(Result.Pval.combined_leaf_its[i,k])){
      if (Result.Pval.combined_leaf_its[i,k] < 0.001){
        Result.asterisk_leaf_its[i,k] <- "***"
      } else if (Result.Pval.combined_leaf_its[i,k] < 0.01){
        Result.asterisk_leaf_its[i,k] <- "**"
      } else if (Result.Pval.combined_leaf_its[i,k] < 0.05){
        Result.asterisk_leaf_its[i,k] <- "*"
      } else {
        Result.asterisk_leaf_its[i,k]<- " "
      }
    } 
  }
}

Result.Pval.combined_soil_16s <- Result.Pval_soil_16s
Result.asterisk_soil_16s <- Result.Pval.combined_soil_16s
for (k in 1:ncol(Result.Pval.combined_soil_16s)){
  for (i in 1:nrow(Result.Pval.combined_soil_16s)){
    if(!is.na(Result.Pval.combined_soil_16s[i,k])){
      if (Result.Pval.combined_soil_16s[i,k] < 0.001){
        Result.asterisk_soil_16s[i,k] <- "***"
      } else if (Result.Pval.combined_soil_16s[i,k] < 0.01){
        Result.asterisk_soil_16s[i,k] <- "**"
      } else if (Result.Pval.combined_soil_16s[i,k] < 0.05){
        Result.asterisk_soil_16s[i,k] <- "*"
      } else {
        Result.asterisk_soil_16s[i,k]<- " "
      }
    } 
  }
}

Result.Pval.combined_leaf_16s <- Result.Pval_leaf_16s
Result.asterisk_leaf_16s <- Result.Pval.combined_leaf_16s
for (k in 1:ncol(Result.Pval.combined_leaf_16s)){
  for (i in 1:nrow(Result.Pval.combined_leaf_16s)){
    if(!is.na(Result.Pval.combined_leaf_16s[i,k])){
      if (Result.Pval.combined_leaf_16s[i,k] < 0.001){
        Result.asterisk_leaf_16s[i,k] <- "***"
      } else if (Result.Pval.combined_leaf_16s[i,k] < 0.01){
        Result.asterisk_leaf_16s[i,k] <- "**"
      } else if (Result.Pval.combined_leaf_16s[i,k] < 0.05){
        Result.asterisk_leaf_16s[i,k] <- "*"
      } else {
        Result.asterisk_leaf_16s[i,k]<- " "
      }
    } 
  }
}
# Combine netowrk propeties + others
Result.combined_soil_its <- Result_soil_its
melt_soil_its <- melt(Result.combined_soil_its)
melt.asterisk_soil_its <- melt(Result.asterisk_soil_its)

Result.combined_leaf_its <- Result_leaf_its
melt_leaf_its <- melt(Result.combined_leaf_its)
melt.asterisk_leaf_its <- melt(Result.asterisk_leaf_its)

Result.combined_soil_16s <- Result_soil_16s
melt_soil_16s <- melt(Result.combined_soil_16s)
melt.asterisk_soil_16s <- melt(Result.asterisk_soil_16s)

Result.combined_leaf_16s <- Result_leaf_16s
melt_leaf_16s <- melt(Result.combined_leaf_16s)
melt.asterisk_leaf_16s <- melt(Result.asterisk_leaf_16s)

melt_all <- rbind(melt_leaf_its, melt_soil_its, melt_leaf_16s, melt_soil_16s)
melt_asterisk_all <- rbind(melt.asterisk_leaf_its, melt.asterisk_soil_its, 
                           melt.asterisk_leaf_16s, melt.asterisk_soil_16s)

# Visualize
vdata_its <- cbind(melt_all, melt_asterisk_all$value)
colnames(vdata_its) <- c("microbe", "environment", "coef", "Pval")

order_of_env_factors <- vdata_its[which(vdata_its$microbe == "Fungal animal pathogens in leaves"),]

vdata_its$environment <- factor(vdata_its$environment, 
                            levels = order_of_env_factors$environment[order(order_of_env_factors$coef)])

heatmap_soilO <- ggplot(vdata_its) +
  geom_tile(aes(x=microbe, y=environment, fill=coef)) +
  geom_text(aes(x=microbe, y=environment, label = Pval), size = 3) +
  labs(x="",y="") + theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, hjust=1,vjust=0.3)) +
  theme(axis.text.x = element_text(angle = 90, hjust=1,vjust=0.3)) + 
  scale_fill_gradient2(low = "#003049", mid = "#FDF0D5", high = "#C00000", 
                       name = "Correlation\ncoefficient") +
  theme(text = element_text(family = "Arial"),
        axis.title.x = element_text(size = 8), 
        axis.title.y = element_markdown(size = 8), 
        axis.text.y = element_markdown(size = 8, color = "black"),
        axis.text.x = element_text(size = 8, color = "black"),
        legend.text = element_text(size = 8), 
        legend.title = element_text(size = 8), 
        plot.title = element_text(size = 8)) + 
  ggtitle("") + coord_fixed()
heatmap_soilO

ggsave("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Figures/heatmap_manuscript_fig2c_new.pdf", device = cairo_pdf, width = 5, height = 10, units = "in", dpi = 300)
```

Figure 3a
```{r inorganic N soil and deposition}
metadata_16s_200$site_edge <- paste0(metadata_16s_200$tree_urban_type, " ", metadata_16s_200$tree_edge_type)
metadata_16s_200$site_edge <- gsub("Urban Street Tree Urban Street Tree", "Urban Street Tree",
                            metadata_16s_200$site_edge)
metadata_16s_200$site_edge <- factor(metadata_16s_200$site_edge, levels = c("Urban Street Tree", 
                                                      "Urban Forest Edge", 
                                                      "Urban Forest Interior", 
                                                      "Rural Forest Edge", 
                                                      "Rural Forest Interior"))
metadata_soil_16s <- metadata_16s_200[which(metadata_16s_200$sample_type == "Soil, 0-15 cm"),]
length(unique(metadata_soil_16s$tree_id))


# Format data
standard_error <- function(x) sd(x) / sqrt(length(x)) 
Results_anova <- NULL
Results_multcomp <- NULL

data <- cbind(metadata_soil_16s$soil_no3)
colnames(data) <- c("Available soil NO<sub>3</sub><sup>-</sup>")

mean <- aggregate(data~site_edge,data = metadata_soil_16s,FUN= "mean",na.action = na.omit)
se <- aggregate(data~site_edge,data = metadata_soil_16s,FUN= standard_error,na.action = na.omit)

melt.mean <- melt(mean)
melt.se <- melt(se)

vdata <- cbind(melt.mean, melt.se[,ncol(melt.se)])
colnames(vdata) <- c("site_edge","Nutirent","Abundance","SE")

vdata$tukey <- c(
  "a[\"\\u25A0\"]", 
  "ab[\"\\u25A0\"]", 
  "b[\"\\u25A0\"]", 
  "ab[\"\\u25A0\"]", 
  "b[\"\\u25A0\"]"
)
vdata$site_edge <- gsub(" ", "\n", vdata$site_edge)
vdata$site_edge <- factor(vdata$site_edge, 
                               levels = c("Urban\nStreet\nTree", 
                                          "Urban\nForest\nEdge",
                                          "Urban\nForest\nInterior",
                                          "Rural\nForest\nEdge",
                                          "Rural\nForest\nInterior"))

# Visualize
pal <- c("#FF8080","#FF0000","#C00000","#800000","#000000")

nitrogen_avail <- ggplot(vdata) +
  geom_line(data = vdata, 
             aes(x=site_edge, y=Abundance, group=Nutirent), 
            linetype = 2, color = "black") +
  geom_point(data = vdata, 
             aes(x=site_edge, y=Abundance, color=site_edge, shape=Nutirent), size=5) + 
  geom_errorbar(data = vdata, 
                aes(x=site_edge, ymin=Abundance-SE, ymax=Abundance+SE, 
                    color=site_edge), width = 0.2) +
  geom_text(data = vdata, direction = "y",
                  aes(x=site_edge, y=Abundance + SE + 0.02, label = tukey), 
            size = 3, parse = TRUE) +
  theme_classic() + labs (y="Soil inorganic N content (ug/g)", x="") +
  theme(text = element_text(family = "Arial"),
        axis.title.x = element_text(size = 10, color="black"), 
        axis.title.y = element_text(size = 10), 
        axis.text.x = element_text(size = 10, color = "black"), 
        axis.text.y = element_text(size = 10, color = "black"), 
        legend.text = element_markdown(size = 7), 
        legend.title = element_text(size = 1), 
        plot.title = element_markdown(size = 10), legend.position = "none", legend.background = element_rect(fill = NA)) +
  scale_color_manual(values = pal) + 
  scale_fill_manual(values = pal) +
  guides(color = "none", shape = guide_legend(title = ""))+
  ggtitle("Available soil NO<sub>3</sub><sup>-</sup>")
nitrogen_avail

ggsave("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Figures/figure3_nitrogen_fragmentation_new.pdf", device = cairo_pdf, plot = nitrogen_avail, dpi = 300, width = 6, height = 3.5, units = "in")

model1 <- lme(formula(soil_nh4 ~ site_edge), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML", data = metadata_soil_16s)
summary(model1)
summary(emmeans(model1, pairwise ~ site_edge))

model1 <- lme(formula(soil_no3 ~ site_edge), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML", data = metadata_soil_16s)
summary(model1)
summary(emmeans(model1, pairwise ~ site_edge))

model1 <- lme(formula(rindy_2019_nh4n_deposition.kgn_ha_yr ~ site_edge), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML", data = metadata_soil_16s)
summary(model1)
summary(emmeans(model1, pairwise ~ site_edge))

model1 <- lme(formula(rindy_2019_no3n_deposition.kgn_ha_yr ~ site_edge), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML", data = metadata_soil_16s)
summary(model1)
summary(emmeans(model1, pairwise ~ site_edge))
```

Figure 3b
```{r soil bacterial biogeochemistry}
metadata_16s_200$site_edge <- paste0(metadata_16s_200$tree_urban_type, " ", metadata_16s_200$tree_edge_type)
metadata_16s_200$site_edge <- gsub("Urban Street Tree Urban Street Tree", "Urban Street Tree",
                            metadata_16s_200$site_edge)
metadata_16s_200$site_edge <- factor(metadata_16s_200$site_edge, levels = c("Urban Street Tree", 
                                                      "Urban Forest Edge", 
                                                      "Urban Forest Interior", 
                                                      "Rural Forest Edge", 
                                                      "Rural Forest Interior"))

metadata_soil_16s <- metadata_16s_200[which(metadata_16s_200$sample_type == "Soil, 0-15 cm"),]

standard_error <- function(x) sd(x) / sqrt(length(x)) 
Results_anova <- NULL
Results_multcomp <- NULL

data <- log10(cbind(metadata_soil_16s$perc_chitinolytic, metadata_soil_16s$perc_dissim_nitrate_reduction,
metadata_soil_16s$perc_partial_nitrification, metadata_soil_16s$perc_denitrification)+0.001)
colnames(data) <- c("Chitinolytic bacteria", "Dissimilatory nitrate reducing bacteria",
"Nitrifying bacteria", "Denitrifying bacteria")

mean <- aggregate(data~site_edge,data = metadata_soil_16s,FUN= "mean",na.action = na.omit)
se <- aggregate(data~site_edge,data = metadata_soil_16s,FUN= standard_error,na.action = na.omit)

melt.mean <- melt(mean)
melt.se <- melt(se)

vdata <- cbind(melt.mean, melt.se[,ncol(melt.se)])
colnames(vdata) <- c("site_edge","Microbe","Abundance","SE")
vdata$Microbe <- factor(vdata$Microbe, levels = c("Chitinolytic bacteria", "Nitrifying bacteria",
                                                  "Dissimilatory nitrate reducing bacteria",
                                                  "Denitrifying bacteria"))

vdata$tukey <- c(
  "a[\"\\u25A0\"]", 
  "b[\"\\u25A0\"]", 
  "b[\"\\u25A0\"]", 
  "b[\"\\u25A0\"]", 
  "b[\"\\u25A0\"]",
  "a[\"\\u25B2\"]", 
  "b[\"\\u25B2\"]", 
  "b[\"\\u25B2\"]", 
  "b[\"\\u25B2\"]", 
  "b[\"\\u25B2\"]",
  "a[\"\\u25CF\"]", 
  "b[\"\\u25CF\"]", 
  "b[\"\\u25CF\"]", 
  "b[\"\\u25CF\"]", 
  "b[\"\\u25CF\"]",
  "a[\"\\u25C6\"]", 
  "b[\"\\u25C6\"]", 
  "b[\"\\u25C6\"]", 
  "b[\"\\u25C6\"]", 
  "b[\"\\u25C6\"]"
)
vdata$site_edge <- gsub(" ", "\n", vdata$site_edge)
vdata$site_edge <- factor(vdata$site_edge, 
                               levels = c("Urban\nStreet\nTree", 
                                          "Urban\nForest\nEdge",
                                          "Urban\nForest\nInterior",
                                          "Rural\nForest\nEdge",
                                          "Rural\nForest\nInterior"))

# Visualize
pal <- c("#FF8080","#FF0000","#C00000","#800000","#000000")

n_bac <- ggplot(vdata) +
  geom_line(data = vdata, 
             aes(x=site_edge, y=Abundance, group=Microbe), 
            linetype = 2, color = "black") +
  geom_point(data = vdata, 
             aes(x=site_edge, y=Abundance, color=site_edge, shape=Microbe), size=5) + 
  geom_errorbar(data = vdata, 
                aes(x=site_edge, ymin=Abundance-SE, ymax=Abundance+SE, 
                    color=site_edge), width = 0.2) +
  geom_text_repel(data = vdata, direction = "y",
                  aes(x=site_edge, y=Abundance, label = tukey), 
            hjust = 1, segment.color = NA,
            size = 3, parse = TRUE, nudge_x = -0.1) +
  theme_classic() + labs (y="log(Functional group abundance (%))", x="") +
  theme(text = element_text(family = "Arial"),
        axis.title.x = element_text(size = 10, color="black"), 
        axis.title.y = element_text(size = 10), 
        axis.text.x = element_text(size = 10, color = "black"), 
        axis.text.y = element_text(size = 10, color = "black"), 
        legend.text = element_markdown(size = 7), 
        legend.title = element_text(size = 1), 
        plot.title = element_text(size = 10), legend.position = c(0.45, 0.6), legend.background = element_rect(fill = NA)) +
  scale_color_manual(values = pal) + 
  scale_fill_manual(values = pal) + 
  scale_shape_manual(values = c(15,16,17,18)) + 
  guides(color = "none", shape = guide_legend(title = ""))+
  ggtitle("Nitrogen cycling bacteria in soil")

n_bac
ggsave("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Figures/figure3_nbacteria_fragmentation.pdf", device = cairo_pdf, plot = n_bac, dpi = 300, width = 5.75, height = 3.5, units = "in")

model1 <- lme(formula(perc_chitinolytic ~ site_edge), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML", data = metadata_soil_16s)
summary(model1)
summary(emmeans(model1, pairwise ~ site_edge))

model1 <- lme(formula(perc_dissim_nitrate_reduction ~ site_edge), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML", data = metadata_soil_16s)
summary(model1)
summary(emmeans(model1, pairwise ~ site_edge))

model1 <- lme(formula(perc_partial_nitrification ~ site_edge), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML", data = metadata_soil_16s)
summary(model1)
summary(emmeans(model1, pairwise ~ site_edge))

model1 <- lme(formula(perc_denitrification ~ site_edge), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML", data = metadata_soil_16s)
summary(model1)
summary(emmeans(model1, pairwise ~ site_edge))
```

Figure 3c
```{r saprotroph stacked bar chart}
metadata_its$site_edge <- paste0(metadata_its$tree_urban_type, " ", metadata_its$tree_edge_type)
metadata_its$site_edge <- gsub("Urban Street Tree Urban Street Tree", "Urban Street Tree",
                            metadata_its$site_edge)
metadata_its$site_edge <- factor(metadata_its$site_edge, levels = c("Urban Street Tree", 
                                                      "Urban Forest Edge", 
                                                      "Urban Forest Interior", 
                                                      "Rural Forest Edge", 
                                                      "Rural Forest Interior"))
metadata_soil_its <- metadata_its[which(metadata_its$sample_type == "Soil, 0-15 cm"),]

# Format data
# standard_error <- function(x) sd(x) / sqrt(length(x)) 

data <- cbind(#metadata_soil_its$perc_nectar_tap_saprotroph, 
# metadata_soil_its$perc_pollen_saprotroph, 
metadata_soil_its$perc_dung_saprotroph, 
metadata_soil_its$perc_wood_saprotroph, metadata_soil_its$perc_litter_saprotroph, metadata_soil_its$perc_soil_saprotroph)
colnames(data) <- c(#"Nectar",#"Pollen",
"Dung","Wood","Litter","Soil")

mean <- aggregate(data~site_edge,data = metadata_soil_its,FUN= "mean",na.action = na.omit)

vdata <- melt(mean)

fill_pal <- c("#B3CDDE", "#669BBC", "#003049", "#001825")

vdata <- vdata[order(vdata$variable, decreasing = TRUE),]

vdata <- vdata %>%
  group_by(site_edge) %>%
  mutate(cumulative = cumsum(value),            # Cumulative sum
         midpoint = cumulative - value / 2)     # Midpoint of each segment
vdata$midpoint[17:20] <- vdata$cumulative[17:20] + 0.4
vdata_cumsum <- vdata %>%
  group_by(site_edge) %>%
  summarise(cumulative_sum = sum(value))

vdata$tukey <- rep(c("a", "b", "b", "b", "b"),4)
vdata$color <- c(rep("white",10), rep("black",10))

vdata$site_edge <- gsub(" ", "\n", vdata$site_edge)
vdata$site_edge <- factor(vdata$site_edge, 
                               levels = c("Urban\nStreet\nTree", 
                                          "Urban\nForest\nEdge",
                                          "Urban\nForest\nInterior",
                                          "Rural\nForest\nEdge",
                                          "Rural\nForest\nInterior"))
library(ggsignif)
# ggplot
saps <- ggplot (vdata,  mapping=aes(x=site_edge, y=value, fill=variable))+
geom_bar(aes(), stat="identity", position="stack",color="black")+
  geom_text(aes(y = midpoint, label = tukey), color = vdata$color, family = "Arial") +
theme_classic()+ geom_signif(
    comparisons = list(c("Urban\nStreet\nTree", "Urban\nForest\nEdge"),
                       c("Urban\nStreet\nTree", "Urban\nForest\nInterior"),
                       c("Urban\nStreet\nTree", "Rural\nForest\nEdge"),
                       c("Urban\nStreet\nTree", "Rural\nForest\nInterior")),  # Specify pairwise comparisons
    y_position = c(max(vdata_cumsum$cumulative_sum) + 0.1,
                   max(vdata_cumsum$cumulative_sum) + 0.7,
                   max(vdata_cumsum$cumulative_sum) + 1.3,
                   max(vdata_cumsum$cumulative_sum) + 1.9),  # Position above the bars
    annotations = c("**","***", "**", "***"), textsize = 2.5) +
theme(text = element_text(family = "Arial"),
        axis.title.x = element_text(size = 10, color="black"), 
        axis.title.y = element_text(size = 10), 
        axis.text.x = element_text(size = 10, color = "black"), 
        axis.text.y = element_text(size = 10, color = "black"), 
        legend.text = element_markdown(size = 8), 
        legend.title = element_text(size = 1), 
        plot.title = element_text(size = 10), 
        legend.position = c(0.7, 0.7), legend.direction="horizontal") + scale_fill_manual(values = fill_pal) + 
labs (x="",y="Fungal decomposer abundance (%)", fill="") + ggtitle("Soil fungal decomposers")
saps
ggsave("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Figures/figure3_saps_fragmentation.pdf", device = cairo_pdf, plot = saps, dpi = 300, width = 5.75, height = 3.5, units = "in")

metadata_soil_its$sap_tot <- metadata_soil_its$perc_dung_saprotroph + metadata_soil_its$perc_wood_saprotroph + metadata_soil_its$perc_litter_saprotroph + metadata_soil_its$perc_soil_saprotroph

model1 <- lme(formula(perc_dung_saprotroph ~ site_edge), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML", data = metadata_soil_its)
summary(model1)
summary(emmeans(model1, pairwise ~ site_edge))

model1 <- lme(formula(perc_wood_saprotroph ~ site_edge), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML", data = metadata_soil_its)
summary(model1)
summary(emmeans(model1, pairwise ~ site_edge))

model1 <- lme(formula(perc_litter_saprotroph ~ site_edge), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML", data = metadata_soil_its)
summary(model1)
summary(emmeans(model1, pairwise ~ site_edge))

model1 <- lme(formula(perc_soil_saprotroph ~ site_edge), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML", data = metadata_soil_its)
summary(model1)
summary(emmeans(model1, pairwise ~ site_edge))

model1 <- lme(formula(sap_tot ~ site_edge), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML", data = metadata_soil_its)
summary(model1)
summary(emmeans(model1, pairwise ~ site_edge))
```

Figure 3d
```{r bacterial methanotrophs}
metadata_16s_200$site_edge <- paste0(metadata_16s_200$tree_urban_type, " ", metadata_16s_200$tree_edge_type)
metadata_16s_200$site_edge <- gsub("Urban Street Tree Urban Street Tree", "Urban Street Tree",
                            metadata_16s_200$site_edge)
metadata_16s_200$site_edge <- factor(metadata_16s_200$site_edge, levels = c("Urban Street Tree", 
                                                      "Urban Forest Edge", 
                                                      "Urban Forest Interior", 
                                                      "Rural Forest Edge", 
                                                      "Rural Forest Interior"))
metadata_leaf_16s <- metadata_16s_200[which(metadata_16s_200$sample_type == "Leaf"),]

# Format data
standard_error <- function(x) sd(x) / sqrt(length(x)) 
Results_anova <- NULL
Results_multcomp <- NULL

mean_leaf <- aggregate(log10(perc_methanotroph+0.001)~site_edge,data = metadata_leaf_16s,FUN= "mean",na.action = na.omit)
se_leaf <- aggregate(log10(perc_methanotroph+0.001)~site_edge,data = metadata_leaf_16s,FUN= standard_error,na.action = na.omit)

leaf <- cbind(mean_leaf, se_leaf[,ncol(se_leaf)],"Leaf")

colnames(leaf) <- c("site_edge","Abundance","SE","Microbe")

vdata <- leaf
vdata$max <- c(-2.7, 0.5, 0.5, 0.5, 0.6)

vdata$tukey <- c("b", "a", "a", "ab", "a")
vdata$site_edge <- gsub(" ", "\n", vdata$site_edge)
vdata$site_edge <- factor(vdata$site_edge, 
                               levels = c("Urban\nStreet\nTree", 
                                          "Urban\nForest\nEdge",
                                          "Urban\nForest\nInterior",
                                          "Rural\nForest\nEdge",
                                          "Rural\nForest\nInterior"))
metadata_leaf_16s$site_edge <- gsub(" ", "\n", metadata_leaf_16s$site_edge)
metadata_leaf_16s$site_edge <- factor(metadata_leaf_16s$site_edge, 
                               levels = c("Urban\nStreet\nTree", 
                                          "Urban\nForest\nEdge",
                                          "Urban\nForest\nInterior",
                                          "Rural\nForest\nEdge",
                                          "Rural\nForest\nInterior"))

# Visualize
pal <- c("#FF8080","#FF0000","#C00000","#800000","#000000")

meth <- ggplot() +
  geom_line(data = vdata, 
             aes(x=site_edge, y=Abundance, group=Microbe), 
            linetype = 2, color = "black") +
  geom_point(data = vdata,
             aes(x=site_edge, y=Abundance, color=site_edge, shape=Microbe), size=5) +
  geom_point(data = metadata_leaf_16s,
             aes(x = site_edge, y = log10(perc_methanotroph+0.001), color = site_edge),
             position = position_jitter(width = 0.2), size = 2, alpha = 0.2, stroke = NA) +
  geom_errorbar(data = vdata,
                aes(x=site_edge, ymin=Abundance-SE, ymax=Abundance+SE,
                    color=site_edge), width = 0.2) +
  geom_text(data = vdata, aes(x = site_edge, y = max, label = tukey), size = 3,
            family = "Arial") +
  theme_classic() + labs (y="log(Methanotrophic bacteria abundance (%))", x="") +
  theme(text = element_text(family = "Arial"),
        axis.title.x = element_text(size = 10, color="black"), 
        axis.title.y = element_text(size = 10), 
        axis.text.x = element_text(size = 10, color = "black"), 
        axis.text.y = element_text(size = 10, color = "black"), 
        legend.text = element_markdown(size = 8), 
        legend.title = element_text(size = 1), 
        plot.title = element_text(size = 10),
        legend.position = "none") +
  scale_color_manual(values = pal) + 
  scale_fill_manual(values = pal) +
  guides(color = "none", shape = guide_legend(title = ""))+
  ggtitle("Methanotrophic bacterial abundance in leaves")
meth
ggsave("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Figures/figure3_methanebacteria_fragmentation.pdf", device = cairo_pdf, plot = meth, dpi = 300, width = 5.75, height = 3.5, units = "in")


model1 <- lme(formula(perc_methanotroph ~ site_edge), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML", data = metadata_leaf_16s)
summary(model1)
summary(emmeans(model1, pairwise ~ site_edge))
```

Figure 4a
```{r diversity -- fungi leaf}
metadata_leaf_its <- metadata_its[which(metadata_its$sample_type == 
                                               "Leaf"),]
metadata_leaf_its_div <- metadata_leaf_its[,colnames(metadata_leaf_its) %in% 
                                                       c("sample_name", 
                                                         "tree_dist_to_boston_km", "shannon",
                                                         "shannon_nopath")]

diversity_long <- metadata_leaf_its_div %>% 
  pivot_longer(!c(sample_name, tree_dist_to_boston_km), 
               names_to = "diversity_type", values_to = "value")
diversity_long$diversity_type <- gsub("shannon_nopath", "non-pathogenic fungi",
                                      diversity_long$diversity_type)
diversity_long$diversity_type <- gsub("shannon", "all fungi",
                                      diversity_long$diversity_type)

div_fungi_phyllo <- ggplot(diversity_long, aes(x = log(tree_dist_to_boston_km +1), y = value, 
                          color = diversity_type, fill = diversity_type)) + 
  geom_point(shape = 18, alpha = 0.5, stroke = NA, size = 2.5) +
  geom_smooth(method = "lm", alpha = 0.12, aes(ymax = after_stat(y + se * sqrt(length(y))),
                  ymin = after_stat(y - se * sqrt(length(y))))) + 
  scale_color_manual(values = plot_pal) + scale_fill_manual(values = plot_pal) + 
  theme_classic() + 
  labs(x = "log(Distance from Boston (km))", y = "Shannon diversity", legend = NA) + 
 theme(text = element_text(family = "Arial"),
        axis.title.x = element_text(size = 10, color="black"), 
        axis.title.y = element_text(size = 10), 
        axis.text.x = element_text(size = 10, color = "black"), 
        axis.text.y = element_text(size = 10, color = "black"), 
        legend.text = element_text(size = 7), 
        legend.title = element_text(size = 1), 
        plot.title = element_text(size = 10), legend.position = "none") + ggtitle("Fungal diversity in leaves")
div_fungi_phyllo

ggsave("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Figures/figure4_fungi_phyllo_dfb.pdf", device = cairo_pdf, plot = div_fungi_phyllo, dpi = 300, width = 5.75, height = 3.5, units = "in")

model1 <- lme(formula(shannon ~ log(tree_dist_to_boston_km+1) * as.numeric(tree_distance_from_edge)), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML", data = metadata_leaf_its)
anova(model1)
r.squaredGLMM(model1)

model1 <- lme(formula(shannon_nopath ~ log(tree_dist_to_boston_km+1) * as.numeric(tree_distance_from_edge)), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML", data = metadata_leaf_its)
anova(model1)
r.squaredGLMM(model1)
```

Figure 4b
```{r diversity -- bacteria leaf}
metadata_leaf_16s <- metadata_16s_200[which(metadata_16s_200$sample_type == 
                                               "Leaf"),]
metadata_leaf_16s_div <- metadata_leaf_16s[,colnames(metadata_leaf_16s) %in% 
                                                       c("sample_name", 
                                                         "tree_dist_to_boston_km", "shannon.x",
                                                         "shannon_nopath")]

diversity_long <- metadata_leaf_16s_div %>% 
  pivot_longer(!c(sample_name, tree_dist_to_boston_km), 
               names_to = "diversity_type", values_to = "value")
diversity_long$diversity_type <- gsub("shannon_nopath", "non-pathogenic fungi",
                                      diversity_long$diversity_type)
diversity_long$diversity_type <- gsub("shannon", "all fungi",
                                      diversity_long$diversity_type)

div_bac_phyllo <- ggplot(diversity_long, aes(x = log(tree_dist_to_boston_km +1), y = value, 
                          color = diversity_type, fill = diversity_type)) + 
  geom_point(shape = 18, alpha = 0.5, stroke = NA, size = 2.5) +
  geom_smooth(method = "lm", alpha = 0.12, aes(ymax = after_stat(y + se * sqrt(length(y))),
                  ymin = after_stat(y - se * sqrt(length(y))))) +  
  scale_color_manual(values = plot_pal) + scale_fill_manual(values = plot_pal) + 
  theme_classic() + 
  labs(x = "log(Distance from Boston (km))", y = "Shannon diversity", legend = NA) + 
  theme(text = element_text(family = "Arial"),
        axis.title.x = element_text(size = 10, color="black"), 
        axis.title.y = element_text(size = 10), 
        axis.text.x = element_text(size = 10, color = "black"), 
        axis.text.y = element_text(size = 10, color = "black"), 
        legend.text = element_text(size = 7), 
        legend.title = element_text(size = 1), 
        plot.title = element_text(size = 10), legend.position = "none") + ggtitle("Bacterial diversity in leaves")
div_bac_phyllo
ggsave("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Figures/figure4_bac_phyllo_dfb.pdf", device = cairo_pdf, plot = div_bac_phyllo, dpi = 300, width = 5.75, height = 3.5, units = "in")

model1 <- lme(formula(shannon.x ~ log(tree_dist_to_boston_km+1) * as.numeric(tree_distance_from_edge)), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML", data = metadata_leaf_16s)
anova(model1)
r.squaredGLMM(model1)

model1 <- lme(formula(shannon_nopath ~ log(tree_dist_to_boston_km+1) * as.numeric(tree_distance_from_edge)), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML", data = metadata_leaf_16s)
anova(model1)
r.squaredGLMM(model1)
```

Figure 4c
```{r diversity -- fungi soil}
metadata_soil_its <- metadata_its[which(metadata_its$sample_type != 
                                               "Leaf"),]
metadata_soil_its_div <- metadata_soil_its[,colnames(metadata_soil_its) %in% 
                                                       c("sample_name", "sample_type", 
                                                         "tree_dist_to_boston_km", "shannon",
                                                         "shannon_nopath")]

diversity_long <- metadata_soil_its_div %>% 
  pivot_longer(!c(sample_name, sample_type, tree_dist_to_boston_km), 
               names_to = "diversity_type", values_to = "value")
diversity_long$diversity_type <- gsub("shannon_nopath", "non-pathogenic fungi",
                                      diversity_long$diversity_type)
diversity_long$diversity_type <- gsub("shannon", "all fungi",
                                      diversity_long$diversity_type)

div_fungi_bg <- ggplot(diversity_long, aes(x = log(tree_dist_to_boston_km +1), y = value, 
                          color = diversity_type, fill = diversity_type)) + 
  geom_point(aes(shape = sample_type), alpha = 0.5, size = 2) +
  geom_smooth(method = "lm", alpha = 0.12, aes(ymax = after_stat(y + se * sqrt(length(y))),                   
                                 ymin = after_stat(y - se * sqrt(length(y))))) + 
  scale_color_manual(values = plot_pal) + scale_fill_manual(values = plot_pal) + 
  theme_classic() + 
  labs(x = "log(Distance from Boston (km))", y = "Shannon diversity", legend = NA) + 
  theme(text = element_text(family = "Arial"),
        axis.title.x = element_text(size = 10, color="black"), 
        axis.title.y = element_text(size = 10), 
        axis.text.x = element_text(size = 10, color = "black"), 
        axis.text.y = element_text(size = 10, color = "black"), 
        legend.text = element_text(size = 7), 
        legend.title = element_text(size = 1), 
        plot.title = element_text(size = 10), legend.position = "none") + ggtitle("Fungal diversity belowground") +
  scale_shape_manual(values = shapes)
div_fungi_bg
ggsave("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Figures/figure4_fungi_bg_dfb.pdf", device = cairo_pdf, plot = div_fungi_bg, dpi = 300, width = 5.75, height = 3.5, units = "in")

div_fungi_bg <- ggplot(diversity_long, aes(x = log(tree_dist_to_boston_km +1), y = value, 
                          color = diversity_type, fill = diversity_type)) + 
  geom_point(aes(shape = sample_type), alpha = 0.5, size = 2) +
  geom_smooth(method = "lm", alpha = 0.12, aes(ymax = after_stat(y + se * sqrt(length(y))),                   
                                 ymin = after_stat(y - se * sqrt(length(y))))) + 
  scale_color_manual(values = plot_pal, guide = "none") + scale_fill_manual(values = plot_pal, guide = "none") + 
  theme_classic() + 
  labs(x = "log(Distance from Boston (km))", y = "Shannon diversity", shape = NA) + 
  theme(text = element_text(family = "Arial"),
        axis.title.x = element_text(size = 10, color="black"), 
        axis.title.y = element_text(size = 10), 
        axis.text.x = element_text(size = 10, color = "black"), 
        axis.text.y = element_text(size = 10, color = "black"), 
        legend.text = element_text(size = 7), 
        legend.title = element_text(size = 1), 
        plot.title = element_text(size = 10), legend.position = "bottom") + ggtitle("Fungal diversity belowground") +
  scale_shape_manual(values = shapes)
div_fungi_bg
ggsave("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Figures/figure4_fungi_bg_dfb_wlegend.pdf", device = cairo_pdf, plot = div_fungi_bg, dpi = 300, width = 5.75, height = 3.5, units = "in")

model1 <- lme(formula(shannon ~ log(tree_dist_to_boston_km+1) * as.numeric(tree_distance_from_edge)), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML", data = metadata_soil_its)
anova(model1)
r.squaredGLMM(model1)

model1 <- lme(formula(shannon_nopath ~ log(tree_dist_to_boston_km+1) * as.numeric(tree_distance_from_edge)), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML", data = metadata_soil_its)
anova(model1)
r.squaredGLMM(model1)
```

Figure 4d
```{r diversity -- bacteria soil}
metadata_soil_16s <- metadata_16s_200[which(metadata_16s_200$sample_type != 
                                               "Leaf"),]
metadata_soil_16s_div <- metadata_soil_16s[,colnames(metadata_soil_16s) %in% 
                                                       c("sample_name", "sample_type",
                                                         "tree_dist_to_boston_km", "shannon.x",
                                                         "shannon_nopath")]

diversity_long <- metadata_soil_16s_div %>% 
  pivot_longer(!c(sample_name, sample_type, tree_dist_to_boston_km), 
               names_to = "diversity_type", values_to = "value")
diversity_long$diversity_type <- gsub("shannon_nopath", "non-pathogenic fungi",
                                      diversity_long$diversity_type)
diversity_long$diversity_type <- gsub("shannon", "all fungi",
                                      diversity_long$diversity_type)

div_bac_bg <- ggplot(diversity_long, aes(x = log(tree_dist_to_boston_km +1), y = value, 
                          color = diversity_type, fill = diversity_type)) + 
  geom_point(aes(shape = sample_type), alpha = 0.5, size = 2) +
  geom_smooth(method = "lm", alpha = 0.12, aes(ymax = after_stat(y + se * sqrt(length(y))),                   
                                 ymin = after_stat(y - se * sqrt(length(y)))))  + 
  scale_color_manual(values = plot_pal, guide = "none") + scale_fill_manual(values = plot_pal, guide = "none") + 
  theme_classic() + 
  labs(x = "log(Distance from Boston (km))", y = "Shannon diversity", legend = NA) + 
  theme(text = element_text(family = "Arial"),
        axis.title.x = element_text(size = 10, color="black"), 
        axis.title.y = element_text(size = 10), 
        axis.text.x = element_text(size = 10, color = "black"), 
        axis.text.y = element_text(size = 10, color = "black"), 
        legend.text = element_text(size = 7), 
        legend.title = element_text(size = 1), 
        plot.title = element_text(size = 10), legend.position = "none") + ggtitle("Bacterial diversity belowground") +
  scale_shape_manual(values = shapes)
div_bac_bg
ggsave("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Figures/figure4_bac_bg_dfb.pdf", device = cairo_pdf, plot = div_bac_bg, dpi = 300, width = 5.75, height = 3.5, units = "in")

model1 <- lme(formula(shannon.x ~ log(tree_dist_to_boston_km+1) * as.numeric(tree_distance_from_edge)), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML", data = metadata_soil_16s)
anova(model1)
r.squaredGLMM(model1)

model1 <- lme(formula(shannon_nopath ~ log(tree_dist_to_boston_km+1) * as.numeric(tree_distance_from_edge)), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML", data = metadata_soil_16s)
anova(model1)
r.squaredGLMM(model1)
```

Figure S1a
```{r PCA -- fungi leaf}
leaf_aitch_its <- all_aitch_its
metadata_leaf_its <- metadata_its
metadata_leaf_its <- metadata_leaf_its[which(metadata_leaf_its$sample_type == 
                                               "Leaf"),]
leaf_aitch_its <- leaf_aitch_its[which(row.names(leaf_aitch_its) %in%
                                         metadata_leaf_its$sample_name),
                                 which(colnames(leaf_aitch_its) %in%
                                         metadata_leaf_its$sample_name)]

pca_result_leaf_its <- prcomp(leaf_aitch_its, scale. = TRUE)
pca_leaf_df_its <- data.frame(Sample = rownames(leaf_aitch_its), 
                              PC1 = pca_result_leaf_its$x[,1], 
                              PC2 = pca_result_leaf_its$x[,2])

pca_leaf_df_its <- merge(pca_leaf_df_its, metadata_leaf_its, 
                         by.x = "Sample", by.y = "sample_name", all.x = T)

pca_summary_leaf_its <- summary(pca_result_leaf_its)
pca_variance_leaf_its <- pca_summary_leaf_its$importance[2, ]  # Proportion of variance
pca_variance_cumulative_leaf_its <- pca_summary_leaf_its$importance[3, ]  # Cumulative proportion of variance

# env_names <- c("tree_distance_from_edge", "tree_distance_to_boston_km",
#          "tree_dbh_2021", "tree_growth_rate", "soil_temperature",
#          "soil_bulk_density", "soil_ph", "soil_percent_moisture",
#          "soil_percent_organic_matter", "soil_nh4", "soil_no3",
#          "normalized_root_mass", "caron_2018_2019_soil_soluble_salt",
#          "rindy_2019_no_ppb", "rindy_2019_no2_ppb", "rindy_2019_NO<sub>x</sub> Concentration_ppb",
#          "rindy_2019_o3_ppb", "caron_2018_2019_foliar_n_percent",
#          "caron_2019_2019_foliar_c_n", 
#          "conrad_rooney_nadp_ca_deposition_kg.ha",    
#          "conrad_rooney_nadp_cl_deposition_kg.ha",       
#          "conrad_rooney_nadp_hplus_deposition_kg.ha",   
#          "conrad_rooney_nadp_k_deposition_kg.ha",  
#          "conrad_rooney_nadp_mg_deposition_kg.ha",
#          "conrad_rooney_nadp_na_deposition_kg.ha", 
#          "conrad_rooney_nadp_so4_deposition_kg.ha",
#          "rindy_conrad_rooney_nh4_deposition_kg.ha",
#          "rindy_conrad_rooney_no3_deposition_kg.ha",
#          "rindy_conrad_rooney_totaln_deposition_kg.ha", 
#          "perc_algal_parasite", "perc_animal_endosymbiont",
#          "perc_animal_parasite", "perc_amf", "perc_arthropod_associated",
#          "perc_dung_saprotroph", "perc_ecm", "perc_epiphyte",
#          "perc_foliar_endophyte", "perc_lichen_parasite",
#          "perc_lichenized", "perc_litter_saprotroph",
#          "perc_moss_symbiont", "perc_mycoparasite",
#          "perc_nectar_tap_saprotroph", "perc_plant_pathogen",
#          "perc_pollen_saprotroph", "perc_root_endophyte",
#          "perc_soil_saprotroph", "perc_sooty_mold",
#          "perc_wood_saprotroph", "perc_all_saprotroph",
#          "perc_all_pathotroph", "perc_all_symbiont",
#          "perc_animal_pathogen", "perc_human_pathogen",
#          "perc_opportunistic_pathogen", "perc_root_associated_pathogen",
#          "perc_leaf_fruit_seed_pathogen", "perc_wood_pathogen", "shannon",
#          "shannon_nopath")
# 
# env <- pca_leaf_df_its[,which(colnames(pca_leaf_df_its) %in% env_names)]
# 
# env_fit <- envfit(pca_summary_leaf_its, env, perm=100000, na.rm=TRUE)
# 
# env.values <- (env_fit$vectors[1]$arrows)
# pval <- (env_fit$vectors[4]$pvals)
# env.arrows <- cbind (env.values, pval)
# env.arrows <- data.frame(subset (env.arrows, pval < 0.05))
# row.names(env.arrows) <- c("available soil NO<sub>3</sub><sup>-</sup>", "soil soluble salt",
#                            "NO deposition", "NH<sub>4</sub><sup>+</sup> deposition", "total N deposition",
#                            "foliar endophytes", "pollen saprotrophs", "Shannon diversity")
# 
# # Visualize
# rate <- 5
# col <- c(rep("gray40",5),rep("black",3))

pca_fungi_leaf <- ggplot(pca_leaf_df_its, 
                         aes(x = PC1, y = PC2, color = tree_pit_type)) +
  geom_point(shape = 18, alpha = 0.5, size = 2.5) +
  xlab(paste("Principal Component 1\n(", 
             round(pca_variance_leaf_its[1] * 100, 2), "% variance)", sep = "")) +
  ylab(paste("Principal Component 2\n(", 
             round(pca_variance_leaf_its[2] * 100, 2), "% variance)", sep = "")) +
  # geom_segment(data=env.arrows, aes(x = 0, y = 0, xend = (PC1*rate), yend = (PC2*rate)), 
  #              arrow = arrow(length = unit(0.2,"cm")),color=col) +
  # geom_text_repel(data=env.arrows, aes(x=(PC1*rate), y=(PC2*rate), 
  #                                      label=rownames(env.arrows)),  size=3, color=col) +
  theme_classic() + stat_ellipse() + 
  labs(shape = "Sample Type", color = "Tree Type") +
  theme(text = element_text(family = "Arial"),
        axis.title.x = element_text(size = 10, color="black"), 
        axis.title.y = element_text(size = 10), 
        axis.text.x = element_text(size = 10, color = "black"), 
        axis.text.y = element_text(size = 10, color = "black"), 
        legend.text = element_text(size = 7), 
        legend.title = element_text(size = 1), 
        plot.title = element_text(size = 10), legend.position = "bottom") +
  ggtitle("Fungal community in leaves") + scale_color_manual(values = pal)
pca_fungi_leaf

ggsave("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Figures/figureS1_fungalleafPCA_wlegend.pdf", device = cairo_pdf, plot = pca_fungi_leaf, dpi = 300, width = 8, height = 3.5, units = "in")

dec.degrees.its<-as.matrix(cbind(pca_leaf_df_its$tree_longitude, pca_leaf_df_its$tree_latitude))
rownames(dec.degrees.its) <-pca_leaf_df_its$Sample 
space.matrix_its<-rdist.earth(dec.degrees.its[,1:2],miles=FALSE)

design_its <- as.data.frame(matrix(nrow = nrow(pca_leaf_df_its), ncol = 5, data = 0))
rownames(design_its) <- pca_leaf_df_its$Sample
colnames(design_its) <- c("street_tree", "urban_edge", "urban_interior", 
                          "rural_edge", "rural_interior")
for(i in 1:nrow(design_its)){
  tree_type <- pca_leaf_df_its$tree_pit_type[which(pca_leaf_df_its$Sample == rownames(design_its)[i])]
  if(tree_type == "Street Tree"){
    design_its$street_tree[i] <- 1
  } else if(tree_type == "Urban Edge"){
    design_its$urban_edge[i] <- 1
  } else if(tree_type == "Urban Interior"){
    design_its$urban_interior[i] <- 1
  } else if(tree_type == "Rural Edge"){
    design_its$rural_edge[i] <- 1
  } else if(tree_type == "Rural Interior"){
    design_its$rural_interior[i] <- 1
  }
}

design.dist_its <- dist(design_its)
# mantel <- mantel(leaf_aitch_its, space.matrix_its, method = "spearman", permutations = 99999, na.rm = TRUE)
# mantel

MRM <- lm(as.numeric(as.dist(leaf_aitch_its)) ~  as.numeric(design.dist_its) + as.numeric(as.dist(space.matrix_its)))
anova(MRM)
```

Figure S1b
```{r PCA -- bacteria leaf}
metadata_leaf_16s <- metadata_16s_200
metadata_leaf_16s <- metadata_leaf_16s[which(metadata_leaf_16s$sample_type == 
                                               "Leaf"),]
leaf_aitch_16s <- leaf_aitch_16s[which(row.names(leaf_aitch_16s) %in%
                                         metadata_leaf_16s$sample_name),
                                 which(colnames(leaf_aitch_16s) %in%
                                         metadata_leaf_16s$sample_name)]
metadata_leaf_16s <- metadata_leaf_16s[which(metadata_leaf_16s$sample_name %in%
                                               colnames(leaf_aitch_16s)),]

pca_result_leaf_16s <- prcomp(leaf_aitch_16s, scale. = TRUE)
pca_leaf_df_16s <- data.frame(Sample = rownames(leaf_aitch_16s), 
                              PC1 = pca_result_leaf_16s$x[,1], 
                              PC2 = pca_result_leaf_16s$x[,2])

pca_leaf_df_16s <- merge(pca_leaf_df_16s, metadata_leaf_16s, 
                         by.x = "Sample", by.y = "sample_name", all = T)

pca_summary_leaf_16s <- summary(pca_result_leaf_16s)
pca_variance_leaf_16s <- pca_summary_leaf_16s$importance[2, ]  # Proportion of variance
pca_variance_cumulative_leaf_16s <- pca_summary_leaf_16s$importance[3, ]  # Cumulative proportion of variance

# env_names <- c("tree_distance_from_edge", "tree_distance_to_boston_km",
#          "tree_dbh_2021", "tree_growth_rate", "soil_temperature",
#          "soil_bulk_density", "soil_ph", "soil_percent_moisture",
#          "soil_percent_organic_matter", "soil_nh4", "soil_no3",
#          "normalized_root_mass", "caron_2018_2019_soil_soluble_salt",
#          "rindy_2019_no_ppb", "rindy_2019_no2_ppb", "rindy_2019_NO<sub>x</sub> Concentration_ppb",
#          "rindy_2019_o3_ppb", "caron_2018_2019_foliar_n_percent",
#          "caron_2019_2019_foliar_c_n", 
#          "conrad_rooney_nadp_ca_deposition_kg.ha",    
#          "conrad_rooney_nadp_cl_deposition_kg.ha",       
#          "conrad_rooney_nadp_hplus_deposition_kg.ha",   
#          "conrad_rooney_nadp_k_deposition_kg.ha",  
#          "conrad_rooney_nadp_mg_deposition_kg.ha",
#          "conrad_rooney_nadp_na_deposition_kg.ha", 
#          "conrad_rooney_nadp_so4_deposition_kg.ha",
#          "rindy_conrad_rooney_nh4_deposition_kg.ha",
#          "rindy_conrad_rooney_no3_deposition_kg.ha",
#          "rindy_conrad_rooney_totaln_deposition_kg.ha", 
#          "perc_c_fixation", "perc_cellulolytic",
#          "perc_chitinolytic", "perc_lignolytic", "perc_carbon_moNO<sub>x</sub> Concentrationide_oxidation",
#          "perc_n_fixation", "perc_copiotroph", "perc_denitrification",
#          "perc_dissim_nitrate_reduction", "perc_hydrocarbon_degradation",
#          "perc_sulfonate_desulfurization", "perc_phosphate_solubilization",
#          "perc_other_n_cycling", "perc_other_p_cycling",
#          "perc_oxidize_reduced_sulfur", "perc_methanotroph",
#          "perc_oligotroph", "perc_partial_nitrification",
#          "perc_archaea", "perc_plant_pathogen",
#          "perc_animal_pathogen", "perc_zoonotic_pathogen",
#          "shannon.x", "shannon_nopath")
# 
# env <- pca_leaf_df_16s[,which(colnames(pca_leaf_df_16s) %in% env_names)]
# 
# env_fit <- envfit(pca_summary_leaf_16s, env, perm=100000, na.rm=TRUE)
# 
# env.values <- (env_fit$vectors[1]$arrows)
# pval <- (env_fit$vectors<sub>4</sub>$pvals)
# env.arrows <- cbind (env.values, pval)
# env.arrows <- data.frame(subset (env.arrows, pval < 0.05))
# row.names(env.arrows) <- c("N fixers", "denitrifiers", "animal pathogens")

# Visualize
# rate <- 7
# col <- c(rep("black",3))

pca_bac_leaf <- ggplot(pca_leaf_df_16s, 
                         aes(x = PC1, y = PC2, color = tree_pit_type)) +
  geom_point(shape = 18, alpha = 0.5, size = 2.5) +
  xlab(paste("Principal Component 1\n(", 
             round(pca_variance_leaf_16s[1] * 100, 2), "% variance)", sep = "")) +
  ylab(paste("Principal Component 2\n(", 
             round(pca_variance_leaf_16s[2] * 100, 2), "% variance)", sep = "")) +
  # geom_segment(data=env.arrows, aes(x = 0, y = 0, xend = (PC1*rate), yend = (PC2*rate)), 
  #              arrow = arrow(length = unit(0.2,"cm")),color=col) +
  # geom_text_repel(data=env.arrows, aes(x=(PC1*rate), y=(PC2*rate), 
  #                                      label=rownames(env.arrows)),  size=3, color=col) +
  theme_classic() + stat_ellipse() + 
  labs(shape = "Sample Type", color = "Tree Type") +
  theme(text = element_text(family = "Arial"),
        axis.title.x = element_text(size = 10, color="black"), 
        axis.title.y = element_text(size = 10), 
        axis.text.x = element_text(size = 10, color = "black"), 
        axis.text.y = element_text(size = 10, color = "black"), 
        legend.text = element_text(size = 5), 
        legend.title = element_text(size = 1), 
        plot.title = element_text(size = 10), legend.position = "none") +
  ggtitle("Bacterial community in leaves") + scale_color_manual(values = pal)
pca_bac_leaf

ggsave("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Figures/figureS1_bacleafPCA.pdf", device = cairo_pdf, plot = pca_bac_leaf, dpi = 300, width = 5, height = 3.5, units = "in")


dec.degrees.16s<-as.matrix(cbind(pca_leaf_df_16s$tree_longitude, pca_leaf_df_16s$tree_latitude))
rownames(dec.degrees.16s) <-pca_leaf_df_16s$Sample 
space.matrix_16s<-rdist.earth(dec.degrees.16s[,1:2],miles=FALSE)

design_16s <- as.data.frame(matrix(nrow = nrow(pca_leaf_df_16s), ncol = 5, data = 0))
rownames(design_16s) <- pca_leaf_df_16s$Sample
colnames(design_16s) <- c("street_tree", "urban_edge", "urban_interior", 
                          "rural_edge", "rural_interior")
for(i in 1:nrow(design_16s)){
  tree_type <- pca_leaf_df_16s$tree_pit_type[which(pca_leaf_df_16s$Sample == rownames(design_16s)[i])]
  if(tree_type == "Street Tree"){
    design_16s$street_tree[i] <- 1
  } else if(tree_type == "Urban Edge"){
    design_16s$urban_edge[i] <- 1
  } else if(tree_type == "Urban Interior"){
    design_16s$urban_interior[i] <- 1
  } else if(tree_type == "Rural Edge"){
    design_16s$rural_edge[i] <- 1
  } else if(tree_type == "Rural Interior"){
    design_16s$rural_interior[i] <- 1
  }
}

design.dist_16s <- dist(design_16s)
# mantel <- mantel(leaf_aitch_16s, space.matrix_16s, method = "spearman", permutations = 99999, na.rm = TRUE)
# mantel

MRM <- lm(as.numeric(as.dist(leaf_aitch_16s)) ~  as.numeric(design.dist_16s) + as.numeric(as.dist(space.matrix_16s)))
anova(MRM)
```

Figure S1c
```{r PCA -- fungi belowground}
shapes <- c(16, 17, 15, 8)
soil_aitch_its <- all_aitch_its
metadata_soil_its <- metadata_its
metadata_soil_its <- metadata_soil_its[which(metadata_soil_its$sample_type != 
                                               "Leaf"),]
soil_aitch_its <- soil_aitch_its[which(row.names(soil_aitch_its) %in%
                                         metadata_soil_its$sample_name),
                                 which(colnames(soil_aitch_its) %in%
                                         metadata_soil_its$sample_name)]

pca_result_soil_its <- prcomp(soil_aitch_its, scale. = TRUE)
pca_soil_df_its <- data.frame(Sample = rownames(soil_aitch_its), 
                              PC1 = pca_result_soil_its$x[,1], 
                              PC2 = pca_result_soil_its$x[,2])

pca_soil_df_its <- merge(pca_soil_df_its, metadata_soil_its, 
                         by.x = "Sample", by.y = "sample_name", all.x = T)

pca_summary_soil_its <- summary(pca_result_soil_its)
pca_variance_soil_its <- pca_summary_soil_its$importance[2, ]  # Proportion of variance
pca_variance_cumulative_soil_its <- pca_summary_soil_its$importance[3, ]  # Cumulative proportion of variance

# env_names <- c("tree_distance_from_edge", "tree_distance_to_boston_km",
#          "tree_dbh_2021", "tree_growth_rate", "soil_temperature",
#          "soil_bulk_density", "soil_ph", "soil_percent_moisture",
#          "soil_percent_organic_matter", "soil_nh4", "soil_no3",
#          "normalized_root_mass", "caron_2018_2019_soil_soluble_salt",
#          "rindy_2019_no_ppb", "rindy_2019_no2_ppb", "rindy_2019_NO<sub>x</sub> Concentration_ppb",
#          "rindy_2019_o3_ppb", "caron_2018_2019_foliar_n_percent",
#          "caron_2019_2019_foliar_c_n", 
#          "conrad_rooney_nadp_ca_deposition_kg.ha",    
#          "conrad_rooney_nadp_cl_deposition_kg.ha",       
#          "conrad_rooney_nadp_hplus_deposition_kg.ha",   
#          "conrad_rooney_nadp_k_deposition_kg.ha",  
#          "conrad_rooney_nadp_mg_deposition_kg.ha",
#          "conrad_rooney_nadp_na_deposition_kg.ha", 
#          "conrad_rooney_nadp_so4_deposition_kg.ha",
#          "rindy_conrad_rooney_nh4_deposition_kg.ha",
#          "rindy_conrad_rooney_no3_deposition_kg.ha",
#          "rindy_conrad_rooney_totaln_deposition_kg.ha", 
#          "perc_algal_parasite", "perc_animal_endosymbiont",
#          "perc_animal_parasite", "perc_amf", "perc_arthropod_associated",
#          "perc_dung_saprotroph", "perc_ecm", "perc_epiphyte",
#          "perc_foliar_endophyte", "perc_lichen_parasite",
#          "perc_lichenized", "perc_litter_saprotroph",
#          "perc_moss_symbiont", "perc_mycoparasite",
#          "perc_nectar_tap_saprotroph", "perc_plant_pathogen",
#          "perc_pollen_saprotroph", "perc_root_endophyte",
#          "perc_soil_saprotroph", "perc_sooty_mold",
#          "perc_wood_saprotroph", "perc_all_saprotroph",
#          "perc_all_pathotroph", "perc_all_symbiont",
#          "perc_animal_pathogen", "perc_human_pathogen",
#          "perc_opportunistic_pathogen", "perc_root_associated_pathogen",
#          "perc_leaf_fruit_seed_pathogen", "perc_wood_pathogen", "shannon",
#          "shannon_nopath")
# 
# env <- pca_soil_df_its[,which(colnames(pca_soil_df_its) %in% env_names)]
# 
# env_fit <- envfit(pca_summary_soil_its, env, perm=100000, na.rm=TRUE)
# 
# env.values <- (env_fit$vectors[1]$arrows)
# pval <- (env_fit$vectors<sub>4</sub>$pvals)
# env.arrows <- cbind (env.values, pval)
# env.arrows <- data.frame(subset (env.arrows, pval < 0.05))
# row.names(env.arrows) <- c("distance from forest edge", "tree DBH",
#                            "tree growth rate", "soil bulk density", "soil pH",
#                            "soil % moisture", "soil % organic matter", 
#                            "available soil NH<sub>4</sub><sup>+</sup>", "root biomass", 
#                            "soil soluble salt", "NO deposition", 
#                            "NO2 deposition", "NO<sub>x</sub> Concentration", 
#                            "O<sub>3</sub> Concentration", "foliar N%", "Ca deposition", 
#                            "Cl deposition", "H+ deposition", "K deposition",
#                            "Mg deposition", "Na deposition", "SO4 deposition",
#                            "NH<sub>4</sub><sup>+</sup> deposition", "NO<sub>3</sub><sup>-</sup> deposition", 
#                            "total N deposition", "animal endosymbionts", 
#                            "animal parasites", "dung saprotrophs", 
#                            "foliar endophytes", "mycoparasites", 
#                            "nectar/tap saprotrophs", "plant pathogens", 
#                            "pollen saprotrophs", "root endophytes", 
#                            "soil saprotrophs", "wood saprotrophs", 
#                            "all saprotrophs", "all symbionts", 
#                            "animal pathogens", "human pathogens", 
#                            "opportunistic pathogens", "root-associated pathogens",
#                            "leaf/fruit/seed pathogens", "Shannon diversity",
#                            "Shannon diversity without pathogens")

# # Visualize
# rate <- 5
# col <- c(rep("gray40",25),rep("black",nrow(env.arrows)-25))

pca_fungi_soil <- ggplot(pca_soil_df_its, 
                         aes(x = PC1, y = PC2, color = tree_pit_type)) +
  geom_point(aes(shape = sample_type), alpha = 0.5, size = 2) +
  xlab(paste("Principal Component 1 \n(", 
             round(pca_variance_soil_its[1] * 100, 2), "% variance)", sep = "")) +
  ylab(paste("Principal Component 2 \n(", 
             round(pca_variance_soil_its[2] * 100, 2), "% variance)", sep = "")) +
  # geom_segment(data=env.arrows, aes(x = 0, y = 0, xend = (PC1*rate), yend = (PC2*rate)), 
  #              arrow = arrow(length = unit(0.2,"cm")),color=col) +
  # geom_text_repel(data=env.arrows, aes(x=(PC1*rate), y=(PC2*rate), 
  #                                      label=rownames(env.arrows)),  size=3, color=col) +
  theme_classic() + stat_ellipse() + 
  labs(shape = "Sample Type", color = "Tree Type") +
  theme(text = element_text(family = "Arial"),
        axis.title.x = element_text(size = 10, color="black"), 
        axis.title.y = element_text(size = 10), 
        axis.text.x = element_text(size = 10, color = "black"), 
        axis.text.y = element_text(size = 10, color = "black"), 
        legend.text = element_text(size = 5), 
        legend.title = element_text(size = 1), 
        plot.title = element_text(size = 10), legend.position = "none") +
  ggtitle("Fungal community belowground") + scale_color_manual(values = pal, guide = "none") + scale_shape_manual(values = shapes)
pca_fungi_soil

ggsave("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Figures/figureS1_fungibgPCA.pdf", device = cairo_pdf, plot = pca_fungi_soil, dpi = 300, width = 5, height = 3.5, units = "in")


dec.degrees.its<-as.matrix(cbind(pca_soil_df_its$tree_longitude, pca_soil_df_its$tree_latitude))
rownames(dec.degrees.its) <-pca_soil_df_its$Sample 
space.matrix_its<-rdist.earth(dec.degrees.its[,1:2],miles=FALSE)

design_its <- as.data.frame(matrix(nrow = nrow(pca_soil_df_its), ncol = 5, data = 0))
rownames(design_its) <- pca_soil_df_its$Sample
colnames(design_its) <- c("street_tree", "urban_edge", "urban_interior", 
                          "rural_edge", "rural_interior")
for(i in 1:nrow(design_its)){
  tree_type <- pca_soil_df_its$tree_pit_type[which(pca_soil_df_its$Sample == rownames(design_its)[i])]
  if(tree_type == "Street Tree"){
    design_its$street_tree[i] <- 1
  } else if(tree_type == "Urban Edge"){
    design_its$urban_edge[i] <- 1
  } else if(tree_type == "Urban Interior"){
    design_its$urban_interior[i] <- 1
  } else if(tree_type == "Rural Edge"){
    design_its$rural_edge[i] <- 1
  } else if(tree_type == "Rural Interior"){
    design_its$rural_interior[i] <- 1
  }
}

design.dist_its <- dist(design_its)
# mantel <- mantel(soil_aitch_its, space.matrix_its, method = "spearman", permutations = 99999, na.rm = TRUE)
# mantel

MRM <- lm(as.numeric(as.dist(soil_aitch_its)) ~  as.numeric(design.dist_its) + as.numeric(as.dist(space.matrix_its)))
anova(MRM)
```

Figure S1d
```{r PCA -- bacteria belowground}
shapes <- c(16, 17, 15, 8)
soil_aitch_16s <- bg_aitch_16s
metadata_soil_16s <- metadata_16s_200
metadata_soil_16s <- metadata_soil_16s[which(metadata_soil_16s$sample_type != 
                                               "Leaf"),]
soil_aitch_16s <- soil_aitch_16s[which(row.names(soil_aitch_16s) %in%
                                         metadata_soil_16s$sample_name),
                                 which(colnames(soil_aitch_16s) %in%
                                         metadata_soil_16s$sample_name)]
metadata_soil_16s <- metadata_soil_16s[which(metadata_soil_16s$sample_name %in%
                                               colnames(soil_aitch_16s)),]

pca_result_soil_16s <- prcomp(soil_aitch_16s, scale. = TRUE)
pca_soil_df_16s <- data.frame(Sample = rownames(soil_aitch_16s), 
                              PC1 = pca_result_soil_16s$x[,1], 
                              PC2 = pca_result_soil_16s$x[,2])

pca_soil_df_16s <- merge(pca_soil_df_16s, metadata_soil_16s, 
                         by.x = "Sample", by.y = "sample_name", all.x = T)

pca_summary_soil_16s <- summary(pca_result_soil_16s)
pca_variance_soil_16s <- pca_summary_soil_16s$importance[2, ]  # Proportion of variance
pca_variance_cumulative_soil_16s <- pca_summary_soil_16s$importance[3, ]  # Cumulative proportion of variance

# env_names <- c("tree_distance_from_edge", "tree_distance_to_boston_km",
#          "tree_dbh_2021", "tree_growth_rate", "soil_temperature",
#          "soil_bulk_density", "soil_ph", "soil_percent_moisture",
#          "soil_percent_organic_matter", "soil_nh4", "soil_no3",
#          "normalized_root_mass", "caron_2018_2019_soil_soluble_salt",
#          "rindy_2019_no_ppb", "rindy_2019_no2_ppb", "rindy_2019_NO<sub>x</sub> Concentration_ppb",
#          "rindy_2019_o3_ppb", "caron_2018_2019_foliar_n_percent",
#          "caron_2019_2019_foliar_c_n", 
#          "conrad_rooney_nadp_ca_deposition_kg.ha",    
#          "conrad_rooney_nadp_cl_deposition_kg.ha",       
#          "conrad_rooney_nadp_hplus_deposition_kg.ha",   
#          "conrad_rooney_nadp_k_deposition_kg.ha",  
#          "conrad_rooney_nadp_mg_deposition_kg.ha",
#          "conrad_rooney_nadp_na_deposition_kg.ha", 
#          "conrad_rooney_nadp_so4_deposition_kg.ha",
#          "rindy_conrad_rooney_nh4_deposition_kg.ha",
#          "rindy_conrad_rooney_no3_deposition_kg.ha",
#          "rindy_conrad_rooney_totaln_deposition_kg.ha", 
#          "perc_c_fixation", "perc_cellulolytic",
#          "perc_chitinolytic", "perc_lignolytic", "perc_carbon_moNO<sub>x</sub> Concentrationide_oxidation",
#          "perc_n_fixation", "perc_copiotroph", "perc_denitrification",
#          "perc_dissim_nitrate_reduction", "perc_hydrocarbon_degradation",
#          "perc_sulfonate_desulfurization", "perc_phosphate_solubilization",
#          "perc_other_n_cycling", "perc_other_p_cycling",
#          "perc_oxidize_reduced_sulfur", "perc_methanotroph",
#          "perc_oligotroph", "perc_partial_nitrification",
#          "perc_archaea", "perc_plant_pathogen",
#          "perc_animal_pathogen", "perc_zoonotic_pathogen",
#          "shannon.x", "shannon_nopath")
# 
# env <- pca_soil_df_16s[,which(colnames(pca_soil_df_16s) %in% env_names)]
# 
# env_fit <- envfit(pca_summary_soil_16s, env, perm=100000, na.rm=TRUE)
# 
# env.values <- (env_fit$vectors[1]$arrows)
# pval <- (env_fit$vectors<sub>4</sub>$pvals)
# env.arrows <- cbind (env.values, pval)
# env.arrows <- data.frame(subset (env.arrows, pval < 0.05))
# row.names(env.arrows) <- c("distance from forest edge", "available soil NO<sub>3</sub><sup>-</sup>",
#                            "root biomass", "soil soluble salt", "copiotrophs",
#                            "methanotrophs", "animal pathogens", 
#                            "Shannon diversity", 
#                            "Shannon diversity without pathogens")
# 
# # Visualize
# rate <- 7
# col <- c(rep("gray40",4),rep("black",5))

pca_bac_soil <- ggplot(pca_soil_df_16s, 
                         aes(x = PC1, y = PC2, color = tree_pit_type)) +
  geom_point(aes(shape = sample_type), alpha = 0.5, size = 2) +
  xlab(paste("Principal Component 1 \n(", 
             round(pca_variance_soil_16s[1] * 100, 2), "% variance)", sep = "")) +
  ylab(paste("Principal Component 2 \n(", 
             round(pca_variance_soil_16s[2] * 100, 2), "% variance)", sep = "")) +
  # geom_segment(data=env.arrows, aes(x = 0, y = 0, xend = (PC1*rate), yend = (PC2*rate)), 
  #              arrow = arrow(length = unit(0.2,"cm")),color=col) +
  # geom_text_repel(data=env.arrows, aes(x=(PC1*rate), y=(PC2*rate), 
  #                                      label=rownames(env.arrows)),  size=3, color=col) +
  theme_classic() + stat_ellipse() + 
  labs(shape = "Sample Type", color = "Tree Type") +
  theme(text = element_text(family = "Arial"),
        axis.title.x = element_text(size = 10, color="black"), 
        axis.title.y = element_text(size = 10), 
        axis.text.x = element_text(size = 10, color = "black"), 
        axis.text.y = element_text(size = 10, color = "black"), 
        legend.text = element_text(size = 5), 
        legend.title = element_text(size = 1), 
        plot.title = element_text(size = 10), legend.position = "none") +
  ggtitle("Bacterial community belowground") + scale_color_manual(values = pal, guide = "none") + scale_shape_manual(values = shapes)
pca_bac_soil

ggsave("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Figures/figureS1_bacbgPCA.pdf", device = cairo_pdf, plot = pca_bac_soil, dpi = 300, width = 5, height = 3.5, units = "in")


dec.degrees.16s<-as.matrix(cbind(pca_soil_df_16s$tree_longitude, pca_soil_df_16s$tree_latitude))
rownames(dec.degrees.16s) <-pca_soil_df_16s$Sample 
space.matrix_16s<-rdist.earth(dec.degrees.16s[,1:2],miles=FALSE)

design_16s <- as.data.frame(matrix(nrow = nrow(pca_soil_df_16s), ncol = 5, data = 0))
rownames(design_16s) <- pca_soil_df_16s$Sample
colnames(design_16s) <- c("street_tree", "urban_edge", "urban_interior", 
                          "rural_edge", "rural_interior")
for(i in 1:nrow(design_16s)){
  tree_type <- pca_soil_df_16s$tree_pit_type[which(pca_soil_df_16s$Sample == rownames(design_16s)[i])]
  if(tree_type == "Street Tree"){
    design_16s$street_tree[i] <- 1
  } else if(tree_type == "Urban Edge"){
    design_16s$urban_edge[i] <- 1
  } else if(tree_type == "Urban Interior"){
    design_16s$urban_interior[i] <- 1
  } else if(tree_type == "Rural Edge"){
    design_16s$rural_edge[i] <- 1
  } else if(tree_type == "Rural Interior"){
    design_16s$rural_interior[i] <- 1
  }
}

design.dist_16s <- dist(design_16s)
# mantel <- mantel(soil_aitch_16s, space.matrix_16s, method = "spearman", permutations = 99999, na.rm = TRUE)
# mantel

MRM <- lm(as.numeric(as.dist(soil_aitch_16s)) ~  as.numeric(design.dist_16s) + as.numeric(as.dist(space.matrix_16s)))
anova(MRM)
```

Figure S2a
```{r beta dispersion -- leaf fungi}
leaf_aitch_its <- leaf_aitch_its[order(row.names(leaf_aitch_its),
                                       colnames(leaf_aitch_its))]
pca_leaf_df_its <- pca_leaf_df_its[order(pca_leaf_df_its$Sample),]
beta_leaf_its <- betadisper(as.dist(leaf_aitch_its), 
                           group = pca_leaf_df_its$tree_pit_type, 
                           type = "centroid")
anova(beta_leaf_its)
tukey_its <- TukeyHSD(beta_leaf_its)
tukey_df_its <- as.data.frame(tukey_its$group)
x_its <- tukey_df_its$`p adj`
names(x_its) <- rownames(tukey_df_its)
tukey_groups_its <- multcompLetters(x_its)$Letters

beta_leaf_dist_its <- as.data.frame(beta_leaf_its$distances)
beta_leaf_df_its <- merge(beta_leaf_dist_its, metadata_its, 
                         by.x = 0, by.y = "sample_name", leaf = T)
beta_leaf_df_its <- beta_leaf_df_its[order(beta_leaf_df_its$tree_pit_type),]
beta_leaf_df_its$tree_pit_type <- gsub("Interior", "Forest Interior", beta_leaf_df_its$tree_pit_type)
beta_leaf_df_its$tree_pit_type <- gsub("Edge", "Forest Edge", beta_leaf_df_its$tree_pit_type)
beta_leaf_df_its$tree_pit_type <- gsub("Street", "Urban Street", beta_leaf_df_its$tree_pit_type)
beta_leaf_df_its$tree_pit_type <- gsub(" ", "\n", beta_leaf_df_its$tree_pit_type)
beta_leaf_df_its$tree_pit_type <- factor(beta_leaf_df_its$tree_pit_type, 
                               levels = c("Urban\nStreet\nTree", 
                                          "Urban\nForest\nEdge",
                                          "Urban\nForest\nInterior",
                                          "Rural\nForest\nEdge",
                                          "Rural\nForest\nInterior"))

text_positions <- beta_leaf_df_its %>%
  group_by(tree_pit_type) %>%
  summarise(max_value = max(`beta_leaf_its$distances`), .groups = 'drop') %>%
  mutate(text_position = max_value + 0.5) # Add buffer for text
text_positions$label <- c("a", "a", "a", "ab", "b")

beta_dispersion_leaf_fungi <- ggplot(beta_leaf_df_its, 
                aes(x = tree_pit_type, y = `beta_leaf_its$distances`, fill = tree_pit_type)) + 
  geom_boxplot() + theme_classic() + xlab("") + 
  ylab("Beta dispersion") + 
  geom_text(data = text_positions, aes(x = tree_pit_type, y = text_position, label = label), 
            inherit.aes = FALSE, vjust = -0.5, family = "Arial") +
  theme(text = element_text(family = "Arial"),
        axis.title.x = element_text(size = 10, color="black"), 
        axis.title.y = element_text(size = 10), 
        axis.text.x = element_text(size = 10, color = "black"), 
        axis.text.y = element_text(size = 10, color = "black"), 
        legend.text = element_text(size = 7), 
        legend.title = element_text(size = 1), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  scale_fill_manual(values = pal) + ylim(c(20, 55)) +
  ggtitle("Fungal community in leaves") 
beta_dispersion_leaf_fungi

ggsave("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Figures/figureS2_fungalleafbeta.pdf", device = cairo_pdf, plot = beta_dispersion_leaf_fungi, dpi = 300, width = 5, height = 3.5, units = "in")
```

Figure S2b
```{r beta dispersion -- aboveground bacteria}
leaf_aitch_16s <- leaf_aitch_16s[order(row.names(leaf_aitch_16s),
                                       colnames(leaf_aitch_16s))]
pca_leaf_df_16s <- pca_leaf_df_16s[order(pca_leaf_df_16s$Sample),]
beta_leaf_16s <- betadisper(as.dist(leaf_aitch_16s), 
                           group = pca_leaf_df_16s$tree_pit_type, 
                           type = "centroid")
anova(beta_leaf_16s)
tukey_16s <- TukeyHSD(beta_leaf_16s)
tukey_df_16s <- as.data.frame(tukey_16s$group)
x_16s <- tukey_df_16s$`p adj`
names(x_16s) <- rownames(tukey_df_16s)
tukey_groups_16s <- multcompLetters(x_16s)$Letters

beta_leaf_dist_16s <- as.data.frame(beta_leaf_16s$distances)
beta_leaf_df_16s <- merge(beta_leaf_dist_16s, metadata_16s_200[which(metadata_16s_200$sample_name %in% colnames(leaf_aitch_16s)),], 
                         by.x = 0, by.y = "sample_name", leaf = T)
beta_leaf_df_16s <- beta_leaf_df_16s[order(beta_leaf_df_16s$tree_pit_type),]
beta_leaf_df_16s$tree_pit_type <- gsub("Interior", "Forest Interior", beta_leaf_df_16s$tree_pit_type)
beta_leaf_df_16s$tree_pit_type <- gsub("Edge", "Forest Edge", beta_leaf_df_16s$tree_pit_type)
beta_leaf_df_16s$tree_pit_type <- gsub("Street", "Urban Street", beta_leaf_df_16s$tree_pit_type)
beta_leaf_df_16s$tree_pit_type <- gsub(" ", "\n", beta_leaf_df_16s$tree_pit_type)
beta_leaf_df_16s$tree_pit_type <- factor(beta_leaf_df_16s$tree_pit_type, 
                               levels = c("Urban\nStreet\nTree", 
                                          "Urban\nForest\nEdge",
                                          "Urban\nForest\nInterior",
                                          "Rural\nForest\nEdge",
                                          "Rural\nForest\nInterior"))

text_positions <- beta_leaf_df_16s %>%
  group_by(tree_pit_type) %>%
  summarise(max_value = max(`beta_leaf_16s$distances`), .groups = 'drop') %>%
  mutate(text_position = max_value + 0.5) # Add buffer for text
text_positions$label <- c("", "", "", "", "")

beta_dispersion_leaf_bac <- ggplot(beta_leaf_df_16s, 
                aes(x = tree_pit_type, y = `beta_leaf_16s$distances`, fill = tree_pit_type)) + 
  geom_boxplot() + theme_classic() + xlab("") + 
  ylab("Beta dispersion") + 
  geom_text(data = text_positions, aes(x = tree_pit_type, y = text_position, label = label), 
            inherit.aes = FALSE, vjust = -0.5) +
  theme(text = element_text(family = "Arial"),
        axis.title.x = element_text(size = 10, color="black"), 
        axis.title.y = element_text(size = 10), 
        axis.text.x = element_text(size = 10, color = "black"), 
        axis.text.y = element_text(size = 10, color = "black"), 
        legend.text = element_text(size = 7), 
        legend.title = element_text(size = 1), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  scale_fill_manual(values = pal) + ylim(c(5, 35)) +
  ggtitle("Bacterial community in leaves") 
beta_dispersion_leaf_bac

ggsave("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Figures/figureS2_bacleafbeta.pdf", device = cairo_pdf, plot = beta_dispersion_leaf_bac, dpi = 300, width = 5, height = 3.5, units = "in")
```

Figure S2c
```{r beta dispersion -- soil fungi}
soil_aitch_its <- soil_aitch_its[order(row.names(soil_aitch_its),
                                       colnames(soil_aitch_its))]
pca_soil_df_its <- pca_soil_df_its[order(pca_soil_df_its$Sample),]
beta_soil_its <- betadisper(as.dist(soil_aitch_its), 
                           group = pca_soil_df_its$tree_pit_type, 
                           type = "centroid")
anova(beta_soil_its)
tukey_its <- TukeyHSD(beta_soil_its)
tukey_df_its <- as.data.frame(tukey_its$group)
x_its <- tukey_df_its$`p adj`
names(x_its) <- rownames(tukey_df_its)
tukey_groups_its <- multcompLetters(x_its)$Letters

beta_soil_dist_its <- as.data.frame(beta_soil_its$distances)
beta_soil_df_its <- merge(beta_soil_dist_its, metadata_its, 
                         by.x = 0, by.y = "sample_name", soil = T)
beta_soil_df_its <- beta_soil_df_its[order(beta_soil_df_its$tree_pit_type),]
beta_soil_df_its$tree_pit_type <- gsub("Interior", "Forest Interior", beta_soil_df_its$tree_pit_type)
beta_soil_df_its$tree_pit_type <- gsub("Edge", "Forest Edge", beta_soil_df_its$tree_pit_type)
beta_soil_df_its$tree_pit_type <- gsub("Street", "Urban Street", beta_soil_df_its$tree_pit_type)
beta_soil_df_its$tree_pit_type <- gsub(" ", "\n", beta_soil_df_its$tree_pit_type)
beta_soil_df_its$tree_pit_type <- factor(beta_soil_df_its$tree_pit_type, 
                               levels = c("Urban\nStreet\nTree", 
                                          "Urban\nForest\nEdge",
                                          "Urban\nForest\nInterior",
                                          "Rural\nForest\nEdge",
                                          "Rural\nForest\nInterior"))

text_positions <- beta_soil_df_its %>%
  group_by(tree_pit_type) %>%
  summarise(max_value = max(`beta_soil_its$distances`), .groups = 'drop') %>%
  mutate(text_position = max_value + 0.5) # Add buffer for text
text_positions$label <- c("c", "a", "ab", "b", "c")

beta_dispersion_soil_fungi <- ggplot(beta_soil_df_its, 
                aes(x = tree_pit_type, y = `beta_soil_its$distances`, fill = tree_pit_type)) + 
  geom_boxplot() + theme_classic() + xlab("") + 
  ylab("Beta dispersion") + 
  geom_text(data = text_positions, aes(x = tree_pit_type, y = text_position, label = label,
                                       family = "Arial"), 
            inherit.aes = FALSE, vjust = -0.5) +
  theme(text = element_text(family = "Arial"),
        axis.title.x = element_text(size = 10, color="black"), 
        axis.title.y = element_text(size = 10), 
        axis.text.x = element_text(size = 10, color = "black"), 
        axis.text.y = element_text(size = 10, color = "black"), 
        legend.text = element_text(size = 7), 
        legend.title = element_text(size = 1), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  scale_fill_manual(values = pal) + ylim(c(20, 72)) +
  ggtitle("Fungal community belowground") 
beta_dispersion_soil_fungi

ggsave("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Figures/figureS2_fungalsoilbeta.pdf", device = cairo_pdf, plot = beta_dispersion_soil_fungi, dpi = 300, width = 5, height = 3.5, units = "in")
```

Figure S2d
```{r beta dispersion -- soil bac}
soil_aitch_16s <- soil_aitch_16s[order(row.names(soil_aitch_16s),
                                       colnames(soil_aitch_16s))]
pca_soil_df_16s <- pca_soil_df_16s[order(pca_soil_df_16s$Sample),]
pca_soil_df_16s <- distinct(pca_soil_df_16s)
beta_soil_16s <- betadisper(as.dist(soil_aitch_16s), 
                           group = pca_soil_df_16s$tree_pit_type, 
                           type = "centroid")
anova(beta_soil_16s)
tukey_16s <- TukeyHSD(beta_soil_16s)
tukey_df_16s <- as.data.frame(tukey_16s$group)
x_16s <- tukey_df_16s$`p adj`
names(x_16s) <- rownames(tukey_df_16s)
tukey_groups_16s <- multcompLetters(x_16s)$Letters

beta_soil_dist_16s <- as.data.frame(beta_soil_16s$distances)
beta_soil_df_16s <- merge(beta_soil_dist_16s, metadata_16s_200, 
                         by.x = 0, by.y = "sample_name", soil = T)
beta_soil_df_16s <- beta_soil_df_16s[order(beta_soil_df_16s$tree_pit_type),]
beta_soil_df_16s$tree_pit_type <- gsub("Interior", "Forest Interior", beta_soil_df_16s$tree_pit_type)
beta_soil_df_16s$tree_pit_type <- gsub("Edge", "Forest Edge", beta_soil_df_16s$tree_pit_type)
beta_soil_df_16s$tree_pit_type <- gsub("Street", "Urban Street", beta_soil_df_16s$tree_pit_type)
beta_soil_df_16s$tree_pit_type <- gsub(" ", "\n", beta_soil_df_16s$tree_pit_type)
beta_soil_df_16s$tree_pit_type <- factor(beta_soil_df_16s$tree_pit_type, 
                               levels = c("Urban\nStreet\nTree", 
                                          "Urban\nForest\nEdge",
                                          "Urban\nForest\nInterior",
                                          "Rural\nForest\nEdge",
                                          "Rural\nForest\nInterior"))

text_positions <- beta_soil_df_16s %>%
  group_by(tree_pit_type) %>%
  summarise(max_value = max(`beta_soil_16s$distances`), .groups = 'drop') %>%
  mutate(text_position = max_value + 0.5) # Add buffer for text
text_positions$label <- c("c", "a", "ab", "a", "bc")

beta_dispersion_soil_bac <- ggplot(beta_soil_df_16s, 
                aes(x = tree_pit_type, y = `beta_soil_16s$distances`, fill = tree_pit_type)) + 
  geom_boxplot() + theme_classic() + xlab("") + 
  ylab("Beta dispersion") + 
  geom_text(data = text_positions, aes(x = tree_pit_type, y = text_position, label = label,
                                       family = "Arial"), 
            inherit.aes = FALSE, vjust = -0.5) +
  theme(text = element_text(family = "Arial"),
        axis.title.x = element_text(size = 10, color="black"), 
        axis.title.y = element_text(size = 10), 
        axis.text.x = element_text(size = 10, color = "black"), 
        axis.text.y = element_text(size = 10, color = "black"), 
        legend.text = element_text(size = 7), 
        legend.title = element_text(size = 1), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  scale_fill_manual(values = pal) + ylim(c(25, 140)) +
  ggtitle("Bacterial community belowground") 
beta_dispersion_soil_bac

ggsave("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Figures/figureS2_bacsoilbeta.pdf", device = cairo_pdf, plot = beta_dispersion_soil_bac, dpi = 300, width = 5, height = 3.5, units = "in")
```

Figure S3a
```{r fungi wood saprotroph roots dfb}
metadata_soil_its <- metadata_its[which(metadata_its$sample_type %in% c("Roots, 0-15 cm",
                                                                        "Roots, 15-30 cm")),]
metadata_soil_its$tree_distance_from_edge <- as.numeric(metadata_soil_its$tree_distance_from_edge)

metadata_soil_16s <- metadata_16s_200[which(metadata_16s_200$sample_type %in% c("Roots, 0-15 cm",
                                                                        "Roots, 15-30 cm")),]
metadata_soil_16s$tree_distance_from_edge <- as.numeric(metadata_soil_16s$tree_distance_from_edge)

fungal_trait_soil <- metadata_soil_its[, colnames(metadata_soil_its) %in% 
                                    fungal_trait_names]
bacterial_trait_soil <- metadata_soil_16s[, colnames(metadata_soil_16s) %in% 
                                    bacterial_trait_names]
standard_error <- function(x) sd(x) / sqrt(length(x)) 

mean_fungi <- aggregate(fungal_trait_soil,
                  by=list(metadata_soil_its$tree_urban_type,
                          metadata_soil_its$tree_edge_type), FUN= "mean")
mean_bac <- aggregate(bacterial_trait_soil,
                  by=list(metadata_soil_16s$tree_urban_type,
                          metadata_soil_16s$tree_edge_type), FUN= "mean")
se_fungi <- aggregate(fungal_trait_soil,
                by=list(metadata_soil_its$tree_urban_type,
                        metadata_soil_its$tree_edge_type), FUN= standard_error)
se_bac <- aggregate(bacterial_trait_soil,
                by=list(metadata_soil_16s$tree_urban_type,
                        metadata_soil_16s$tree_edge_type), FUN= standard_error)

vdata <- cbind(mean_fungi[,1:2], #mean_fungi$perc_ecm, se_fungi$perc_ecm,
               mean_fungi$perc_wood_saprotroph, se_fungi$perc_wood_saprotroph,
               mean_bac$perc_cellulolytic, se_bac$perc_cellulolytic,
               mean_bac$perc_lignolytic, se_bac$perc_lignolytic)

colnames(vdata) <- c("Site", "Edge", #"ecm", "ecm_se", 
                     "wood_sap", "wood_sap_se", "cellulolytic",
                     "cellulolytic_se", "lignolytic", "lignolytic_se")
vdata$Edge <- factor(vdata$Edge, 
                     levels = c("Urban Street Tree", "Forest Edge", 
                                "Forest Interior"))

vdata_long <- as.data.frame(matrix(data = NA, nrow = 15,
                                   #nrow = 20, 
                                   ncol = 5))
colnames(vdata_long) <- c("Site", "Edge", "Guild", "Abundance", "SE")
vdata_long$Site <- rep(vdata$Site, #4
                       3)
vdata_long$Edge <- rep(vdata$Edge, #4
                       3)
vdata_long$Guild <- c(#rep("ECM", 5), 
                      rep("Fungal wood decomposers", 5),
                      rep("Cellulolytic bacteria", 5),
                      rep("Lignolytic bacteria", 5))
vdata_long$Abundance <- c(#vdata$ecm, 
                          vdata$wood_sap, 
                          vdata$cellulolytic, 
                          vdata$lignolytic)
vdata_long$SE <- c(#vdata$ecm_se, 
  vdata$wood_sap_se, 
  vdata$cellulolytic_se, 
  vdata$lignolytic_se)

vdata_long$site_edge <- paste0(vdata_long$Site, " ", vdata_long$Edge)
vdata_long$site_edge <- gsub("Urban Street Tree Urban Street Tree", 
                             "Urban Street Tree",
                             vdata_long$site_edge)
vdata_long$site_edge <- factor(vdata_long$site_edge, 
                               levels = c("Urban Street Tree", 
                                          "Urban Forest Edge",
                                          "Urban Forest Interior",
                                          "Rural Forest Edge", 
                                          "Rural Forest Interior"))


metadata_soil_its$site_edge <- paste0(metadata_soil_its$tree_urban_type, " ", metadata_soil_its$tree_edge_type)
metadata_soil_its$site_edge[which(metadata_soil_its$site_edge == "Urban Street Tree Urban Street Tree")] <- "Urban Street Tree"
metadata_soil_16s$site_edge <- paste0(metadata_soil_16s$tree_urban_type, " ", metadata_soil_16s$tree_edge_type)
metadata_soil_16s$site_edge[which(metadata_soil_16s$site_edge == "Urban Street Tree Urban Street Tree")] <- "Urban Street Tree"

shapes <- c(18, 16, 17, 15, 8)

vdata_long$Guild <- factor(vdata_long$Guild, levels = c("Cellulolytic bacteria", 
                                                        "Lignolytic bacteria",
                                                        "Fungal wood decomposers"))
vdata_long$tukey <- c("ab[\"\\u25A0\"]", 
  "ab[\"\\u25A0\"]", 
  "ab[\"\\u25A0\"]", 
  "b[\"\\u25A0\"]", 
  "a[\"\\u25A0\"]",
  "b[\"\\u25CF\"]", 
  "b[\"\\u25CF\"]", 
  "b[\"\\u25CF\"]", 
  "b[\"\\u25CF\"]", 
  "a[\"\\u25CF\"]", 
  "b[\"\\u25B2\"]", 
  "b[\"\\u25B2\"]", 
  "b[\"\\u25B2\"]", 
  "b[\"\\u25B2\"]", 
  "a[\"\\u25B2\"]"
)
vdata_long$site_edge <- gsub(" ", "\n", vdata_long$site_edge)
vdata_long$site_edge <- factor(vdata_long$site_edge, 
                               levels = c("Urban\nStreet\nTree", 
                                          "Urban\nForest\nEdge",
                                          "Urban\nForest\nInterior",
                                          "Rural\nForest\nEdge",
                                          "Rural\nForest\nInterior"))

fungi_ecm_v_frag_bg_v4_woutliers <- ggplot() +
   geom_line(data = vdata_long, 
             aes(x=site_edge, y=Abundance, shape = Guild, group = Guild), 
            linetype = 2) +
  geom_point(data = vdata_long, 
             aes(x=site_edge, y=Abundance, color=site_edge, shape = Guild, 
                 group = Guild), size=5) + 
  geom_errorbar(data = vdata_long, 
                aes(x=site_edge, ymin=Abundance-SE, ymax=Abundance+SE, 
                    color=site_edge), width = 0.2) +
  geom_text_repel(data=vdata_long, direction = "y", segment.color = NA, family = "Arial",
            aes(x = site_edge, y = Abundance, label = tukey, hjust = 1      # Adjust text to the left of the points
            ), size = 3, parse = TRUE, nudge_x = -0.1, nudge_y = 0.1) +
  # geom_point(data = metadata_soil_its, 
  #            aes(x = site_edge, y = perc_ecm, color = site_edge), 
  #            position = position_jitter(width = 0.2), size = 2, alpha = 0.2, stroke = NA) +
  theme_classic() + labs (y="log(Functional group abundance (%)", x="") + 
  theme(text = element_text(family = "Arial"),
        axis.title.x = element_text(size = 10, color="black"), 
        axis.title.y = element_text(size = 10), 
        axis.text.x = element_text(size = 10, color = "black"), 
        axis.text.y = element_text(size = 10, color = "black"), 
        legend.text = element_text(size = 7), 
        legend.title = element_text(size = 1), 
        plot.title = element_text(size = 10), 
        legend.position = c(0.45, 0.8), 
        legend.background = element_rect(fill = NA)) +
  scale_color_manual(values = pal, guide = "none") + 
  scale_fill_manual(values = pal, guide = "none") + # scale_shape_manual(values = shapes) +
  ggtitle("Wood decomposers on tree roots")

fungi_ecm_v_frag_bg_v4_woutliers
ggsave("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Figures/figures3_rootdecomp_fragmentation.pdf", device = cairo_pdf, plot = fungi_ecm_v_frag_bg_v4_woutliers, dpi = 300, width = 6, height = 3.5, units = "in")


model1 <- lme(formula(perc_ecm ~ site_edge), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML", data = metadata_soil_its)
summary(model1)
summary(emmeans(model1, pairwise ~ site_edge))

model1 <- lme(formula(perc_wood_saprotroph ~ site_edge), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML", data = metadata_soil_its)
summary(model1)
summary(emmeans(model1, pairwise ~ site_edge))

model1 <- lme(formula(perc_cellulolytic ~ site_edge), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML", data = metadata_soil_16s)
summary(model1)
summary(emmeans(model1, pairwise ~ site_edge))

model1 <- lme(formula(perc_lignolytic ~ site_edge), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML", data = metadata_soil_16s)
summary(model1)
summary(emmeans(model1, pairwise ~ site_edge))

```

Figure S3b
```{r fungi wood saprotroph  vs. ECM belowground}
metadata_soil_its <- metadata_its[which(metadata_its$sample_type != "Leaf"),]

wood_saprotroph_vs_dfb_ag <- ggplot(metadata_soil_its, 
                          aes(x = perc_ecm, 
                              y = perc_wood_saprotroph)) + 
  geom_point(alpha = 0.5, col = "#003049", aes(shape = sample_type)) +
  xlab("ECM abundance (%)))") + ylab("log(Wood decomposer abundance (%))") +
  labs(shape = "") +
  geom_smooth(method = "lm", alpha = 0.12, col = "#003049", fill = "#003049", 
              aes(x = perc_ecm,
                 y = perc_wood_saprotroph,
                 col = Guild, fill = Guild,
                 ymax = after_stat(y + se * sqrt(length(y))),
                  ymin = after_stat(y - se * sqrt(length(y))))) +
  theme_classic() + 
  theme(text = element_text(family = "Arial"),
        axis.title.x = element_text(size = 10, color="black"), 
        axis.title.y = element_text(size = 10), 
        axis.text.x = element_text(size = 10, color = "black"), 
        axis.text.y = element_text(size = 10, color = "black"), 
        legend.text = element_text(size = 7), 
        legend.title = element_text(size = 1), 
        plot.title = element_text(size = 10),
        legend.position = "bottom") + 
  scale_shape_manual(values = shapes[2:5]) +
  ggtitle("Fungal wood decomposer and ECM abundance belowground")
wood_saprotroph_vs_dfb_ag

ggsave("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Figures/figures3_ecm_v_sap_bg.pdf", device = cairo_pdf, plot = wood_saprotroph_vs_dfb_ag, dpi = 300, width = 6, height = 4, units = "in")

metadata_soil_its$tree_distance_from_edge <- as.numeric(metadata_soil_its$tree_distance_from_edge)
model1 <- lme(formula(perc_ecm ~ perc_wood_saprotroph), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML", data = metadata_soil_its)
anova(model1)
r.squaredGLMM(model1)
```

Figure S3c
```{r wood saprotroph pathogens}
o_sap_paths <- subset(o_rare_its, primary_lifestyle %in% c("wood_saprotroph"))
o_sap_wood_paths <- subset(o_sap_paths, Plant_pathogenic_capacity_template %in% c("leaf/fruit/seed_pathogen", "root_pathogen", "wood_pathogen"))
o_sap_human_paths <- subset(o_sap_paths, Animal_biotrophic_capacity_template %in% c("opportunistic_human_parasite"))
plant_path_abund <- colSums(o_sap_wood_paths[12:253])
plant_path_abund <- data.frame(plant_path_abund)
human_path_abund <- colSums(o_sap_human_paths[12:253])
human_path_abund <- data.frame(human_path_abund)

microbe.data<-merge(human_path_abund, plant_path_abund, by = 0)
metadata_its$sample_name <- gsub(" ","", metadata_its$sample_name)

sample_type<-metadata_its$sample_type[which(metadata_its$sample_name %in% microbe.data$Row.names)]
soil_horizon<-metadata_its$soil_horizon[which(metadata_its$sample_name %in% microbe.data$Row.names)]
DFB<-metadata_its$tree_dist_to_boston_km[which(metadata_its$sample_name %in% microbe.data$Row.names)]
DFE<-metadata_its$tree_distance_from_edge[which(metadata_its$sample_name %in% microbe.data$Row.names)]
plot_name<-metadata_its$plot_name[which(metadata_its$sample_name %in% microbe.data$Row.names)]
site_name<-metadata_its$site_name[which(metadata_its$sample_name %in% microbe.data$Row.names)]
tree_id<-metadata_its$tree_id[which(metadata_its$sample_name %in% microbe.data$Row.names)]
tree_edge_type<-metadata_its$tree_edge_type[which(metadata_its$sample_name %in% microbe.data$Row.names)]
tree_urban_type<-metadata_its$tree_urban_type[which(metadata_its$sample_name %in% microbe.data$Row.names)]
new_tree_type<-paste(tree_urban_type, tree_edge_type, sep = " ")
new_tree_type<-gsub("Urban Street Tree Urban Street Tree", "Urban Street Tree", new_tree_type)
wood_paths<-microbe.data$plant_path_abund[which(microbe.data$Row.names %in% metadata_its$sample_name)]
human_paths<-microbe.data$human_path_abund[which(microbe.data$Row.names %in% metadata_its$sample_name)]

standard_error <- function(x) sd(x) / sqrt(length(x)) 
Results_anova <- NULL
Results_multcomp <- NULL

mean_root_hp <- aggregate(log(human_paths+1)~new_tree_type,FUN= "mean",na.action = na.omit)
se_root_hp <- aggregate(log(human_paths+1)~new_tree_type,FUN= standard_error,na.action = na.omit)
mean_root_pp <- aggregate(log(wood_paths+1)~new_tree_type,FUN= "mean",na.action = na.omit)
se_root_pp <- aggregate(log(wood_paths+1)~new_tree_type,FUN= standard_error,na.action = na.omit)

root_hp <- cbind(mean_root_hp, se_root_hp[,ncol(se_root_hp)],"Potential human pathogen")
root_pp <- cbind(mean_root_pp, se_root_pp[,ncol(se_root_pp)],"Potential plant pathogen")

colnames(root_hp) <- c("site_edge","Abundance","SE","Microbe")
colnames(root_pp) <- c("site_edge","Abundance","SE","Microbe")

vdata <- rbind(root_hp, root_pp)
vdata$max <- vdata$Abundance + vdata$SE

vdata$tukey <- c("", "", "", "", "", "", "", "", "", "")
vdata$site_edge <- gsub(" ", "\n", vdata$site_edge)
vdata$site_edge <- factor(vdata$site_edge, 
                               levels = c("Urban\nStreet\nTree", 
                                          "Urban\nForest\nEdge",
                                          "Urban\nForest\nInterior",
                                          "Rural\nForest\nEdge",
                                          "Rural\nForest\nInterior"))
vdata$Microbe <- factor(vdata$Microbe, levels = c("Potential plant pathogen", "Potential human pathogen"))


pathogens <- ggplot() +
  geom_line(data = vdata, 
             aes(x=site_edge, y=Abundance, group=Microbe), 
            linetype = 2, color = "black") +
  geom_point(data = vdata,
             aes(x=site_edge, y=Abundance, color=site_edge, shape=Microbe), size=5) +
  geom_errorbar(data = vdata,
                aes(x=site_edge, ymin=Abundance-SE, ymax=Abundance+SE,
                    color=site_edge), width = 0.2) +
  geom_text(data = vdata, aes(x = site_edge, y = max, label = tukey), size = 3,
            family = "Arial") +
  theme_classic() + labs (y="log(Potential pathogens abundance (%))", x="") +
  theme(text = element_text(family = "Arial"),
        axis.title.x = element_text(size = 10, color="black"), 
        axis.title.y = element_text(size = 10), 
        axis.text.x = element_text(size = 10, color = "black"), 
        axis.text.y = element_text(size = 10, color = "black"), 
        legend.text = element_markdown(size = 8), 
        legend.title = element_text(size = 1), 
        plot.title = element_text(size = 10),
        legend.position = c(0.75, 0.75)) +
  scale_color_manual(values = pal) + 
  scale_fill_manual(values = pal) +
  guides(color = "none", shape = guide_legend(title = ""))+
  ggtitle("Wood decomposer potential human and plant pathogen abundance in soils")
pathogens

vdata2 <- cbind(DFB, human_paths, wood_paths)
pathogens2 <- ggplot(data = vdata2) +
  geom_smooth(method = "lm",  color = plot_pal[1], fill = plot_pal[1], alpha = 0.12, aes(x=log(DFB+1), y=log(human_paths+1), 
                                               ymax = after_stat(y + se * sqrt(length(y))),
                                 ymin = after_stat(y - se * sqrt(length(y)))))  + 
  geom_smooth(method = "lm",  color = plot_pal[2], fill = plot_pal[2], alpha = 0.12, aes(x=log(DFB+1), y=log(wood_paths+1), 
                                               ymax = after_stat(y + se * sqrt(length(y))),
                                 ymin = after_stat(y - se * sqrt(length(y)))))  + 
  geom_point(aes(x=log(DFB+1), y=log(human_paths+1)), 
             shape = shapes[1], color = plot_pal[1], size=2, alpha = 0.5) +
  geom_point(aes(x=log(DFB+1), y=log(wood_paths+1)), 
             shape = shapes[2], color = plot_pal[2], size=2, alpha = 0.5) +
  theme_classic() + labs (y="log(Potential pathogens abundance (%))", x="log(distance from Boston (km))") +
  theme(text = element_text(family = "Arial"),
        axis.title.x = element_text(size = 10, color="black"), 
        axis.title.y = element_text(size = 10), 
        axis.text.x = element_text(size = 10, color = "black"), 
        axis.text.y = element_text(size = 10, color = "black"), 
        legend.text = element_markdown(size = 8), 
        legend.title = element_text(size = 1), 
        plot.title = element_text(size = 10),
        legend.position = c(0.75, 0.75)) +
  guides(color = "none", shape = guide_legend(title = ""))+
  ggtitle("Wood saprotroph potential human and plant pathogen abundance in soils")
pathogens2
ggsave("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Figures/figureS3_rootpathogens_fragmentation.pdf", device = cairo_pdf, plot = pathogens, dpi = 300, width = 5.75, height = 3.5, units = "in")
ggsave("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Figures/figureS3_rootpathogens_dfb.pdf", device = cairo_pdf, plot = pathogens2, dpi = 300, width = 5.75, height = 3.5, units = "in")


model1 <- lme(formula(log(human_paths+1) ~ new_tree_type), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML")
summary(model1)
summary(emmeans(model1, pairwise ~ new_tree_type))


model2 <- lme(formula(log(wood_paths+1) ~ new_tree_type), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML")
summary(model2)
summary(emmeans(model2, pairwise ~ new_tree_type))

model1 <- lme(formula(log(human_paths+1) ~ log(DFB)*DFE), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML")
summary(model1)

model1 <- lme(formula(log(wood_paths+1) ~ log(DFB)*DFE), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML")
summary(model1)


model2 <- aov(log(human_paths+1) ~ log(DFB)*DFE+Error(site_name)); summary(model2)

```

Figure S4a
```{r inorganic N soil and deposition}
metadata_16s_200$site_edge <- paste0(metadata_16s_200$tree_urban_type, " ", metadata_16s_200$tree_edge_type)
metadata_16s_200$site_edge <- gsub("Urban Street Tree Urban Street Tree", "Urban Street Tree",
                            metadata_16s_200$site_edge)
metadata_16s_200$site_edge <- factor(metadata_16s_200$site_edge, levels = c("Urban Street Tree", 
                                                      "Urban Forest Edge", 
                                                      "Urban Forest Interior", 
                                                      "Rural Forest Edge", 
                                                      "Rural Forest Interior"))
metadata_soil_16s <- metadata_16s_200[which(metadata_16s_200$sample_type == "Soil, 0-15 cm"),]
length(unique(metadata_soil_16s$tree_id))


# Format data
standard_error <- function(x) sd(x) / sqrt(length(x)) 
Results_anova <- NULL
Results_multcomp <- NULL

data <- cbind(metadata_soil_16s$rindy_2019_nh4n_deposition.kgn_ha_yr, metadata_soil_16s$rindy_2019_no3n_deposition.kgn_ha_yr)
colnames(data) <- c("NH<sub>4</sub><sup>+</sup> annual throughfall", "NO<sub>3</sub><sup>-</sup> annual throughfall")

mean <- aggregate(data~site_edge,data = metadata_soil_16s,FUN= "mean",na.action = na.omit)
se <- aggregate(data~site_edge,data = metadata_soil_16s,FUN= standard_error,na.action = na.omit)

melt.mean <- melt(mean)
melt.se <- melt(se)

vdata <- cbind(melt.mean, melt.se[,ncol(melt.se)])
colnames(vdata) <- c("site_edge","Nutirent","Abundance","SE")

vdata$site_edge <- gsub(" ", "\n", vdata$site_edge)
vdata$site_edge <- factor(vdata$site_edge, 
                               levels = c("Urban\nStreet\nTree", 
                                          "Urban\nForest\nEdge",
                                          "Urban\nForest\nInterior",
                                          "Rural\nForest\nEdge",
                                          "Rural\nForest\nInterior"))

vdata$Nutirent <- factor(vdata$Nutirent, levels = c("NH<sub>4</sub><sup>+</sup> annual throughfall", "NO<sub>3</sub><sup>-</sup> annual throughfall"))

# Visualize
pal <- c("#FF8080","#FF0000","#C00000","#800000","#000000")

nitrogen_avail <- ggplot(vdata) +
  geom_line(data = vdata, 
             aes(x=site_edge, y=Abundance, group=Nutirent), 
            linetype = 2, color = "black") +
  geom_point(data = vdata, 
             aes(x=site_edge, y=Abundance, color=site_edge, shape=Nutirent), size=5) + 
  geom_errorbar(data = vdata, 
                aes(x=site_edge, ymin=Abundance-SE, ymax=Abundance+SE, 
                    color=site_edge), width = 0.2) +
  theme_classic() + labs (y="Atmospheric inorganic N deposition (kg/ha)", x="") +
  theme(text = element_text(family = "Arial"),
        axis.title.x = element_text(size = 10, color="black"), 
        axis.title.y = element_text(size = 10), 
        axis.text.x = element_text(size = 10, color = "black"), 
        axis.text.y = element_text(size = 10, color = "black"), 
        legend.text = element_markdown(size = 7), 
        legend.title = element_text(size = 1), 
        plot.title = element_markdown(size = 10), legend.position = c(0.65, 0.5), legend.background = element_rect(fill = NA)) +
  scale_color_manual(values = pal) + 
  scale_fill_manual(values = pal) +
  guides(color = "none", shape = guide_legend(title = ""))+
  ggtitle("Annual atmospheric N throughfall")
nitrogen_avail

ggsave("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Figures/figures3c_new.pdf", device = cairo_pdf, plot = nitrogen_avail, dpi = 300, width = 6, height = 3.5, units = "in")

model1 <- lme(formula(rindy_2019_nh4n_deposition.kgn_ha_yr ~ site_edge), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML", data = metadata_soil_16s)
summary(model1)
summary(emmeans(model1, pairwise ~ site_edge))

model1 <- lme(formula(rindy_2019_no3n_deposition.kgn_ha_yr ~ site_edge), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML", data = metadata_soil_16s)
summary(model1)
summary(emmeans(model1, pairwise ~ site_edge))
```

Figure S4b
```{r soil organic matter}
metadata_16s_200$site_edge <- paste0(metadata_16s_200$tree_urban_type, " ", metadata_16s_200$tree_edge_type)
metadata_16s_200$site_edge <- gsub("Urban Street Tree Urban Street Tree", "Urban Street Tree",
                            metadata_16s_200$site_edge)
metadata_16s_200$site_edge <- factor(metadata_16s_200$site_edge, levels = c("Urban Street Tree", 
                                                      "Urban Forest Edge", 
                                                      "Urban Forest Interior", 
                                                      "Rural Forest Edge", 
                                                      "Rural Forest Interior"))
metadata_soil_16s <- metadata_16s_200[which(metadata_16s_200$sample_type == "Soil, 0-15 cm"),]

# Format data
standard_error <- function(x) sd(x) / sqrt(length(x)) 
Results_anova <- NULL
Results_multcomp <- NULL

data <- cbind(100*metadata_soil_16s$soil_percent_organic_matter)

mean <- aggregate(data~site_edge,data = metadata_soil_16s,FUN= "mean",na.action = na.omit)
se <- aggregate(data~site_edge,data = metadata_soil_16s,FUN= standard_error,na.action = na.omit)

melt.mean <- melt(mean)
melt.se <- melt(se)

vdata <- cbind(melt.mean, melt.se[,ncol(melt.se)])
colnames(vdata) <- c("site_edge","Nutirent","Abundance","SE")

vdata$Nutirent <- c(rep("Soil organic matter",5))
vdata$tukey <- c("b", 
  "a", 
  "a", 
  "a", 
  "a"
)
vdata$site_edge <- gsub(" ", "\n", vdata$site_edge)
vdata$site_edge <- factor(vdata$site_edge, 
                               levels = c("Urban\nStreet\nTree", 
                                          "Urban\nForest\nEdge",
                                          "Urban\nForest\nInterior",
                                          "Rural\nForest\nEdge",
                                          "Rural\nForest\nInterior"))


# Visualize
pal <- c("#FF8080","#FF0000","#C00000","#800000","#000000")
library(ggtext)
fungi_ecm_v_frag_bg_v4_woutliers <- ggplot() +
   geom_line(data = vdata, 
             aes(x=site_edge, y=Abundance, shape = Nutirent, group = Nutirent), 
            linetype = 2) +
  geom_point(data = vdata, 
             aes(x=site_edge, y=Abundance, color=site_edge, shape = Nutirent, 
                 group = Nutirent), size=5) + 
  geom_errorbar(data = vdata, 
                aes(x=site_edge, ymin=Abundance-SE, ymax=Abundance+SE, 
                    color=site_edge), width = 0.2) +
  geom_text(data=vdata, segment.color = NA, family = "Arial",
            aes(x = site_edge, y = Abundance + SE + 1, label = tukey, vjust = 1        # Adjust text to the left of the points
            ), size = 3, parse = TRUE) +
  # geom_point(data = metadata_soil_its, 
  #            aes(x = site_edge, y = perc_ecm, color = site_edge), 
  #            position = position_jitter(width = 0.2), size = 2, alpha = 0.2, stroke = NA) +
  theme_classic() + labs(y="Soil organic matter content (%)", x="", shape = "") +
  theme(text = element_text(family = "Arial"),
        axis.title.x = element_text(size = 10, color="black"), 
        axis.title.y = element_markdown(size = 10), 
        axis.text.x = element_text(size = 10, color = "black"), 
        axis.text.y = element_text(size = 10, color = "black"), 
        legend.text = element_markdown(size = 7), 
        legend.title = element_text(size = 1), 
        plot.title = element_text(size = 10), 
        legend.position = "none", 
        legend.background = element_rect(fill = NA)) +
  scale_color_manual(values = pal, guide = "none") + 
  scale_fill_manual(values = pal, guide = "none") + # scale_shape_manual(values = shapes) +
  ggtitle("Soil nutrient content")
fungi_ecm_v_frag_bg_v4_woutliers

ggsave("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Figures/figures4_soilnutrients_fragmentation.pdf", device = cairo_pdf, plot = fungi_ecm_v_frag_bg_v4_woutliers, dpi = 300, width = 6, height = 3.5, units = "in")

model1 <- lme(formula(soil_percent_organic_matter ~ site_edge), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML", data = metadata_soil_16s)
summary(model1)
summary(emmeans(model1, pairwise ~ site_edge))
```

Figure S4c
```{r soil 0-15cm ectos and wood rots heatmap}
metadata_soil_its <- metadata_its[which(metadata_its$sample_type == "Soil, 0-15 cm"),]
metadata_soil_16s <- metadata_16s_200[which(metadata_16s_200$sample_type == "Soil, 0-15 cm"),]

metadata_leaf_16s <- metadata_16s_200[which(metadata_16s_200$sample_type == "Leaf"),]

fungal_trait_soil_its <- metadata_soil_its[,colnames(metadata_soil_its) %in%
                                        c(fungal_trait_names)]
bacterial_trait_soil_16s <- metadata_soil_16s[,colnames(metadata_soil_16s) %in% c(bacterial_trait_names)]

bacterial_trait_leaf_16s <- metadata_leaf_16s[,colnames(metadata_leaf_16s) %in% c(bacterial_trait_names)]
func_soil_its <- fungal_trait_soil_its

func_soil_its <- as.data.frame(func_soil_its[,which(colnames(func_soil_its) %in% c("perc_dung_saprotroph", "perc_wood_saprotroph",
                                                                                   "perc_litter_saprotroph", "perc_soil_saprotroph"))])

colnames(func_soil_its) <- c("Dung decomposers", "Wood decomposers",
                             "Litter decomposers", "Soil decomposers")
colnames(func_soil_its) <- paste0(colnames(func_soil_its), " in soil")

func_soil_16s <- bacterial_trait_soil_16s
func_leaf_16s <- bacterial_trait_leaf_16s

func_soil_16s <- as.data.frame(func_soil_16s[,which(colnames(func_soil_16s) %in% c("perc_chitinolytic", "perc_n_fixation",
                                                                                   "perc_dissim_nitrate_reduction", "perc_denitrification"))])
func_leaf_16s <- as.data.frame(func_leaf_16s[,which(colnames(func_leaf_16s) %in% c("perc_methanotroph"))])
colnames(func_soil_16s) <- c("Chitinolytic bacteria", 
                             "Nitrifying bacteria", 
                             "Dissimilatory nitrate reducing bacteria",
                             "Denitrifying bacteria")
colnames(func_soil_16s) <- paste0(colnames(func_soil_16s), " in soil")
colnames(func_leaf_16s) <- c("Methanotrophic bacteria")
colnames(func_leaf_16s) <- paste0(colnames(func_leaf_16s), " in leaves")

metadata_soil_heatmap_its <- metadata_soil_its[,c(17,18,20,23,30:36,38,49,53:55,59:61,63:65,67:76)]
metadata_soil_heatmap_16s <- metadata_soil_16s[,c(17,18,20,23,30:36,38,49,53:55,59:61,63:65,67:76)]
metadata_leaf_heatmap_its <- metadata_leaf_its[,c(17,18,20,23,30:36,38,49,53:55,59:61,63:65,67:76)]
metadata_leaf_heatmap_16s <- metadata_leaf_16s[,c(17,18,20,23,30:36,38,49,53:55,59:61,63:65,67:76)]
colnames(metadata_soil_heatmap_its) <- c("Tree distance from edge", 
                                         "Tree distance from Boston", 
                                         "Tree DBH", "Tree growth rate",
                                         "Soil temperature", 
                                         "Soil bulk density", 
                                         "Soil pH", "Soil moisture", 
                                         "Soil organic matter", 
                                         "Soil available NH<sub>4</sub><sup>+</sup>",
                                         "Soil available NO<sub>3</sub><sup>-</sup>", 
                                         "Normalized root biomass", 
                                         "Soil soluble salt",
                                         "NO concentration",
                                         "NO<sub>2</sub> concentration",
                                         "NO<sub>x</sub> concentration",
                                         "NH<sub>4</sub><sup>+</sup> annual throughfall",
                                         "NO<sub>3</sub><sup>-</sup> annual throughfall",
                                         "Total inorganic N annual throughfall",
                                         "O<sub>3</sub> concentration",
                                         "Foliar N %",
                                         "Foliar C:N",
                                         "Ca<sup>2+</sup> annual wet deposition",
                                         "Cl<sup>-</sup> annual wet deposition",
                                         "H<sup>+</sup> annual wet deposition",
                                         "K<sup>+</sup> annual wet deposition",
                                         "Mg<sup>2+</sup> annual wet deposition",
                                         "Na<sup>+</sup> annual wet deposition",
                                         "NH<sub>4</sub><sup>+</sup> annual wet deposition",
                                         "NO<sub>3</sub><sup>-</sup> annual wet deposition",
                                         "SO<sub>4</sub><sup>2-</sup> annual wet deposition",
                                         "Total inorganic N annual wet deposition")
colnames(metadata_soil_heatmap_16s) <- c("Tree distance from edge", 
                                         "Tree distance from Boston", 
                                         "Tree DBH", "Tree growth rate",
                                         "Soil temperature", 
                                         "Soil bulk density", 
                                         "Soil pH", "Soil moisture", 
                                         "Soil organic matter", 
                                         "Soil available NH<sub>4</sub><sup>+</sup>",
                                         "Soil available NO<sub>3</sub><sup>-</sup>", 
                                         "Normalized root biomass", 
                                         "Soil soluble salt",
                                         "NO concentration",
                                         "NO<sub>2</sub> concentration",
                                         "NO<sub>x</sub> concentration",
                                         "NH<sub>4</sub><sup>+</sup> annual throughfall",
                                         "NO<sub>3</sub><sup>-</sup> annual throughfall",
                                         "Total inorganic N annual throughfall",
                                         "O<sub>3</sub> concentration",
                                         "Foliar N %",
                                         "Foliar C:N",
                                         "Ca<sup>2+</sup> annual wet deposition",
                                         "Cl<sup>-</sup> annual wet deposition",
                                         "H<sup>+</sup> annual wet deposition",
                                         "K<sup>+</sup> annual wet deposition",
                                         "Mg<sup>2+</sup> annual wet deposition",
                                         "Na<sup>+</sup> annual wet deposition",
                                         "NH<sub>4</sub><sup>+</sup> annual wet deposition",
                                         "NO<sub>3</sub><sup>-</sup> annual wet deposition",
                                         "SO<sub>4</sub><sup>2-</sup> annual wet deposition",
                                         "Total inorganic N annual wet deposition")
colnames(metadata_leaf_heatmap_16s) <- c("Tree distance from edge", 
                                         "Tree distance from Boston", 
                                         "Tree DBH", "Tree growth rate",
                                         "Soil temperature", 
                                         "Soil bulk density", 
                                         "Soil pH", "Soil moisture", 
                                         "Soil organic matter", 
                                         "Soil available NH<sub>4</sub><sup>+</sup>",
                                         "Soil available NO<sub>3</sub><sup>-</sup>", 
                                         "Normalized root biomass", 
                                         "Soil soluble salt",
                                         "NO concentration",
                                         "NO<sub>2</sub> concentration",
                                         "NO<sub>x</sub> concentration",
                                         "NH<sub>4</sub><sup>+</sup> annual throughfall",
                                         "NO<sub>3</sub><sup>-</sup> annual throughfall",
                                         "Total inorganic N annual throughfall",
                                         "O<sub>3</sub> concentration",
                                         "Foliar N %",
                                         "Foliar C:N",
                                         "Ca<sup>2+</sup> annual wet deposition",
                                         "Cl<sup>-</sup> annual wet deposition",
                                         "H<sup>+</sup> annual wet deposition",
                                         "K<sup>+</sup> annual wet deposition",
                                         "Mg<sup>2+</sup> annual wet deposition",
                                         "Na<sup>+</sup> annual wet deposition",
                                         "NH<sub>4</sub><sup>+</sup> annual wet deposition",
                                         "NO<sub>3</sub><sup>-</sup> annual wet deposition",
                                         "SO<sub>4</sub><sup>2-</sup> annual wet deposition",
                                         "Total inorganic N annual wet deposition")
# Make coefficient matrix
Result_soil_its <- NULL
for (i in 1:ncol(func_soil_its)){
  estimate_soil_its <- NULL
  for (j in 1:ncol(metadata_soil_heatmap_its)){
    cor_soil_its <- cor.test(func_soil_its[,i],metadata_soil_heatmap_its[,j], method="p")
    estimate_soil_its <- c(estimate_soil_its, cor_soil_its$estimate)
  }
  Result_soil_its <- rbind(Result_soil_its, estimate_soil_its)
}
colnames(Result_soil_its) <- colnames(metadata_soil_heatmap_its)
rownames(Result_soil_its) <- c("Dung decomposers in soil", 
                               "Wood decomposers in soil",
                               "Litter decomposers in soil", 
                               "Soil decomposers in soil")

Result_soil_16s <- NULL
for (i in 1:ncol(func_soil_16s)){
  estimate_soil_16s <- NULL
  for (j in 1:ncol(metadata_soil_heatmap_16s)){
    cor_soil_16s <- cor.test(func_soil_16s[,i],metadata_soil_heatmap_16s[,j], method="p")
    estimate_soil_16s <- c(estimate_soil_16s, cor_soil_16s$estimate)
  }
  Result_soil_16s <- rbind(Result_soil_16s, estimate_soil_16s)
}
colnames(Result_soil_16s) <- colnames(metadata_soil_heatmap_16s)
rownames(Result_soil_16s) <- c("Chitinolytic bacteria in soil", 
                             "Nitrifying bacteria in soil", 
                             "Dissimilatory nitrate reducing bacteria in soil",
                             "Denitrifying bacteria in soil")

Result_leaf_16s <- NULL
  estimate_leaf_16s <- NULL
  for (j in 1:ncol(metadata_leaf_heatmap_16s)){
    cor_leaf_16s <- cor.test(func_leaf_16s[,1],metadata_leaf_heatmap_16s[,j], method="p")
    estimate_leaf_16s <- c(estimate_leaf_16s, cor_leaf_16s$estimate)
  }
  Result_leaf_16s <- rbind(Result_leaf_16s, estimate_leaf_16s)
colnames(Result_leaf_16s) <- colnames(metadata_leaf_heatmap_16s)
rownames(Result_leaf_16s) <- "Methanotrophic bacteria in leaves" 

# Make Pval matrix
Result.Pval_soil_its <- NULL
for (i in 1:ncol(func_soil_its)){
  Pval_soil_its <- NULL
  for (j in 1:ncol(metadata_soil_heatmap_its)){
    cor_soil_its <- cor.test(func_soil_its[,i],metadata_soil_heatmap_its[,j], method="p")
    Pval_soil_its <- c(Pval_soil_its, cor_soil_its$p.value)
  }
  Result.Pval_soil_its <- rbind (Result.Pval_soil_its, Pval_soil_its)
}
colnames(Result.Pval_soil_its) <- colnames(metadata_soil_heatmap_its)
rownames(Result.Pval_soil_its) <- c("Dung decomposers in soil", 
                               "Wood decomposers in soil",
                               "Litter decomposers in soil", 
                               "Soil decomposers in soil")

Result.Pval_soil_16s <- NULL
for (i in 1:ncol(func_soil_16s)){
  Pval_soil_16s <- NULL
  for (j in 1:ncol(metadata_soil_heatmap_16s)){
    cor_soil_16s <- cor.test(func_soil_16s[,i],metadata_soil_heatmap_16s[,j], method="p")
    Pval_soil_16s <- c(Pval_soil_16s, cor_soil_16s$p.value)
  }
  Result.Pval_soil_16s <- rbind (Result.Pval_soil_16s, Pval_soil_16s)
}
colnames(Result.Pval_soil_16s) <- colnames(metadata_soil_heatmap_16s)
rownames(Result.Pval_soil_16s) <- c("Chitinolytic bacteria in soil", 
                             "Nitrifying bacteria in soil", 
                             "Dissimilatory nitrate reducing bacteria in soil",
                             "Denitrifying bacteria in soil")

Result.Pval_leaf_16s <- NULL
  Pval_leaf_16s <- NULL
  for (j in 1:ncol(metadata_leaf_heatmap_16s)){
    cor_leaf_16s <- cor.test(func_leaf_16s[,1],metadata_leaf_heatmap_16s[,j], method="p")
    Pval_leaf_16s <- c(Pval_leaf_16s, cor_leaf_16s$p.value)
  }
  Result.Pval_leaf_16s <- rbind (Result.Pval_leaf_16s, Pval_leaf_16s)
colnames(Result.Pval_leaf_16s) <- colnames(metadata_leaf_heatmap_16s)
rownames(Result.Pval_leaf_16s) <- "Methanotrophic bacteria in leaves"

# Make Asterisk matrix
Result.Pval.combined_soil_its <- Result.Pval_soil_its
Result.asterisk_soil_its <- Result.Pval.combined_soil_its
for (k in 1:ncol(Result.Pval.combined_soil_its)){
  for (i in 1:nrow(Result.Pval.combined_soil_its)){
    if(!is.na(Result.Pval.combined_soil_its[i,k])){
      if (Result.Pval.combined_soil_its[i,k] < 0.001){
        Result.asterisk_soil_its[i,k] <- "***"
      } else if (Result.Pval.combined_soil_its[i,k] < 0.01){
        Result.asterisk_soil_its[i,k] <- "**"
      } else if (Result.Pval.combined_soil_its[i,k] < 0.05){
        Result.asterisk_soil_its[i,k] <- "*"
      } else {
        Result.asterisk_soil_its[i,k]<- " "
      }
    } 
  }
}

Result.Pval.combined_soil_16s <- Result.Pval_soil_16s
Result.asterisk_soil_16s <- Result.Pval.combined_soil_16s
for (k in 1:ncol(Result.Pval.combined_soil_16s)){
  for (i in 1:nrow(Result.Pval.combined_soil_16s)){
    if(!is.na(Result.Pval.combined_soil_16s[i,k])){
      if (Result.Pval.combined_soil_16s[i,k] < 0.001){
        Result.asterisk_soil_16s[i,k] <- "***"
      } else if (Result.Pval.combined_soil_16s[i,k] < 0.01){
        Result.asterisk_soil_16s[i,k] <- "**"
      } else if (Result.Pval.combined_soil_16s[i,k] < 0.05){
        Result.asterisk_soil_16s[i,k] <- "*"
      } else {
        Result.asterisk_soil_16s[i,k]<- " "
      }
    } 
  }
}

Result.Pval.combined_leaf_16s <- Result.Pval_leaf_16s
Result.asterisk_leaf_16s <- Result.Pval.combined_leaf_16s
for (k in 1:ncol(Result.Pval.combined_leaf_16s)){
  for (i in 1:nrow(Result.Pval.combined_leaf_16s)){
    if(!is.na(Result.Pval.combined_leaf_16s[i,k])){
      if (Result.Pval.combined_leaf_16s[i,k] < 0.001){
        Result.asterisk_leaf_16s[i,k] <- "***"
      } else if (Result.Pval.combined_leaf_16s[i,k] < 0.01){
        Result.asterisk_leaf_16s[i,k] <- "**"
      } else if (Result.Pval.combined_leaf_16s[i,k] < 0.05){
        Result.asterisk_leaf_16s[i,k] <- "*"
      } else {
        Result.asterisk_leaf_16s[i,k]<- " "
      }
    } 
  }
}
# Combine netowrk propeties + others
Result.combined_soil_its <- Result_soil_its
melt_soil_its <- melt(Result.combined_soil_its)
melt.asterisk_soil_its <- melt(Result.asterisk_soil_its)

Result.combined_soil_16s <- Result_soil_16s
melt_soil_16s <- melt(Result.combined_soil_16s)
melt.asterisk_soil_16s <- melt(Result.asterisk_soil_16s)

Result.combined_leaf_16s <- Result_leaf_16s
melt_leaf_16s <- melt(Result.combined_leaf_16s)
melt.asterisk_leaf_16s <- melt(Result.asterisk_leaf_16s)

melt_all <- rbind(melt_soil_16s, melt_soil_its, melt_leaf_16s)
melt_asterisk_all <- rbind(melt.asterisk_soil_16s, melt.asterisk_soil_its, 
                           melt.asterisk_leaf_16s)

# Visualize
vdata_its <- cbind(melt_all, melt_asterisk_all$value)
colnames(vdata_its) <- c("microbe", "environment", "coef", "Pval")

order_of_env_factors <- vdata_its[which(vdata_its$microbe == "Methanotrophic bacteria in leaves"),]

vdata_its$environment <- factor(vdata_its$environment, 
                            levels = order_of_env_factors$environment[order(order_of_env_factors$coef)])

vdata_its$microbe <- factor(vdata_its$microbe, levels = c("Chitinolytic bacteria in soil",
                                                          "Dissimilatory nitrate reducing bacteria in soil",
                                                          "Denitrifying bacteria in soil",
                                                          "Dung decomposers in soil",
                                                          "Wood decomposers in soil",
                                                          "Soil decomposers in soil",
                                                          "Nitrifying bacteria in soil",
                                                          "Litter decomposers in soil",
                                                          "Methanotrophic bacteria in leaves"))

heatmap_N <- ggplot(vdata_its) +
  geom_tile(aes(y=microbe, x=environment, fill=coef)) +
  geom_text(aes(y=microbe, x=environment, label = Pval), size = 3) +
  labs(x="",y="") + theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, hjust=1,vjust=0.3)) +
  theme(axis.text.x = element_text(angle = 90, hjust=1,vjust=0.3)) + 
  scale_fill_gradient2(low = "#003049", mid = "#FDF0D5", high = "#C00000", 
                       name = "Correlation\ncoefficient") +
  theme(text = element_text(family = "Arial"),
        axis.title.x = element_markdown(size = 8, color = "black"), 
        axis.title.y = element_markdown(size = 8, color = "black"), 
        axis.text.y = element_markdown(size = 8, color = "black"),
        axis.text.x = element_markdown(size = 8, color = "black"),
        legend.text = element_text(size = 8), 
        legend.title = element_text(size = 8), 
        plot.title = element_text(size = 8)) + 
  ggtitle("") + coord_fixed()
heatmap_N

ggsave("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Figures/heatmap_manuscript_figS4b_new.pdf", device = cairo_pdf, width = 10, height = 5, units = "in", dpi = 300)
```


Table S3 - fungal O horizon pathogens
```{r pathogens}
pathogens_fungi_df <- o_rare_its

plant_pathogens_fungi_df <- subset(pathogens_fungi_df, !is.na(Plant_pathogenic_capacity_template))
human_pathogens_fungi_df <- subset(pathogens_fungi_df, Animal_biotrophic_capacity_template %in% c("animal_parasite/opportunistic_human_parasite",
                                                                                                  "invertebrate_parasite", 
                                                                                                  "opportunistic_human_parasite",
                                                                                                  "animal_parasite", "fish_parasite",
                                                                                                  "animal-associated/opportunistic_human_parasite",
                                                                                                  "opportunistic_animal_parasite", "arthropod_parasite",
                                                                                                  "opportunistic_animal_parasite_G._destructans"))
sum(plant_pathogens_fungi_df$Row.names %in% human_pathogens_fungi_df$Row.names)
plant_pathogens_fungi_df$pathogen_type <- paste0(plant_pathogens_fungi_df$Plant_pathogenic_capacity_template, ", ", plant_pathogens_fungi_df$Animal_biotrophic_capacity_template)
human_pathogens_fungi_df$pathogen_type <- paste0(human_pathogens_fungi_df$Plant_pathogenic_capacity_template, ", ", human_pathogens_fungi_df$Animal_biotrophic_capacity_template)
pathogens_fungi_df <- rbind(plant_pathogens_fungi_df, human_pathogens_fungi_df)
plant_pathogens_fungi_df <- distinct(pathogens_fungi_df)

plant_pathogens_fungi_df <- plant_pathogens_fungi_df[,c(5,11:255)]
plant_pathogens_fungi_df <- aggregate(plant_pathogens_fungi_df, .~Row.names+Species+pathogen_type, FUN = "sum")
plant_pathogens_fungi_df_clean <- plant_pathogens_fungi_df %>% filter(!is.na(Species))
plant_paths_df_w_rownames <- column_to_rownames(plant_pathogens_fungi_df_clean, var = "Row.names")
plant_paths_df_t <- as.data.frame(t(plant_paths_df_w_rownames))
plant_paths_df_t$tree_edge_type<-metadata_its$tree_edge_type[match(rownames(plant_paths_df_t), metadata_its$sample_name)]
plant_paths_df_t$tree_urban_type<-metadata_its$tree_urban_type[match(rownames(plant_paths_df_t), metadata_its$sample_name)]
plant_paths_df_t <- plant_paths_df_t %>% filter(!is.na(tree_edge_type))
plant_paths_df_t <- plant_paths_df_t %>% filter(!is.na(tree_urban_type))
plant_paths_df_t$new_tree_type<-paste(plant_paths_df_t$tree_edge_type,plant_paths_df_t$tree_urban_type)
plant_paths_df_t <- plant_paths_df_t[,-c(2312:2313)]
plant_paths_df_t[,1:2311] <- plant_paths_df_t[,1:2311] %>% mutate_if(is.character, as.numeric)
plant_paths_summed <- aggregate(plant_paths_df_t, .~new_tree_type, FUN = "sum")
plant_paths_sums <- colSums(plant_paths_summed[2:2312])
plant_paths_sums <- c(50, plant_paths_sums)
plant_paths_summed <- plant_paths_summed[,which(plant_paths_sums > 0)]
plant_paths_w_rownames <- column_to_rownames(plant_paths_summed, var = "new_tree_type")
plant_paths_summed_df <- as.data.frame(t(plant_paths_w_rownames))
plant_paths_summed_df$Pathogen_type<-plant_pathogens_fungi_df$pathogen_type[match(rownames(plant_paths_summed_df),plant_pathogens_fungi_df$Row.names)]
plant_paths_summed_df$Species<-plant_pathogens_fungi_df$Species[match(rownames(plant_paths_summed_df),plant_pathogens_fungi_df$Row.names)]
plant_paths_summed_df$perc_id<-its_pathogens$perc_id[match(rownames(plant_paths_summed_df), its_pathogens$asv_id)]

plant_paths_summed_df_sum <- aggregate(plant_paths_summed_df, .~Pathogen_type+Species, FUN = "sum")
plant_paths_summed_df_mean <- aggregate(plant_paths_summed_df, .~Pathogen_type+Species, FUN = "mean")
plant_paths_summed_df_sd <- aggregate(plant_paths_summed_df, .~Pathogen_type+Species, FUN = "sd")
plant_paths_summed_df_sum$perc_id <- plant_paths_summed_df_mean$perc_id
plant_paths_summed_df_sum$perc_id_sd <- plant_paths_summed_df_sd$perc_id
plant_paths_summed_df_sum$Pathogen_type <- gsub("NA, ", "", plant_paths_summed_df_sum$Pathogen_type)
plant_paths_summed_df_sum$Pathogen_type <- gsub(", NA", "", plant_paths_summed_df_sum$Pathogen_type)
plant_paths_summed_df_sum$perc_id <- paste0(round(plant_paths_summed_df_sum$perc_id,2), "% +/- ", round(plant_paths_summed_df_sum$perc_id_sd,2))
plant_paths_summed_df_sum$perc_id <- gsub("% +/- NA", "%", plant_paths_summed_df_sum$perc_id)
plant_paths_summed_df_sum$Species <- gsub("s__", "", plant_paths_summed_df_sum$Species)
plant_paths_summed_df_sum$Species <- gsub("_", " ", plant_paths_summed_df_sum$Species)
plant_paths_summed_df_sum$Pathogen_type <- gsub("_", " ", plant_paths_summed_df_sum$Pathogen_type)
plant_paths_summed_df_sum$Species <- gsub(" sp$", " sp.", plant_paths_summed_df_sum$Species)


final <- plant_paths_summed_df_sum[,c(2, 1, 8, 7, 4, 6, 3, 5)]
colnames(final) <- c("Fungal Species", "Pathogen Type", "% Identity", "Urban Street Tree", "Urban Forest Edge", "Urban Forest Interior", "Rural Forest Edge", "Rural Forest Interior")

write.csv(final, "/projectnb/talbot-lab-data/Katies_data/Street_Trees/Analysis_dysbiosis/pathogen_tables/o_fungi.csv")
```

Table S3 - fungal M horizon pathogens
```{r pathogens}
pathogens_fungi_df <- m_rare_its

plant_pathogens_fungi_df <- subset(pathogens_fungi_df, !is.na(Plant_pathogenic_capacity_template))
human_pathogens_fungi_df <- subset(pathogens_fungi_df, Animal_biotrophic_capacity_template %in% c("animal_parasite/opportunistic_human_parasite",
                                                                                                  "invertebrate_parasite", 
                                                                                                  "opportunistic_human_parasite",
                                                                                                  "animal_parasite", "fish_parasite",
                                                                                                  "animal-associated/opportunistic_human_parasite",
                                                                                                  "opportunistic_animal_parasite", "arthropod_parasite",
                                                                                                  "opportunistic_animal_parasite_G._destructans"))
sum(plant_pathogens_fungi_df$Row.names %in% human_pathogens_fungi_df$Row.names)
plant_pathogens_fungi_df$pathogen_type <- paste0(plant_pathogens_fungi_df$Plant_pathogenic_capacity_template, ", ", plant_pathogens_fungi_df$Animal_biotrophic_capacity_template)
human_pathogens_fungi_df$pathogen_type <- paste0(human_pathogens_fungi_df$Plant_pathogenic_capacity_template, ", ", human_pathogens_fungi_df$Animal_biotrophic_capacity_template)
pathogens_fungi_df <- rbind(plant_pathogens_fungi_df, human_pathogens_fungi_df)
plant_pathogens_fungi_df <- distinct(pathogens_fungi_df)

plant_pathogens_fungi_df <- plant_pathogens_fungi_df[,c(5,11:227)]
plant_pathogens_fungi_df <- aggregate(plant_pathogens_fungi_df, .~Row.names+Species+pathogen_type, FUN = "sum")
plant_pathogens_fungi_df_clean <- plant_pathogens_fungi_df %>% filter(!is.na(Species))
plant_paths_df_w_rownames <- column_to_rownames(plant_pathogens_fungi_df_clean, var = "Row.names")
plant_paths_df_t <- as.data.frame(t(plant_paths_df_w_rownames))
plant_paths_df_t$tree_edge_type<-metadata_its$tree_edge_type[match(rownames(plant_paths_df_t), metadata_its$sample_name)]
plant_paths_df_t$tree_urban_type<-metadata_its$tree_urban_type[match(rownames(plant_paths_df_t), metadata_its$sample_name)]
plant_paths_df_t <- plant_paths_df_t %>% filter(!is.na(tree_edge_type))
plant_paths_df_t <- plant_paths_df_t %>% filter(!is.na(tree_urban_type))
plant_paths_df_t$new_tree_type<-paste(plant_paths_df_t$tree_edge_type,plant_paths_df_t$tree_urban_type)
plant_paths_df_t <- plant_paths_df_t[,-c(1420:1421)]
plant_paths_df_t[,1:1419] <- plant_paths_df_t[,1:1419] %>% mutate_if(is.character, as.numeric)
plant_paths_summed <- aggregate(plant_paths_df_t, .~new_tree_type, FUN = "sum")
plant_paths_sums <- colSums(plant_paths_summed[2:1420])
plant_paths_sums <- c(50, plant_paths_sums)
plant_paths_summed <- plant_paths_summed[,which(plant_paths_sums > 0)]
plant_paths_w_rownames <- column_to_rownames(plant_paths_summed, var = "new_tree_type")
plant_paths_summed_df <- as.data.frame(t(plant_paths_w_rownames))
plant_paths_summed_df$Pathogen_type<-plant_pathogens_fungi_df$pathogen_type[match(rownames(plant_paths_summed_df),plant_pathogens_fungi_df$Row.names)]
plant_paths_summed_df$Species<-plant_pathogens_fungi_df$Species[match(rownames(plant_paths_summed_df),plant_pathogens_fungi_df$Row.names)]
plant_paths_summed_df$perc_id<-its_pathogens$perc_id[match(rownames(plant_paths_summed_df), its_pathogens$asv_id)]

plant_paths_summed_df_sum <- aggregate(plant_paths_summed_df, .~Pathogen_type+Species, FUN = "sum")
plant_paths_summed_df_mean <- aggregate(plant_paths_summed_df, .~Pathogen_type+Species, FUN = "mean")
plant_paths_summed_df_sd <- aggregate(plant_paths_summed_df, .~Pathogen_type+Species, FUN = "sd")
plant_paths_summed_df_sum$perc_id <- plant_paths_summed_df_mean$perc_id
plant_paths_summed_df_sum$perc_id_sd <- plant_paths_summed_df_sd$perc_id
plant_paths_summed_df_sum$Pathogen_type <- gsub("NA, ", "", plant_paths_summed_df_sum$Pathogen_type)
plant_paths_summed_df_sum$Pathogen_type <- gsub(", NA", "", plant_paths_summed_df_sum$Pathogen_type)
plant_paths_summed_df_sum$perc_id <- paste0(round(plant_paths_summed_df_sum$perc_id,2), "% +/- ", round(plant_paths_summed_df_sum$perc_id_sd,2))
plant_paths_summed_df_sum$perc_id <- gsub("% +/- NA", "%", plant_paths_summed_df_sum$perc_id)
plant_paths_summed_df_sum$Species <- gsub("s__", "", plant_paths_summed_df_sum$Species)
plant_paths_summed_df_sum$Species <- gsub("_", " ", plant_paths_summed_df_sum$Species)
plant_paths_summed_df_sum$Pathogen_type <- gsub("_", " ", plant_paths_summed_df_sum$Pathogen_type)
plant_paths_summed_df_sum$Species <- gsub(" sp$", " sp.", plant_paths_summed_df_sum$Species)


final <- plant_paths_summed_df_sum[,c(2, 1, 8, 7, 4, 6, 3, 5)]
colnames(final) <- c("Fungal Species", "Pathogen Type", "% Identity", "Urban Street Tree", "Urban Forest Edge", "Urban Forest Interior", "Rural Forest Edge", "Rural Forest Interior")

write.csv(final, "/projectnb/talbot-lab-data/Katies_data/Street_Trees/Analysis_dysbiosis/pathogen_tables/m_fungi.csv")
```

Table S3 - fungal root pathogens
```{r pathogens}
pathogens_fungi_df <- root_rare_its

plant_pathogens_fungi_df <- subset(pathogens_fungi_df, !is.na(Plant_pathogenic_capacity_template))
human_pathogens_fungi_df <- subset(pathogens_fungi_df, Animal_biotrophic_capacity_template %in% c("animal_parasite/opportunistic_human_parasite",
                                                                                                  "invertebrate_parasite", 
                                                                                                  "opportunistic_human_parasite",
                                                                                                  "animal_parasite", "fish_parasite",
                                                                                                  "animal-associated/opportunistic_human_parasite",
                                                                                                  "opportunistic_animal_parasite", "arthropod_parasite",
                                                                                                  "opportunistic_animal_parasite_G._destructans"))
sum(plant_pathogens_fungi_df$Row.names %in% human_pathogens_fungi_df$Row.names)
plant_pathogens_fungi_df$pathogen_type <- paste0(plant_pathogens_fungi_df$Plant_pathogenic_capacity_template, ", ", plant_pathogens_fungi_df$Animal_biotrophic_capacity_template)
human_pathogens_fungi_df$pathogen_type <- paste0(human_pathogens_fungi_df$Plant_pathogenic_capacity_template, ", ", human_pathogens_fungi_df$Animal_biotrophic_capacity_template)
pathogens_fungi_df <- rbind(plant_pathogens_fungi_df, human_pathogens_fungi_df)
plant_pathogens_fungi_df <- distinct(pathogens_fungi_df)

plant_pathogens_fungi_df <- plant_pathogens_fungi_df[,c(5,11:100)]
plant_pathogens_fungi_df <- aggregate(plant_pathogens_fungi_df, .~Row.names+Species+pathogen_type, FUN = "sum")
plant_pathogens_fungi_df_clean <- plant_pathogens_fungi_df %>% filter(!is.na(Species))
plant_paths_df_w_rownames <- column_to_rownames(plant_pathogens_fungi_df_clean, var = "Row.names")
plant_paths_df_t <- as.data.frame(t(plant_paths_df_w_rownames))
plant_paths_df_t$tree_edge_type<-metadata_its$tree_edge_type[match(rownames(plant_paths_df_t), metadata_its$sample_name)]
plant_paths_df_t$tree_urban_type<-metadata_its$tree_urban_type[match(rownames(plant_paths_df_t), metadata_its$sample_name)]
plant_paths_df_t <- plant_paths_df_t %>% filter(!is.na(tree_edge_type))
plant_paths_df_t <- plant_paths_df_t %>% filter(!is.na(tree_urban_type))
plant_paths_df_t$new_tree_type<-paste(plant_paths_df_t$tree_edge_type,plant_paths_df_t$tree_urban_type)
plant_paths_df_t <- plant_paths_df_t[,-c(637:638)]
plant_paths_df_t[,1:636] <- plant_paths_df_t[,1:636] %>% mutate_if(is.character, as.numeric)
plant_paths_summed <- aggregate(plant_paths_df_t, .~new_tree_type, FUN = "sum")
plant_paths_sums <- colSums(plant_paths_summed[2:637])
plant_paths_sums <- c(50, plant_paths_sums)
plant_paths_summed <- plant_paths_summed[,which(plant_paths_sums > 0)]
plant_paths_w_rownames <- column_to_rownames(plant_paths_summed, var = "new_tree_type")
plant_paths_summed_df <- as.data.frame(t(plant_paths_w_rownames))
plant_paths_summed_df$Pathogen_type<-plant_pathogens_fungi_df$pathogen_type[match(rownames(plant_paths_summed_df),plant_pathogens_fungi_df$Row.names)]
plant_paths_summed_df$Species<-plant_pathogens_fungi_df$Species[match(rownames(plant_paths_summed_df),plant_pathogens_fungi_df$Row.names)]
plant_paths_summed_df$perc_id<-its_pathogens$perc_id[match(rownames(plant_paths_summed_df), its_pathogens$asv_id)]

plant_paths_summed_df_sum <- aggregate(plant_paths_summed_df, .~Pathogen_type+Species, FUN = "sum")
plant_paths_summed_df_mean <- aggregate(plant_paths_summed_df, .~Pathogen_type+Species, FUN = "mean")
plant_paths_summed_df_sd <- aggregate(plant_paths_summed_df, .~Pathogen_type+Species, FUN = "sd")
plant_paths_summed_df_sum$perc_id <- plant_paths_summed_df_mean$perc_id
plant_paths_summed_df_sum$perc_id_sd <- plant_paths_summed_df_sd$perc_id
plant_paths_summed_df_sum$Pathogen_type <- gsub("NA, ", "", plant_paths_summed_df_sum$Pathogen_type)
plant_paths_summed_df_sum$Pathogen_type <- gsub(", NA", "", plant_paths_summed_df_sum$Pathogen_type)
plant_paths_summed_df_sum$perc_id <- paste0(round(plant_paths_summed_df_sum$perc_id,2), "% +/- ", round(plant_paths_summed_df_sum$perc_id_sd,2))
plant_paths_summed_df_sum$perc_id <- gsub("% +/- NA", "%", plant_paths_summed_df_sum$perc_id)
plant_paths_summed_df_sum$Species <- gsub("s__", "", plant_paths_summed_df_sum$Species)
plant_paths_summed_df_sum$Species <- gsub("_", " ", plant_paths_summed_df_sum$Species)
plant_paths_summed_df_sum$Pathogen_type <- gsub("_", " ", plant_paths_summed_df_sum$Pathogen_type)
plant_paths_summed_df_sum$Species <- gsub(" sp$", " sp.", plant_paths_summed_df_sum$Species)


final <- plant_paths_summed_df_sum[,c(2, 1, 8, 7, 4, 6, 3, 5)]
colnames(final) <- c("Fungal Species", "Pathogen Type", "% Identity", "Urban Street Tree", "Urban Forest Edge", "Urban Forest Interior", "Rural Forest Edge", "Rural Forest Interior")

write.csv(final, "/projectnb/talbot-lab-data/Katies_data/Street_Trees/Analysis_dysbiosis/pathogen_tables/root_fungi.csv")
```

Table S3 - fungal leaf pathogens
```{r pathogens}
pathogens_fungi_df <- leaf_rare_its

plant_pathogens_fungi_df <- subset(pathogens_fungi_df, !is.na(Plant_pathogenic_capacity_template))
human_pathogens_fungi_df <- subset(pathogens_fungi_df, Animal_biotrophic_capacity_template %in% c("animal_parasite/opportunistic_human_parasite",
                                                                                                  "invertebrate_parasite", 
                                                                                                  "opportunistic_human_parasite",
                                                                                                  "animal_parasite", "fish_parasite",
                                                                                                  "animal-associated/opportunistic_human_parasite",
                                                                                                  "opportunistic_animal_parasite", "arthropod_parasite",
                                                                                                  "opportunistic_animal_parasite_G._destructans"))
sum(plant_pathogens_fungi_df$Row.names %in% human_pathogens_fungi_df$Row.names)
plant_pathogens_fungi_df$pathogen_type <- paste0(plant_pathogens_fungi_df$Plant_pathogenic_capacity_template, ", ", plant_pathogens_fungi_df$Animal_biotrophic_capacity_template)
human_pathogens_fungi_df$pathogen_type <- paste0(human_pathogens_fungi_df$Plant_pathogenic_capacity_template, ", ", human_pathogens_fungi_df$Animal_biotrophic_capacity_template)
pathogens_fungi_df <- rbind(plant_pathogens_fungi_df, human_pathogens_fungi_df)
plant_pathogens_fungi_df <- distinct(pathogens_fungi_df)

plant_pathogens_fungi_df <- plant_pathogens_fungi_df[,c(5,11:253)]
plant_pathogens_fungi_df <- aggregate(plant_pathogens_fungi_df, .~Row.names+Species+pathogen_type, FUN = "sum")
plant_pathogens_fungi_df_clean <- plant_pathogens_fungi_df %>% filter(!is.na(Species))
plant_paths_df_w_rownames <- column_to_rownames(plant_pathogens_fungi_df_clean, var = "Row.names")
plant_paths_df_t <- as.data.frame(t(plant_paths_df_w_rownames))
plant_paths_df_t$tree_edge_type<-metadata_its$tree_edge_type[match(rownames(plant_paths_df_t), metadata_its$sample_name)]
plant_paths_df_t$tree_urban_type<-metadata_its$tree_urban_type[match(rownames(plant_paths_df_t), metadata_its$sample_name)]
plant_paths_df_t <- plant_paths_df_t %>% filter(!is.na(tree_edge_type))
plant_paths_df_t <- plant_paths_df_t %>% filter(!is.na(tree_urban_type))
plant_paths_df_t$new_tree_type<-paste(plant_paths_df_t$tree_edge_type,plant_paths_df_t$tree_urban_type)
plant_paths_df_t <- plant_paths_df_t[,-c(1490:1491)]
plant_paths_df_t[,1:1489] <- plant_paths_df_t[,1:1489] %>% mutate_if(is.character, as.numeric)
plant_paths_summed <- aggregate(plant_paths_df_t, .~new_tree_type, FUN = "sum")
plant_paths_sums <- colSums(plant_paths_summed[2:1490])
plant_paths_sums <- c(50, plant_paths_sums)
plant_paths_summed <- plant_paths_summed[,which(plant_paths_sums > 0)]
plant_paths_w_rownames <- column_to_rownames(plant_paths_summed, var = "new_tree_type")
plant_paths_summed_df <- as.data.frame(t(plant_paths_w_rownames))
plant_paths_summed_df$Pathogen_type<-plant_pathogens_fungi_df$pathogen_type[match(rownames(plant_paths_summed_df),plant_pathogens_fungi_df$Row.names)]
plant_paths_summed_df$Species<-plant_pathogens_fungi_df$Species[match(rownames(plant_paths_summed_df),plant_pathogens_fungi_df$Row.names)]
plant_paths_summed_df$perc_id<-its_pathogens$perc_id[match(rownames(plant_paths_summed_df), its_pathogens$asv_id)]

plant_paths_summed_df_sum <- aggregate(plant_paths_summed_df, .~Pathogen_type+Species, FUN = "sum")
plant_paths_summed_df_mean <- aggregate(plant_paths_summed_df, .~Pathogen_type+Species, FUN = "mean")
plant_paths_summed_df_sd <- aggregate(plant_paths_summed_df, .~Pathogen_type+Species, FUN = "sd")
plant_paths_summed_df_sum$perc_id <- plant_paths_summed_df_mean$perc_id
plant_paths_summed_df_sum$perc_id_sd <- plant_paths_summed_df_sd$perc_id
plant_paths_summed_df_sum$Pathogen_type <- gsub("NA, ", "", plant_paths_summed_df_sum$Pathogen_type)
plant_paths_summed_df_sum$Pathogen_type <- gsub(", NA", "", plant_paths_summed_df_sum$Pathogen_type)
plant_paths_summed_df_sum$perc_id <- paste0(round(plant_paths_summed_df_sum$perc_id,2), "% +/- ", round(plant_paths_summed_df_sum$perc_id_sd,2))
plant_paths_summed_df_sum$perc_id <- gsub("% +/- NA", "%", plant_paths_summed_df_sum$perc_id)
plant_paths_summed_df_sum$Species <- gsub("s__", "", plant_paths_summed_df_sum$Species)
plant_paths_summed_df_sum$Species <- gsub("_", " ", plant_paths_summed_df_sum$Species)
plant_paths_summed_df_sum$Pathogen_type <- gsub("_", " ", plant_paths_summed_df_sum$Pathogen_type)
plant_paths_summed_df_sum$Species <- gsub(" sp$", " sp.", plant_paths_summed_df_sum$Species)


final <- plant_paths_summed_df_sum[,c(2, 1, 8, 7, 4, 6, 3, 5)]
colnames(final) <- c("Fungal Species", "Pathogen Type", "% Identity", "Urban Street Tree", "Urban Forest Edge", "Urban Forest Interior", "Rural Forest Edge", "Rural Forest Interior")

write.csv(final, "/projectnb/talbot-lab-data/Katies_data/Street_Trees/Analysis_dysbiosis/pathogen_tables/leaf_fungi.csv")
```


Table S3 - bacterial O horizon pathogens
```{r pathogens}
pathogens_bac_df <- o_rare_16s
pathogens_bac_df <- merge(pathogens_bac_df, tax_w_paths, by.x = "Row.names", by.y = "asv_id", all.x = TRUE)
pathogens_bac_df <- merge(pathogens_bac_df, pathogens, by.x = "Row.names", by.y = "asv_id", all.x = TRUE)

plant_pathogens_bac_df <- subset(pathogens_bac_df, !is.na(pathogens_bac_df$pathogen_type))

plant_pathogens_bac_df <- plant_pathogens_bac_df[,c(1,8:250,252:253)]
plant_pathogens_bac_df <- distinct(plant_pathogens_bac_df)
# plant_pathogens_bac_df <- aggregate(plant_pathogens_bac_df, .~Row.names+Species+pathogen_type+perc_id, FUN = "sum")
plant_pathogens_bac_df_clean <- plant_pathogens_bac_df %>% filter(!grepl("uncultured", plant_pathogens_bac_df$Species))
plant_pathogens_bac_df_clean <- plant_pathogens_bac_df_clean %>% filter(!grepl("unidentified", plant_pathogens_bac_df_clean$Species))
plant_pathogens_bac_df_clean <- plant_pathogens_bac_df_clean %>% filter(!grepl("metagenome", plant_pathogens_bac_df_clean$Species))
plant_pathogens_bac_df_clean <- plant_pathogens_bac_df_clean %>% filter(!grepl("__bacterium", plant_pathogens_bac_df_clean$Species))
plant_pathogens_bac_df_clean <- plant_pathogens_bac_df_clean %>% filter(!grepl("low_GC", plant_pathogens_bac_df_clean$Species))
plant_pathogens_bac_df_clean <- plant_pathogens_bac_df_clean %>% filter(!grepl("endosymbiont_of", plant_pathogens_bac_df_clean$Species))
plant_pathogens_bac_df_clean <- plant_pathogens_bac_df_clean %>% filter(!grepl("rock_porewater", plant_pathogens_bac_df_clean$Species))
plant_pathogens_bac_df_clean <- plant_pathogens_bac_df_clean %>% filter(!is.na(Species))
plant_paths_df_w_rownames <- column_to_rownames(plant_pathogens_bac_df_clean, var = "Row.names")
plant_paths_df_t <- as.data.frame(t(plant_paths_df_w_rownames))
plant_paths_df_t$tree_edge_type<-metadata_16s_200$tree_edge_type[match(rownames(plant_paths_df_t), metadata_16s_200$sample_name)]
plant_paths_df_t$tree_urban_type<-metadata_16s_200$tree_urban_type[match(rownames(plant_paths_df_t), metadata_16s_200$sample_name)]
plant_paths_df_t <- plant_paths_df_t %>% filter(!is.na(tree_edge_type))
plant_paths_df_t <- plant_paths_df_t %>% filter(!is.na(tree_urban_type))
plant_paths_df_t$new_tree_type<-paste(plant_paths_df_t$tree_edge_type,plant_paths_df_t$tree_urban_type)
plant_paths_df_t <- plant_paths_df_t[,-c(316:317)]
plant_paths_df_t[,1:315] <- plant_paths_df_t[,1:315] %>% mutate_if(is.character, as.numeric)
plant_paths_summed <- aggregate(plant_paths_df_t, .~new_tree_type, FUN = "sum")
plant_paths_sums <- colSums(plant_paths_summed[2:316])
plant_paths_sums <- c(50, plant_paths_sums)
plant_paths_summed <- plant_paths_summed[,which(plant_paths_sums > 0)]
plant_paths_w_rownames <- column_to_rownames(plant_paths_summed, var = "new_tree_type")
plant_paths_summed_df <- as.data.frame(t(plant_paths_w_rownames))
plant_paths_summed_df$Pathogen_type<-plant_pathogens_bac_df$pathogen_type[match(rownames(plant_paths_summed_df),plant_pathogens_bac_df$Row.names)]
plant_paths_summed_df$Species<-plant_pathogens_bac_df$Species[match(rownames(plant_paths_summed_df),plant_pathogens_bac_df$Row.names)]
plant_paths_summed_df$perc_id<-plant_pathogens_bac_df$perc_id[match(rownames(plant_paths_summed_df),plant_pathogens_bac_df$Row.names)]
plant_paths_summed_df_sum <- aggregate(plant_paths_summed_df, .~Pathogen_type+Species, FUN = "sum")
plant_paths_summed_df_mean <- aggregate(plant_paths_summed_df, .~Pathogen_type+Species, FUN = "mean")
plant_paths_summed_df_sd <- aggregate(plant_paths_summed_df, .~Pathogen_type+Species, FUN = "sd")
plant_paths_summed_df_sum$perc_id <- plant_paths_summed_df_mean$perc_id
plant_paths_summed_df_sum$perc_id_sd <- plant_paths_summed_df_sd$perc_id

write.csv(plant_paths_summed_df_sum, "/projectnb/talbot-lab-data/Katies_data/Street_Trees/Analysis_dysbiosis/pathogen_tables/o_bac.csv")
```

Table S3 - bacterial M horizon pathogens
```{r pathogens}
pathogens_bac_df <- m_rare_16s
pathogens_bac_df <- merge(pathogens_bac_df, tax_w_paths, by.x = "Row.names", by.y = "asv_id", all.x = TRUE)
pathogens_bac_df <- merge(pathogens_bac_df, pathogens, by.x = "Row.names", by.y = "asv_id", all.x = TRUE)

plant_pathogens_bac_df <- subset(pathogens_bac_df, !is.na(pathogens_bac_df$pathogen_type))

plant_pathogens_bac_df <- plant_pathogens_bac_df[,c(1,8:223,225:226)]
plant_pathogens_bac_df <- distinct(plant_pathogens_bac_df)
# plant_pathogens_bac_df <- aggregate(plant_pathogens_bac_df, .~Row.names+Species+pathogen_type+perc_id, FUN = "sum")
plant_pathogens_bac_df_clean <- plant_pathogens_bac_df %>% filter(!grepl("uncultured", plant_pathogens_bac_df$Species))
plant_pathogens_bac_df_clean <- plant_pathogens_bac_df_clean %>% filter(!grepl("unidentified", plant_pathogens_bac_df_clean$Species))
plant_pathogens_bac_df_clean <- plant_pathogens_bac_df_clean %>% filter(!grepl("metagenome", plant_pathogens_bac_df_clean$Species))
plant_pathogens_bac_df_clean <- plant_pathogens_bac_df_clean %>% filter(!grepl("__bacterium", plant_pathogens_bac_df_clean$Species))
plant_pathogens_bac_df_clean <- plant_pathogens_bac_df_clean %>% filter(!grepl("low_GC", plant_pathogens_bac_df_clean$Species))
plant_pathogens_bac_df_clean <- plant_pathogens_bac_df_clean %>% filter(!grepl("endosymbiont_of", plant_pathogens_bac_df_clean$Species))
plant_pathogens_bac_df_clean <- plant_pathogens_bac_df_clean %>% filter(!grepl("rock_porewater", plant_pathogens_bac_df_clean$Species))
plant_pathogens_bac_df_clean <- plant_pathogens_bac_df_clean %>% filter(!is.na(Species))
plant_paths_df_w_rownames <- column_to_rownames(plant_pathogens_bac_df_clean, var = "Row.names")
plant_paths_df_t <- as.data.frame(t(plant_paths_df_w_rownames))
plant_paths_df_t$tree_edge_type<-metadata_16s_200$tree_edge_type[match(rownames(plant_paths_df_t), metadata_16s_200$sample_name)]
plant_paths_df_t$tree_urban_type<-metadata_16s_200$tree_urban_type[match(rownames(plant_paths_df_t), metadata_16s_200$sample_name)]
plant_paths_df_t <- plant_paths_df_t %>% filter(!is.na(tree_edge_type))
plant_paths_df_t <- plant_paths_df_t %>% filter(!is.na(tree_urban_type))
plant_paths_df_t$new_tree_type<-paste(plant_paths_df_t$tree_edge_type,plant_paths_df_t$tree_urban_type)
plant_paths_df_t <- plant_paths_df_t[,-c(550:551)]
plant_paths_df_t[,1:549] <- plant_paths_df_t[,1:549] %>% mutate_if(is.character, as.numeric)
plant_paths_summed <- aggregate(plant_paths_df_t, .~new_tree_type, FUN = "sum")
plant_paths_sums <- colSums(plant_paths_summed[2:550])
plant_paths_sums <- c(50, plant_paths_sums)
plant_paths_summed <- plant_paths_summed[,which(plant_paths_sums > 0)]
plant_paths_w_rownames <- column_to_rownames(plant_paths_summed, var = "new_tree_type")
plant_paths_summed_df <- as.data.frame(t(plant_paths_w_rownames))
plant_paths_summed_df$Pathogen_type<-plant_pathogens_bac_df$pathogen_type[match(rownames(plant_paths_summed_df),plant_pathogens_bac_df$Row.names)]
plant_paths_summed_df$Species<-plant_pathogens_bac_df$Species[match(rownames(plant_paths_summed_df),plant_pathogens_bac_df$Row.names)]
plant_paths_summed_df$perc_id<-plant_pathogens_bac_df$perc_id[match(rownames(plant_paths_summed_df),plant_pathogens_bac_df$Row.names)]
plant_paths_summed_df_sum <- aggregate(plant_paths_summed_df, .~Pathogen_type+Species, FUN = "sum")
plant_paths_summed_df_mean <- aggregate(plant_paths_summed_df, .~Pathogen_type+Species, FUN = "mean")
plant_paths_summed_df_sd <- aggregate(plant_paths_summed_df, .~Pathogen_type+Species, FUN = "sd")
plant_paths_summed_df_sum$perc_id <- plant_paths_summed_df_mean$perc_id
plant_paths_summed_df_sum$perc_id_sd <- plant_paths_summed_df_sd$perc_id

write.csv(plant_paths_summed_df_sum, "/projectnb/talbot-lab-data/Katies_data/Street_Trees/Analysis_dysbiosis/pathogen_tables/m_bac.csv")
```

Table S3 - bacterial root pathogens
```{r pathogens}
pathogens_bac_df <- root_rare_16s
pathogens_bac_df <- merge(pathogens_bac_df, tax_w_paths, by.x = "Row.names", by.y = "asv_id", all.x = TRUE)
pathogens_bac_df <- merge(pathogens_bac_df, pathogens, by.x = "Row.names", by.y = "asv_id", all.x = TRUE)

plant_pathogens_bac_df <- subset(pathogens_bac_df, !is.na(pathogens_bac_df$pathogen_type))

plant_pathogens_bac_df <- plant_pathogens_bac_df[,c(1,8:86,88:89)]
plant_pathogens_bac_df <- distinct(plant_pathogens_bac_df)
# plant_pathogens_bac_df <- aggregate(plant_pathogens_bac_df, .~Row.names+Species+pathogen_type+perc_id, FUN = "sum")
plant_pathogens_bac_df_clean <- plant_pathogens_bac_df %>% filter(!grepl("uncultured", plant_pathogens_bac_df$Species))
plant_pathogens_bac_df_clean <- plant_pathogens_bac_df_clean %>% filter(!grepl("unidentified", plant_pathogens_bac_df_clean$Species))
plant_pathogens_bac_df_clean <- plant_pathogens_bac_df_clean %>% filter(!grepl("metagenome", plant_pathogens_bac_df_clean$Species))
plant_pathogens_bac_df_clean <- plant_pathogens_bac_df_clean %>% filter(!grepl("__bacterium", plant_pathogens_bac_df_clean$Species))
plant_pathogens_bac_df_clean <- plant_pathogens_bac_df_clean %>% filter(!grepl("low_GC", plant_pathogens_bac_df_clean$Species))
plant_pathogens_bac_df_clean <- plant_pathogens_bac_df_clean %>% filter(!grepl("endosymbiont_of", plant_pathogens_bac_df_clean$Species))
plant_pathogens_bac_df_clean <- plant_pathogens_bac_df_clean %>% filter(!grepl("rock_porewater", plant_pathogens_bac_df_clean$Species))
plant_pathogens_bac_df_clean <- plant_pathogens_bac_df_clean %>% filter(!is.na(Species))
plant_paths_df_w_rownames <- column_to_rownames(plant_pathogens_bac_df_clean, var = "Row.names")
plant_paths_df_t <- as.data.frame(t(plant_paths_df_w_rownames))
plant_paths_df_t$tree_edge_type<-metadata_16s_200$tree_edge_type[match(rownames(plant_paths_df_t), metadata_16s_200$sample_name)]
plant_paths_df_t$tree_urban_type<-metadata_16s_200$tree_urban_type[match(rownames(plant_paths_df_t), metadata_16s_200$sample_name)]
plant_paths_df_t <- plant_paths_df_t %>% filter(!is.na(tree_edge_type))
plant_paths_df_t <- plant_paths_df_t %>% filter(!is.na(tree_urban_type))
plant_paths_df_t$new_tree_type<-paste(plant_paths_df_t$tree_edge_type,plant_paths_df_t$tree_urban_type)
plant_paths_df_t <- plant_paths_df_t[,-c(122:123)]
plant_paths_df_t[,1:121] <- plant_paths_df_t[,1:121] %>% mutate_if(is.character, as.numeric)
plant_paths_summed <- aggregate(plant_paths_df_t, .~new_tree_type, FUN = "sum")
plant_paths_sums <- colSums(plant_paths_summed[2:122])
plant_paths_sums <- c(50, plant_paths_sums)
plant_paths_summed <- plant_paths_summed[,which(plant_paths_sums > 0)]
plant_paths_w_rownames <- column_to_rownames(plant_paths_summed, var = "new_tree_type")
plant_paths_summed_df <- as.data.frame(t(plant_paths_w_rownames))
plant_paths_summed_df$Pathogen_type<-plant_pathogens_bac_df$pathogen_type[match(rownames(plant_paths_summed_df),plant_pathogens_bac_df$Row.names)]
plant_paths_summed_df$Species<-plant_pathogens_bac_df$Species[match(rownames(plant_paths_summed_df),plant_pathogens_bac_df$Row.names)]
plant_paths_summed_df$perc_id<-plant_pathogens_bac_df$perc_id[match(rownames(plant_paths_summed_df),plant_pathogens_bac_df$Row.names)]
plant_paths_summed_df_sum <- aggregate(plant_paths_summed_df, .~Pathogen_type+Species, FUN = "sum")
plant_paths_summed_df_mean <- aggregate(plant_paths_summed_df, .~Pathogen_type+Species, FUN = "mean")
plant_paths_summed_df_sd <- aggregate(plant_paths_summed_df, .~Pathogen_type+Species, FUN = "sd")
plant_paths_summed_df_sum$perc_id <- plant_paths_summed_df_mean$perc_id
plant_paths_summed_df_sum$perc_id_sd <- plant_paths_summed_df_sd$perc_id

write.csv(plant_paths_summed_df_sum, "/projectnb/talbot-lab-data/Katies_data/Street_Trees/Analysis_dysbiosis/pathogen_tables/root_bac.csv")
```

Table S3 - bacterial leaf pathogens
```{r pathogens}
pathogens_bac_df <- leaf_rare_16s_200
pathogens_bac_df <- merge(pathogens_bac_df, tax_w_paths, by.x = "Row.names", by.y = "asv_id", all.x = TRUE)
pathogens_bac_df <- merge(pathogens_bac_df, pathogens, by.x = "Row.names", by.y = "asv_id", all.x = TRUE)

plant_pathogens_bac_df <- subset(pathogens_bac_df, !is.na(pathogens_bac_df$pathogen_type))

plant_pathogens_bac_df <- plant_pathogens_bac_df[,c(1,8:151,152:153)]
plant_pathogens_bac_df <- distinct(plant_pathogens_bac_df)
# plant_pathogens_bac_df <- aggregate(plant_pathogens_bac_df, .~Row.names+Species+pathogen_type+perc_id, FUN = "sum")
plant_pathogens_bac_df_clean <- plant_pathogens_bac_df %>% filter(!grepl("uncultured", plant_pathogens_bac_df$Species))
plant_pathogens_bac_df_clean <- plant_pathogens_bac_df_clean %>% filter(!grepl("unidentified", plant_pathogens_bac_df_clean$Species))
plant_pathogens_bac_df_clean <- plant_pathogens_bac_df_clean %>% filter(!grepl("metagenome", plant_pathogens_bac_df_clean$Species))
plant_pathogens_bac_df_clean <- plant_pathogens_bac_df_clean %>% filter(!grepl("__bacterium", plant_pathogens_bac_df_clean$Species))
plant_pathogens_bac_df_clean <- plant_pathogens_bac_df_clean %>% filter(!grepl("low_GC", plant_pathogens_bac_df_clean$Species))
plant_pathogens_bac_df_clean <- plant_pathogens_bac_df_clean %>% filter(!grepl("endosymbiont_of", plant_pathogens_bac_df_clean$Species))
plant_pathogens_bac_df_clean <- plant_pathogens_bac_df_clean %>% filter(!grepl("rock_porewater", plant_pathogens_bac_df_clean$Species))
plant_pathogens_bac_df_clean <- plant_pathogens_bac_df_clean %>% filter(!is.na(Species))
plant_paths_df_w_rownames <- column_to_rownames(plant_pathogens_bac_df_clean, var = "Row.names")
plant_paths_df_t <- as.data.frame(t(plant_paths_df_w_rownames))
plant_paths_df_t$tree_edge_type<-metadata_16s_200$tree_edge_type[match(rownames(plant_paths_df_t), metadata_16s_200$sample_name)]
plant_paths_df_t$tree_urban_type<-metadata_16s_200$tree_urban_type[match(rownames(plant_paths_df_t), metadata_16s_200$sample_name)]
plant_paths_df_t <- plant_paths_df_t %>% filter(!is.na(tree_edge_type))
plant_paths_df_t <- plant_paths_df_t %>% filter(!is.na(tree_urban_type))
plant_paths_df_t$new_tree_type<-paste(plant_paths_df_t$tree_edge_type,plant_paths_df_t$tree_urban_type)
plant_paths_df_t <- plant_paths_df_t[,-c(71:72)]
plant_paths_df_t[,1:70] <- plant_paths_df_t[,1:70] %>% mutate_if(is.character, as.numeric)
plant_paths_summed <- aggregate(plant_paths_df_t, .~new_tree_type, FUN = "sum")
plant_paths_sums <- colSums(plant_paths_summed[2:71])
plant_paths_sums <- c(50, plant_paths_sums)
plant_paths_summed <- plant_paths_summed[,which(plant_paths_sums > 0)]
plant_paths_w_rownames <- column_to_rownames(plant_paths_summed, var = "new_tree_type")
plant_paths_summed_df <- as.data.frame(t(plant_paths_w_rownames))
plant_paths_summed_df$Pathogen_type<-plant_pathogens_bac_df$pathogen_type[match(rownames(plant_paths_summed_df),plant_pathogens_bac_df$Row.names)]
plant_paths_summed_df$Species<-plant_pathogens_bac_df$Species[match(rownames(plant_paths_summed_df),plant_pathogens_bac_df$Row.names)]
plant_paths_summed_df$perc_id<-plant_pathogens_bac_df$perc_id[match(rownames(plant_paths_summed_df),plant_pathogens_bac_df$Row.names)]
plant_paths_summed_df_sum <- aggregate(plant_paths_summed_df, .~Pathogen_type+Species, FUN = "sum")
plant_paths_summed_df_mean <- aggregate(plant_paths_summed_df, .~Pathogen_type+Species, FUN = "mean")
plant_paths_summed_df_sd <- aggregate(plant_paths_summed_df, .~Pathogen_type+Species, FUN = "sd")
plant_paths_summed_df_sum$perc_id <- plant_paths_summed_df_mean$perc_id
plant_paths_summed_df_sum$perc_id_sd <- plant_paths_summed_df_sd$perc_id

write.csv(plant_paths_summed_df_sum, "/projectnb/talbot-lab-data/Katies_data/Street_Trees/Analysis_dysbiosis/pathogen_tables/leaf_bac.csv")
```

Figure X
```{r fungi endophytes}
metadata_leaf_its <- metadata_its[which(metadata_its$sample_type %in% c("Leaf")),]
metadata_leaf_its$tree_distance_from_edge <- as.numeric(metadata_leaf_its$tree_distance_from_edge)

fungal_trait_soil <- metadata_soil_its[, colnames(metadata_soil_its) %in% 
                                    fungal_trait_names]
fungal_trait_leaf <- metadata_leaf_its[, colnames(metadata_leaf_its) %in% 
                                    fungal_trait_names]
standard_error <- function(x) sd(x) / sqrt(length(x)) 

mean_leaf <- aggregate(fungal_trait_leaf,
                  by=list(metadata_leaf_its$tree_urban_type,
                          metadata_leaf_its$tree_edge_type), FUN= "mean")
se_leaf <- aggregate(fungal_trait_leaf,
                by=list(metadata_leaf_its$tree_urban_type,
                        metadata_leaf_its$tree_edge_type), FUN= standard_error)

vdata <- cbind(mean_leaf[,1:2], mean_leaf$perc_foliar_endophyte, se_leaf$perc_foliar_endophyte)

colnames(vdata) <- c("Site", "Edge", "endophytes", "endophytes_se")
vdata$Edge <- factor(vdata$Edge, 
                     levels = c("Urban Street Tree", "Forest Edge", 
                                "Forest Interior"))

vdata_long <- as.data.frame(matrix(data = NA, nrow = 10, ncol = 5))
colnames(vdata_long) <- c("Site", "Edge", "Guild", "Abundance", "SE")
vdata_long$Site <- rep(vdata$Site, 1)
vdata_long$Edge <- rep(vdata$Edge, 1)
vdata_long$Guild <- c(rep("Foliar Endophytes", 5))
vdata_long$Abundance <- c(vdata$endophytes)
vdata_long$SE <- c(vdata$endophytes_se)

vdata_long$site_edge <- paste0(vdata_long$Site, " ", vdata_long$Edge)
vdata_long$site_edge <- gsub("Urban Street Tree Urban Street Tree", 
                             "Urban Street Tree",
                             vdata_long$site_edge)
vdata_long$site_edge <- factor(vdata_long$site_edge, 
                               levels = c("Urban Street Tree", 
                                          "Urban Forest Edge",
                                          "Urban Forest Interior",
                                          "Rural Forest Edge", 
                                          "Rural Forest Interior"))

metadata_leaf_its$site_edge <- paste0(metadata_leaf_its$tree_urban_type, " ", metadata_leaf_its$tree_edge_type)
metadata_leaf_its$site_edge[which(metadata_leaf_its$site_edge == "Urban Street Tree Urban Street Tree")] <- "Urban Street Tree"

fungi_ecm_v_frag_bg_v4_woutliers <- ggplot() +
   geom_line(data = vdata_long, 
             aes(x=site_edge, y=Abundance, shape = Guild, group = Guild), 
            linetype = 2) +
  geom_point(data = vdata_long, 
             aes(x=site_edge, y=Abundance, color=site_edge, shape = Guild, 
                 group = Guild), size=5) + 
  geom_errorbar(data = vdata_long, 
                aes(x=site_edge, ymin=Abundance-SE, ymax=Abundance+SE, 
                    color=site_edge), width = 0.2) +
  geom_point(data = metadata_leaf_its,
             aes(x = site_edge, y = perc_foliar_endophyte, color = site_edge),
             position = position_jitter(width = 0.2), size = 2, alpha = 0.2, stroke = NA) +
  theme_classic() + labs (y="log(Foliar Endophyte Abundance (%)", x="") +
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        axis.text.x = element_text(size = 10, angle = 30, 
                                   hjust = 0.5, vjust = 0.5), 
        axis.text.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10), legend.position = "none") +
  scale_color_manual(values = pal, guide = "none") + 
  scale_fill_manual(values = pal, guide = "none") + 
  ggtitle("")

fungi_ecm_v_frag_bg_v4_woutliers

model1 <- lme(formula(perc_foliar_endophyte ~ site_edge), 
              random = ~1|plot_name/tree_id, na.action = na.exclude, 
              method = "ML", data = metadata_leaf_its)
summary(model1)
summary(emmeans(model1, pairwise ~ site_edge))
```

```{r leaf indicator species its}
leaf_indic_its$Family[which(is.na(leaf_indic_its$Family))] <- "Other Fungi"
# leaf_indic_its$Family <- gsub(".*_fam_Incertae_sedis", "Other Fungi", leaf_indic_its$Family)
# leaf_indic_its$Class <- gsub(".*_cls_Incertae_sedis", "Other Fungi", leaf_indic_its$Family)
# leaf_indic_its$Order <- gsub(".*_ord_Incertae_sedis", "Other Fungi", leaf_indic_its$Family)
leaf_indic_its$Family <- gsub("f__", "", leaf_indic_its$Family)

order_of_families <-leaf_indic_its[,which(colnames(leaf_indic_its) %in% c("Phylum", "Class", "Order", "Family"))]

order_of_families <- order_of_families[order(order_of_families$Phylum, order_of_families$Class, order_of_families$Order, order_of_families$Family),]
order_of_families <- distinct(order_of_families)
# order_of_families <- order_of_families[-which(order_of_families$Family == "Other Fungi"),]

leaf_indic_its$Family <- factor(leaf_indic_its$Family, levels = c(order_of_families$Family, "Other Fungi"))

leaf_indic_its$X <- 1 
leaf_indic_its <- leaf_indic_its[, colnames(leaf_indic_its) %in% c("X", "Class", "Family", "Group")]
leaf_indic_its <- aggregate(leaf_indic_its, X ~ Class + Family + Group, FUN = "sum")

leaf_indic_its$Group <- gsub("Rural.edge", "Rural Forest Edge", leaf_indic_its$Group)
leaf_indic_its$Group <- gsub("Rural.interior", "Rural Forest Interior", leaf_indic_its$Group)
leaf_indic_its$Group <- gsub("Street.tree", "Urban Street Tree", leaf_indic_its$Group)
leaf_indic_its$Group <- gsub("Urban.edge", "Urban Forest Edge", leaf_indic_its$Group)
leaf_indic_its$Group <- gsub("Urban.interior", "Urban Forest Interior", leaf_indic_its$Group)

leaf_indic_its$Group <- factor(leaf_indic_its$Group, levels = c("Urban Street Tree", "Urban Forest Edge", "Urban Forest Interior", "Rural Forest Edge", "Rural Forest Interior"))
# group_pal <- group_pal_fungi[which(names(group_pal_fungi) %in% leaf_indic_its$primary_lifestyle)]

l_indic <- ggplot(leaf_indic_its, aes(x = Group, y = X, fill = Family)) + 
  geom_bar(stat = 'identity', position = 'fill') + 
  #scale_fill_manual(values = group_pal) + 
  labs(x = "Tree Type", y = "Proportion of Indicator Species", fill = "Fungal Guild") + 
  theme_classic() + theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        axis.text.x = element_text(size = 10, angle = 30, 
                                   hjust = 0.5, vjust = 0.5), 
        axis.text.y = element_text(size = 10), 
        legend.text = element_text(size = 8), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10), legend.position = "none")
l_indic
```

```{r root indicator species}
bg_indic_its$Family[which(is.na(bg_indic_its$Family))] <- "Other Fungi"
# bg_indic_its$Family <- gsub(".*_fam_Incertae_sedis", "Other Fungi", bg_indic_its$Family)
# bg_indic_its$Class <- gsub(".*_cls_Incertae_sedis", "Other Fungi", bg_indic_its$Family)
# bg_indic_its$Order <- gsub(".*_ord_Incertae_sedis", "Other Fungi", bg_indic_its$Family)
bg_indic_its$Family <- gsub("f__", "", bg_indic_its$Family)

order_of_families <-bg_indic_its[,which(colnames(bg_indic_its) %in% c("Phylum", "Class", "Order", "Family"))]

order_of_families <- order_of_families[order(order_of_families$Phylum, order_of_families$Class, order_of_families$Order, order_of_families$Family),]
order_of_families <- distinct(order_of_families)
# order_of_families <- order_of_families[-which(order_of_families$Family == "Other Fungi"),]

bg_indic_its$Family <- factor(bg_indic_its$Family, levels = c(order_of_families$Family, "Other Fungi"))

bg_indic_its$X <- 1 
bg_indic_its <- bg_indic_its[, colnames(bg_indic_its) %in% c("X", "Class", "Family", "Group")]
bg_indic_its <- aggregate(bg_indic_its, X ~ Class + Family + Group, FUN = "sum")

bg_indic_its$Group <- gsub("Rural.edge", "Rural Forest Edge", bg_indic_its$Group)
bg_indic_its$Group <- gsub("Rural.interior", "Rural Forest Interior", bg_indic_its$Group)
bg_indic_its$Group <- gsub("Street.tree", "Urban Street Tree", bg_indic_its$Group)
bg_indic_its$Group <- gsub("Urban.edge", "Urban Forest Edge", bg_indic_its$Group)
bg_indic_its$Group <- gsub("Urban.interior", "Urban Forest Interior", bg_indic_its$Group)

bg_indic_its$Group <- factor(bg_indic_its$Group, levels = c("Urban Street Tree", "Urban Forest Edge", "Urban Forest Interior", "Rural Forest Edge", "Rural Forest Interior"))
# group_pal <- group_pal_fungi[which(names(group_pal_fungi) %in% bg_indic_its$primary_lifestyle)]

bg_indic <- ggplot(bg_indic_its, aes(x = Group, y = X, fill = Family)) + 
  geom_bar(stat = 'identity', position = 'fill') + 
  #scale_fill_manual(values = group_pal) + 
  labs(x = "Tree Type", y = "Proportion of Indicator Species", fill = "Fungal Guild") + 
  theme_classic() + theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        axis.text.x = element_text(size = 10, angle = 30, 
                                   hjust = 0.5, vjust = 0.5), 
        axis.text.y = element_text(size = 10), 
        legend.text = element_text(size = 8), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10), legend.position = "none")
bg_indic
```

```{r leaf indicator species 16s}
leaf_indic_16s$Family[which(is.na(leaf_indic_16s$Family))] <- "Other Bacteria"
# leaf_indic_16s$Family <- gsub(".*_fam_Incertae_sedis", "Other Fungi", leaf_indic_16s$Family)
# leaf_indic_16s$Class <- gsub(".*_cls_Incertae_sedis", "Other Fungi", leaf_indic_16s$Family)
# leaf_indic_16s$Order <- gsub(".*_ord_Incertae_sedis", "Other Fungi", leaf_indic_16s$Family)
leaf_indic_16s$Family <- gsub("f__", "", leaf_indic_16s$Family)
leaf_indic_16s$Order <- gsub("o__", "", leaf_indic_16s$Order)
leaf_indic_16s$Class <- gsub("c__", "", leaf_indic_16s$Class)
leaf_indic_16s$Phylum <- gsub("p__", "", leaf_indic_16s$Phylum)

for(i in 1:nrow(leaf_indic_16s)){
  if(leaf_indic_16s$Family[i] == "uncultured"){
    leaf_indic_16s$Family[i] <- paste0("Unknown ", leaf_indic_16s$Order[i])
  }
  if(leaf_indic_16s$Family[i] == "Unknown uncultured"){
    leaf_indic_16s$Family[i] <- paste0("Unknown ", leaf_indic_16s$Class[i])
  } 
  if(leaf_indic_16s$Family[i] == "Unknown uncultured"){
    leaf_indic_16s$Family[i] <- paste0("Unknown ", leaf_indic_16s$Phylum[i])
  }
  if(leaf_indic_16s$Family[i] == "Unknown uncultured"){
    leaf_indic_16s$Family[i] <- "Unknown Bacterium"
  }
}

order_of_families <-leaf_indic_16s[,which(colnames(leaf_indic_16s) %in% c("Phylum", "Class", "Order", "Family"))]

order_of_families <- order_of_families[order(order_of_families$Phylum, order_of_families$Class, order_of_families$Order, order_of_families$Family),]
order_of_families <- distinct(order_of_families)
# order_of_families <- order_of_families[-which(order_of_families$Family == "Other Fungi"),]

leaf_indic_16s$Family <- factor(leaf_indic_16s$Family, levels = c(order_of_families$Family))

leaf_indic_16s$X <- 1 
leaf_indic_16s <- leaf_indic_16s[, colnames(leaf_indic_16s) %in% c("X", "Class", "Family", "Group")]
leaf_indic_16s <- aggregate(leaf_indic_16s, X ~ Class + Family + Group, FUN = "sum")

leaf_indic_16s$Group <- gsub("Rural.edge", "Rural Forest Edge", leaf_indic_16s$Group)
leaf_indic_16s$Group <- gsub("Rural.interior", "Rural Forest Interior", leaf_indic_16s$Group)
leaf_indic_16s$Group <- gsub("Street.tree", "Urban Street Tree", leaf_indic_16s$Group)
leaf_indic_16s$Group <- gsub("Urban.edge", "Urban Forest Edge", leaf_indic_16s$Group)
leaf_indic_16s$Group <- gsub("Urban.interior", "Urban Forest Interior", leaf_indic_16s$Group)

leaf_indic_16s$Group <- factor(leaf_indic_16s$Group, levels = c("Urban Street Tree", "Urban Forest Edge", "Urban Forest Interior", "Rural Forest Edge", "Rural Forest Interior"))
# group_pal <- group_pal_fungi[which(names(group_pal_fungi) %in% leaf_indic_16s$primary_lifestyle)]

l_indic <- ggplot(leaf_indic_16s, aes(x = Group, y = X, fill = Family)) + 
  geom_bar(stat = 'identity', position = 'fill') + 
  #scale_fill_manual(values = group_pal) + 
  labs(x = "Tree Type", y = "Proportion of Indicator Species", fill = "Bacterial Family") + 
  theme_classic() + theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        axis.text.x = element_text(size = 10, angle = 30, 
                                   hjust = 0.5, vjust = 0.5), 
        axis.text.y = element_text(size = 10), 
        legend.text = element_text(size = 8), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10), legend.position = "none")
l_indic
```

```{r root indicator species}
bg_indic_16s$Family[which(is.na(bg_indic_16s$Family))] <- "Other Bacteria"
# bg_indic_16s$Family <- gsub(".*_fam_Incertae_sedis", "Other Fungi", bg_indic_16s$Family)
# bg_indic_16s$Class <- gsub(".*_cls_Incertae_sedis", "Other Fungi", bg_indic_16s$Family)
# bg_indic_16s$Order <- gsub(".*_ord_Incertae_sedis", "Other Fungi", bg_indic_16s$Family)
bg_indic_16s$Family <- gsub("f__", "", bg_indic_16s$Family)
bg_indic_16s$Order <- gsub("o__", "", bg_indic_16s$Order)
bg_indic_16s$Class <- gsub("c__", "", bg_indic_16s$Class)
bg_indic_16s$Phylum <- gsub("p__", "", bg_indic_16s$Phylum)

for(i in 1:nrow(bg_indic_16s)){
  if(bg_indic_16s$Family[i] == "uncultured"){
    bg_indic_16s$Family[i] <- paste0("Unknown ", bg_indic_16s$Order[i])
  }
  if(bg_indic_16s$Family[i] == "Unknown uncultured"){
    bg_indic_16s$Family[i] <- paste0("Unknown ", bg_indic_16s$Class[i])
  } 
  if(bg_indic_16s$Family[i] == "Unknown uncultured"){
    bg_indic_16s$Family[i] <- paste0("Unknown ", bg_indic_16s$Phylum[i])
  }
  if(bg_indic_16s$Family[i] == "Unknown uncultured"){
    bg_indic_16s$Family[i] <- "Unknown Bacterium"
  }
}

order_of_families <-bg_indic_16s[,which(colnames(bg_indic_16s) %in% c("Phylum", "Class", "Order", "Family"))]

order_of_families <- order_of_families[order(order_of_families$Phylum, order_of_families$Class, order_of_families$Order, order_of_families$Family),]
order_of_families <- distinct(order_of_families)
# order_of_families <- order_of_families[-which(order_of_families$Family == "Other Bacteria"),]

bg_indic_16s$Family <- factor(bg_indic_16s$Family, levels = c(order_of_families$Family))

bg_indic_16s$X <- 1 
bg_indic_16s <- bg_indic_16s[, colnames(bg_indic_16s) %in% c("X", "Class", "Family", "Group")]
bg_indic_16s <- aggregate(bg_indic_16s, X ~ Class + Family + Group, FUN = "sum")

bg_indic_16s$Group <- gsub("Rural.edge", "Rural Forest Edge", bg_indic_16s$Group)
bg_indic_16s$Group <- gsub("Rural.interior", "Rural Forest Interior", bg_indic_16s$Group)
bg_indic_16s$Group <- gsub("Street.tree", "Urban Street Tree", bg_indic_16s$Group)
bg_indic_16s$Group <- gsub("Urban.edge", "Urban Forest Edge", bg_indic_16s$Group)
bg_indic_16s$Group <- gsub("Urban.interior", "Urban Forest Interior", bg_indic_16s$Group)

bg_indic_16s$Group <- factor(bg_indic_16s$Group, levels = c("Urban Street Tree", "Urban Forest Edge", "Urban Forest Interior", "Rural Forest Edge", "Rural Forest Interior"))
# group_pal <- group_pal_fungi[which(names(group_pal_fungi) %in% bg_indic_16s$primary_lifestyle)]

bg_indic <- ggplot(bg_indic_16s, aes(x = Group, y = X, fill = Family)) + 
  geom_bar(stat = 'identity', position = 'fill') + 
  #scale_fill_manual(values = group_pal) + 
  labs(x = "Tree Type", y = "Proportion of Indicator Species", fill = "Fungal Guild") + 
  theme_classic() + theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        axis.text.x = element_text(size = 10, angle = 30, 
                                   hjust = 0.5, vjust = 0.5), 
        axis.text.y = element_text(size = 10), 
        legend.text = element_text(size = 8), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10), legend.position = "none")
bg_indic
```

Figure S1eXXX
```{r PCA heatmap}
leaf_its_pcs <- pca_leaf_df_its[,colnames(pca_leaf_df_its) %in%
                                        c("PC1", "PC2")]
leaf_16s_pcs <- pca_leaf_df_16s[,colnames(pca_leaf_df_16s) %in%
                                        c("PC1", "PC2")]
soil_its_pcs <- pca_soil_df_its[,colnames(pca_soil_df_its) %in%
                                        c("PC1", "PC2")]
soil_16s_pcs <- pca_soil_df_16s[,colnames(pca_soil_df_16s) %in%
                                        c("PC1", "PC2")]

colnames(leaf_its_pcs) <- c("Fungal Leaf PC1","Fungal Leaf PC2")
colnames(leaf_16s_pcs) <- c("Bacterial Leaf PC1","Bacterial Leaf PC2")
colnames(soil_its_pcs) <- c("Fungal Soil PC1","Fungal Soil PC2")
colnames(soil_16s_pcs) <- c("Bacterial Soil PC1","Bacterial Soil PC2")

metadata_leaf_heatmap_its <- pca_leaf_df_its[,c(19,20,22,25,32:38,40,51,55:57,65:67,69:74,77,80:82)]
metadata_leaf_heatmap_16s <- pca_leaf_df_16s[,c(19,20,22,25,32:38,40,51,55:57,65:67,69:74,77,80:82)]
metadata_soil_heatmap_its <- pca_soil_df_its[,c(19,20,22,25,32:38,40,51,55:57,65:67,69:74,77,80:82)]
metadata_soil_heatmap_16s <- pca_soil_df_16s[,c(19,20,22,25,32:38,40,51,55:57,65:67,69:74,77,80:82)]
colnames(metadata_soil_heatmap_its) <- c("Tree Distance from Edge", 
                                         "Tree Distance from Boston", 
                                         "Tree DBH", "Tree Growth Rate",
                                         "Soil Temperature", 
                                         "Soil Bulk Density", 
                                         "Soil pH", "Soil Moisture", 
                                         "Soil Organic Matter", 
                                         "Soil Available NH<sub>4</sub><sup>+</sup>",
                                         "Soil Available NO<sub>3</sub><sup>-</sup>", 
                                         "Normalized Root Biomass", 
                                         "Soil Soluble Salt",
                                         "NO Concentration",
                                         "NO<sub>2</sub> Concentration",
                                         "NO<sub>x</sub> Concentration",
                                         "O<sub>3</sub> Concentration",
                                         "Foliar Nitrogen %",
                                         "Foliar C:N",
                                         "Ca<sup>2+</sup> Deposition",
                                         "Cl<sup>-</sup> Deposition",
                                         "H<sup>+</sup> Deposition",
                                         "K<sup>+</sup> Deposition",
                                         "Mg<sup>2+</sup> Deposition",
                                         "Na<sup>+</sup> Deposition",
                                         "SO<sub>4</sub><sup>2-</sup> Deposition",
                                         "NH<sub>4</sub><sup>+</sup> Deposition",
                                         "NO<sub>3</sub><sup>-</sup> Deposition",
                                         "Total Inorganic N Deposition")
colnames(metadata_soil_heatmap_16s) <- c("Tree Distance from Edge", 
                                         "Tree Distance from Boston", 
                                         "Tree DBH", "Tree Growth Rate",
                                         "Soil Temperature", 
                                         "Soil Bulk Density", 
                                         "Soil pH", "Soil Moisture", 
                                         "Soil Organic Matter", 
                                         "Soil Available NH<sub>4</sub><sup>+</sup>",
                                         "Soil Available NO<sub>3</sub><sup>-</sup>", 
                                         "Normalized Root Biomass", 
                                         "Soil Soluble Salt",
                                         "NO Concentration",
                                         "NO<sub>2</sub> Concentration",
                                         "NO<sub>x</sub> Concentration",
                                         "O<sub>3</sub> Concentration",
                                         "Foliar Nitrogen %",
                                         "Foliar C:N",
                                         "Ca<sup>2+</sup> Deposition",
                                         "Cl<sup>-</sup> Deposition",
                                         "H<sup>+</sup> Deposition",
                                         "K<sup>+</sup> Deposition",
                                         "Mg<sup>2+</sup> Deposition",
                                         "Na<sup>+</sup> Deposition",
                                         "SO<sub>4</sub><sup>2-</sup> Deposition",
                                         "NH<sub>4</sub><sup>+</sup> Deposition",
                                         "NO<sub>3</sub><sup>-</sup> Deposition",
                                         "Total Inorganic N Deposition")
colnames(metadata_leaf_heatmap_its) <- c("Tree Distance from Edge", 
                                         "Tree Distance from Boston", 
                                         "Tree DBH", "Tree Growth Rate",
                                         "Soil Temperature", 
                                         "Soil Bulk Density", 
                                         "Soil pH", "Soil Moisture", 
                                         "Soil Organic Matter", 
                                         "Soil Available NH<sub>4</sub><sup>+</sup>",
                                         "Soil Available NO<sub>3</sub><sup>-</sup>", 
                                         "Normalized Root Biomass", 
                                         "Soil Soluble Salt",
                                         "NO Concentration",
                                         "NO<sub>2</sub> Concentration",
                                         "NO<sub>x</sub> Concentration",
                                         "O<sub>3</sub> Concentration",
                                         "Foliar Nitrogen %",
                                         "Foliar C:N",
                                         "Ca<sup>2+</sup> Deposition",
                                         "Cl<sup>-</sup> Deposition",
                                         "H<sup>+</sup> Deposition",
                                         "K<sup>+</sup> Deposition",
                                         "Mg<sup>2+</sup> Deposition",
                                         "Na<sup>+</sup> Deposition",
                                         "SO<sub>4</sub><sup>2-</sup> Deposition",
                                         "NH<sub>4</sub><sup>+</sup> Deposition",
                                         "NO<sub>3</sub><sup>-</sup> Deposition",
                                         "Total Inorganic N Deposition")
colnames(metadata_leaf_heatmap_16s) <- c("Tree Distance from Edge", 
                                         "Tree Distance from Boston", 
                                         "Tree DBH", "Tree Growth Rate",
                                         "Soil Temperature", 
                                         "Soil Bulk Density", 
                                         "Soil pH", "Soil Moisture", 
                                         "Soil Organic Matter", 
                                         "Soil Available NH<sub>4</sub><sup>+</sup>",
                                         "Soil Available NO<sub>3</sub><sup>-</sup>", 
                                         "Normalized Root Biomass", 
                                         "Soil Soluble Salt",
                                         "NO Concentration",
                                         "NO<sub>2</sub> Concentration",
                                         "NO<sub>x</sub> Concentration",
                                         "O<sub>3</sub> Concentration",
                                         "Foliar Nitrogen %",
                                         "Foliar C:N",
                                         "Ca<sup>2+</sup> Deposition",
                                         "Cl<sup>-</sup> Deposition",
                                         "H<sup>+</sup> Deposition",
                                         "K<sup>+</sup> Deposition",
                                         "Mg<sup>2+</sup> Deposition",
                                         "Na<sup>+</sup> Deposition",
                                         "SO<sub>4</sub><sup>2-</sup> Deposition",
                                         "NH<sub>4</sub><sup>+</sup> Deposition",
                                         "NO<sub>3</sub><sup>-</sup> Deposition",
                                         "Total Inorganic N Deposition")
# Make coefficient matrix
Result_leaf_its <- NULL
for (i in 1:ncol(leaf_its_pcs)){
  estimate_leaf_its <- NULL
  for (j in 1:ncol(metadata_leaf_heatmap_its)){
    cor_leaf_its <- cor.test(leaf_its_pcs[,i],metadata_leaf_heatmap_its[,j], method="p")
    estimate_leaf_its <- c(estimate_leaf_its, cor_leaf_its$estimate)
  }
  Result_leaf_its <- rbind(Result_leaf_its, estimate_leaf_its)
}
colnames(Result_leaf_its) <- colnames(metadata_leaf_heatmap_its)
rownames(Result_leaf_its) <- colnames(leaf_its_pcs) 

Result_soil_its <- NULL
for (i in 1:ncol(soil_its_pcs)){
  estimate_soil_its <- NULL
  for (j in 1:ncol(metadata_soil_heatmap_its)){
    cor_soil_its <- cor.test(soil_its_pcs[,i],metadata_soil_heatmap_its[,j], method="p")
    estimate_soil_its <- c(estimate_soil_its, cor_soil_its$estimate)
  }
  Result_soil_its <- rbind(Result_soil_its, estimate_soil_its)
}
colnames(Result_soil_its) <- colnames(metadata_soil_heatmap_its)
rownames(Result_soil_its) <- colnames(soil_its_pcs) 

Result_soil_16s <- NULL
for (i in 1:ncol(soil_16s_pcs)){
  estimate_soil_16s <- NULL
  for (j in 1:ncol(metadata_soil_heatmap_16s)){
    cor_soil_16s <- cor.test(soil_16s_pcs[,i],metadata_soil_heatmap_16s[,j], method="p")
    estimate_soil_16s <- c(estimate_soil_16s, cor_soil_16s$estimate)
  }
  Result_soil_16s <- rbind(Result_soil_16s, estimate_soil_16s)
}
colnames(Result_soil_16s) <- colnames(metadata_soil_heatmap_16s)
rownames(Result_soil_16s) <- colnames(soil_16s_pcs) 

Result_leaf_16s <- NULL
for (i in 1:ncol(leaf_16s_pcs)){
  estimate_leaf_16s <- NULL
  for (j in 1:ncol(metadata_leaf_heatmap_16s)){
    cor_leaf_16s <- cor.test(leaf_16s_pcs[,i],metadata_leaf_heatmap_16s[,j], method="p")
    estimate_leaf_16s <- c(estimate_leaf_16s, cor_leaf_16s$estimate)
  }
  Result_leaf_16s <- rbind(Result_leaf_16s, estimate_leaf_16s)
}
colnames(Result_leaf_16s) <- colnames(metadata_leaf_heatmap_16s)
rownames(Result_leaf_16s) <- colnames(leaf_16s_pcs) 

# Make Pval matrix
Result.Pval_soil_its <- NULL
for (i in 1:ncol(soil_its_pcs)){
  Pval_soil_its <- NULL
  for (j in 1:ncol(metadata_soil_heatmap_its)){
    cor_soil_its <- cor.test(soil_its_pcs[,i],metadata_soil_heatmap_its[,j], method="p")
    Pval_soil_its <- c(Pval_soil_its, cor_soil_its$p.value)
  }
  Result.Pval_soil_its <- rbind (Result.Pval_soil_its, Pval_soil_its)
}
colnames(Result.Pval_soil_its) <- colnames(metadata_soil_heatmap_its)
rownames(Result.Pval_soil_its) <- colnames(soil_its_pcs) 

Result.Pval_leaf_its <- NULL
for (i in 1:ncol(leaf_its_pcs)){
  Pval_leaf_its <- NULL
  for (j in 1:ncol(metadata_leaf_heatmap_its)){
    cor_leaf_its <- cor.test(leaf_its_pcs[,i],metadata_leaf_heatmap_its[,j], method="p")
    Pval_leaf_its <- c(Pval_leaf_its, cor_leaf_its$p.value)
  }
  Result.Pval_leaf_its <- rbind (Result.Pval_leaf_its, Pval_leaf_its)
}
colnames(Result.Pval_leaf_its) <- colnames(metadata_leaf_heatmap_its)
rownames(Result.Pval_leaf_its) <- colnames(leaf_its_pcs) 

Result.Pval_soil_16s <- NULL
for (i in 1:ncol(soil_16s_pcs)){
  Pval_soil_16s <- NULL
  for (j in 1:ncol(metadata_soil_heatmap_16s)){
    cor_soil_16s <- cor.test(soil_16s_pcs[,i],metadata_soil_heatmap_16s[,j], method="p")
    Pval_soil_16s <- c(Pval_soil_16s, cor_soil_16s$p.value)
  }
  Result.Pval_soil_16s <- rbind (Result.Pval_soil_16s, Pval_soil_16s)
}
colnames(Result.Pval_soil_16s) <- colnames(metadata_soil_heatmap_16s)
rownames(Result.Pval_soil_16s) <- colnames(soil_16s_pcs) 

Result.Pval_leaf_16s <- NULL
for (i in 1:ncol(leaf_16s_pcs)){
  Pval_leaf_16s <- NULL
  for (j in 1:ncol(metadata_leaf_heatmap_16s)){
    cor_leaf_16s <- cor.test(leaf_16s_pcs[,i],metadata_leaf_heatmap_16s[,j], method="p")
    Pval_leaf_16s <- c(Pval_leaf_16s, cor_leaf_16s$p.value)
  }
  Result.Pval_leaf_16s <- rbind (Result.Pval_leaf_16s, Pval_leaf_16s)
}
colnames(Result.Pval_leaf_16s) <- colnames(metadata_leaf_heatmap_16s)
rownames(Result.Pval_leaf_16s) <- colnames(leaf_16s_pcs) 

# Make Asterisk matrix
Result.Pval.combined_soil_its <- Result.Pval_soil_its
Result.asterisk_soil_its <- Result.Pval.combined_soil_its
for (k in 1:ncol(Result.Pval.combined_soil_its)){
  for (i in 1:nrow(Result.Pval.combined_soil_its)){
    if(!is.na(Result.Pval.combined_soil_its[i,k])){
      if (Result.Pval.combined_soil_its[i,k] < 0.001){
        Result.asterisk_soil_its[i,k] <- "***"
      } else if (Result.Pval.combined_soil_its[i,k] < 0.01){
        Result.asterisk_soil_its[i,k] <- "**"
      } else if (Result.Pval.combined_soil_its[i,k] < 0.05){
        Result.asterisk_soil_its[i,k] <- "*"
      } else {
        Result.asterisk_soil_its[i,k]<- " "
      }
    } 
  }
}

Result.Pval.combined_leaf_its <- Result.Pval_leaf_its
Result.asterisk_leaf_its <- Result.Pval.combined_leaf_its
for (k in 1:ncol(Result.Pval.combined_leaf_its)){
  for (i in 1:nrow(Result.Pval.combined_leaf_its)){
    if(!is.na(Result.Pval.combined_leaf_its[i,k])){
      if (Result.Pval.combined_leaf_its[i,k] < 0.001){
        Result.asterisk_leaf_its[i,k] <- "***"
      } else if (Result.Pval.combined_leaf_its[i,k] < 0.01){
        Result.asterisk_leaf_its[i,k] <- "**"
      } else if (Result.Pval.combined_leaf_its[i,k] < 0.05){
        Result.asterisk_leaf_its[i,k] <- "*"
      } else {
        Result.asterisk_leaf_its[i,k]<- " "
      }
    } 
  }
}

Result.Pval.combined_soil_16s <- Result.Pval_soil_16s
Result.asterisk_soil_16s <- Result.Pval.combined_soil_16s
for (k in 1:ncol(Result.Pval.combined_soil_16s)){
  for (i in 1:nrow(Result.Pval.combined_soil_16s)){
    if(!is.na(Result.Pval.combined_soil_16s[i,k])){
      if (Result.Pval.combined_soil_16s[i,k] < 0.001){
        Result.asterisk_soil_16s[i,k] <- "***"
      } else if (Result.Pval.combined_soil_16s[i,k] < 0.01){
        Result.asterisk_soil_16s[i,k] <- "**"
      } else if (Result.Pval.combined_soil_16s[i,k] < 0.05){
        Result.asterisk_soil_16s[i,k] <- "*"
      } else {
        Result.asterisk_soil_16s[i,k]<- " "
      }
    } 
  }
}

Result.Pval.combined_leaf_16s <- Result.Pval_leaf_16s
Result.asterisk_leaf_16s <- Result.Pval.combined_leaf_16s
for (k in 1:ncol(Result.Pval.combined_leaf_16s)){
  for (i in 1:nrow(Result.Pval.combined_leaf_16s)){
    if(!is.na(Result.Pval.combined_leaf_16s[i,k])){
      if (Result.Pval.combined_leaf_16s[i,k] < 0.001){
        Result.asterisk_leaf_16s[i,k] <- "***"
      } else if (Result.Pval.combined_leaf_16s[i,k] < 0.01){
        Result.asterisk_leaf_16s[i,k] <- "**"
      } else if (Result.Pval.combined_leaf_16s[i,k] < 0.05){
        Result.asterisk_leaf_16s[i,k] <- "*"
      } else {
        Result.asterisk_leaf_16s[i,k]<- " "
      }
    } 
  }
}
# Combine netowrk propeties + others
Result.combined_soil_its <- Result_soil_its
melt_soil_its <- melt(Result.combined_soil_its)
melt.asterisk_soil_its <- melt(Result.asterisk_soil_its)

Result.combined_leaf_its <- Result_leaf_its
melt_leaf_its <- melt(Result.combined_leaf_its)
melt.asterisk_leaf_its <- melt(Result.asterisk_leaf_its)

Result.combined_soil_16s <- Result_soil_16s
melt_soil_16s <- melt(Result.combined_soil_16s)
melt.asterisk_soil_16s <- melt(Result.asterisk_soil_16s)

Result.combined_leaf_16s <- Result_leaf_16s
melt_leaf_16s <- melt(Result.combined_leaf_16s)
melt.asterisk_leaf_16s <- melt(Result.asterisk_leaf_16s)

melt_all <- rbind(melt_soil_its, melt_soil_16s, melt_leaf_its, melt_leaf_16s)
melt_asterisk_all <- rbind(melt.asterisk_soil_its, melt.asterisk_soil_16s, 
                           melt.asterisk_leaf_its, melt.asterisk_leaf_16s)

# Visualize
vdata_its <- cbind(melt_all, melt_asterisk_all$value)
colnames(vdata_its) <- c("microbe", "environment", "coef", "Pval")

vdata_its$microbe <- factor(vdata_its$microbe, levels = c("Fungal Leaf PC1", 
                                                          "Bacterial Leaf PC1",
                                                          "Fungal Soil PC1", 
                                                          "Bacterial Soil PC1",
                                                          "Fungal Leaf PC2", 
                                                          "Bacterial Leaf PC2",
                                                          "Fungal Soil PC2", 
                                                          "Bacterial Soil PC2"))

order_of_env_factors <- vdata_its[which(vdata_its$microbe == "Fungal Leaf PC1"),]

vdata_its$environment <- factor(vdata_its$environment,
                            levels = order_of_env_factors$environment[order(order_of_env_factors$coef)])

heatmap_pcs <- ggplot(vdata_its) +
  geom_tile(aes(x=microbe, y=environment, fill=coef)) +
  geom_text(aes(x=microbe, y=environment, label = Pval), size = 2.5) +
  labs(x="",y="") + theme_minimal() +
  theme(axis.text.x = element_text(angle = 90, hjust=1,vjust=0.3)) +
  theme(axis.text.x = element_text(angle = 90, hjust=1,vjust=0.3)) + 
  scale_fill_gradient2(low = "#003049", mid = "#FDF0D5", high = "#C00000", 
                       name = "Correlation\nCoefficient") +
  theme(axis.title.x = element_text(size = 15), 
        axis.title.y = element_text(size = 15), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10)) + 
  ggtitle("") + coord_fixed()
heatmap_pcs

ggsave("/projectnb/talbot-lab-data/Katies_data/Street_Trees/Figures/heatmap_pcs.pdf", device = cairo_pdf, width = 5, height = 10, units = "in", dpi = 300)
```

Figure XXX
```{r o soil networks}
o_networks <- network_metrics[which(network_metrics$Sample_Type == "Soil, 0-15 cm"),]

ggplot(o_networks, aes(x = n_samples, y = Complexity)) + geom_point() + geom_smooth(method = "lm", aes(ymax = after_stat(y + se * sqrt(length(y))),                   ymin = after_stat(y - se * sqrt(length(y))))) +  


ggplot(o_networks, aes(x = Tree_Type, y = Density, fill = Tree_Type)) + geom_boxplot() + theme_classic() + scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("Average Number of Connections per Microbe")

ggplot(o_networks, aes(x = Tree_Type, y = Complexity, fill = Tree_Type)) + geom_boxplot() + theme_classic() + scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("Average Number of Connections per Microbe")

ggplot(o_networks, 
       aes(x = Tree_Type, y = Transitivity, fill = Tree_Type)) +
  geom_boxplot() + theme_classic() + 
  scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("Network Transitivity (Clustering Coefficient)")

ggplot(o_networks, 
       aes(x = Tree_Type, y = K_assort, fill = Tree_Type)) +
  geom_boxplot() + theme_classic() + 
  scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("Kingdom Assortativity")

ggplot(o_networks, 
       aes(x = Tree_Type, y = ECM_betweenness, fill = Tree_Type)) +
  geom_boxplot() + theme_classic() + 
  scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("ECM Betweenness Centrality")

ggplot(o_networks, 
       aes(x = Tree_Type, y = ECM_degree, fill = Tree_Type)) +
  geom_boxplot() + theme_classic() + 
  scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("ECM Degree Centrality")

ggplot(o_networks, 
       aes(x = Tree_Type, y = plantpath_betweenness, fill = Tree_Type)) +
  geom_boxplot() + theme_classic() + 
  scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("Plant Pathogen Betweenness Centrality")

ggplot(o_networks, 
       aes(x = Tree_Type, y = plantpath_degree, fill = Tree_Type)) +
  geom_boxplot() + theme_classic() + 
  scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("Plant Pathogen Degree Centrality")

ggplot(o_networks, 
       aes(x = Tree_Type, y = animalpath_betweenness, fill = Tree_Type)) +
  geom_boxplot() + theme_classic() + 
  scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("Animal Pathogen Betweenness Centrality")

ggplot(o_networks, 
       aes(x = Tree_Type, y = animalpath_degree, fill = Tree_Type)) +
  geom_boxplot() + theme_classic() + 
  scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("Animal Pathogen Degree Centrality")

ggplot(o_networks, 
       aes(x = Tree_Type, y = sap_degree, fill = Tree_Type)) +
  geom_boxplot() + theme_classic() + 
  scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("Saprotroph Degree Centrality")

ggplot(o_networks, 
       aes(x = Tree_Type, y = sap_betweenness, fill = Tree_Type)) +
  geom_boxplot() + theme_classic() + 
  scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("Saprotroph Betweenness Centrality")

summary(aov(Density ~ Tree_Type, o_networks))
summary(aov(Complexity ~ Tree_Type, o_networks))
summary(aov(Transitivity ~ Tree_Type, o_networks))
summary(aov(K_assort ~ Tree_Type, o_networks))
summary(aov(ECM_betweenness ~ Tree_Type, o_networks))
summary(aov(ECM_degree ~ Tree_Type, o_networks))
summary(aov(plantpath_betweenness ~ Tree_Type, o_networks))
summary(aov(plantpath_degree ~ Tree_Type, o_networks))
summary(aov(animalpath_betweenness ~ Tree_Type, o_networks))
summary(aov(animalpath_degree ~ Tree_Type, o_networks))
summary(aov(sap_betweenness ~ Tree_Type, o_networks))
summary(aov(sap_degree ~ Tree_Type, o_networks))

TukeyHSD(aov(Complexity ~ Tree_Type, o_networks))
TukeyHSD(aov(Transitivity ~ Tree_Type, o_networks))
TukeyHSD(aov(K_assort ~ Tree_Type, o_networks))
TukeyHSD(aov(sap_degree ~ Tree_Type, o_networks))
```

```{r m soil networks}
m_networks <- network_metrics[which(network_metrics$Sample_Type == "Soil, 15-30 cm"),]

ggplot(m_networks, aes(x = n_samples, y = Complexity)) + geom_point() + geom_smooth(method = "lm", aes(ymax = after_stat(y + se * sqrt(length(y))),                   ymin = after_stat(y - se * sqrt(length(y))))) +  

ggplot(m_networks, aes(x = Tree_Type, y = Density, fill = Tree_Type)) + geom_boxplot() + theme_classic() + scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("Average Number of Connections per Microbe")

ggplot(m_networks, aes(x = Tree_Type, y = Complexity, fill = Tree_Type)) + geom_boxplot() + theme_classic() + scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("Average Number of Connections per Microbe")

ggplot(m_networks, 
       aes(x = Tree_Type, y = Transitivity, fill = Tree_Type)) +
  geom_boxplot() + theme_classic() + 
  scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("Network Transitivity (Clustering Coefficient)")

ggplot(m_networks, 
       aes(x = Tree_Type, y = K_assort, fill = Tree_Type)) +
  geom_boxplot() + theme_classic() + 
  scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("Kingdom Assortativity")

ggplot(m_networks, 
       aes(x = Tree_Type, y = ECM_betweenness, fill = Tree_Type)) +
  geom_boxplot() + theme_classic() + 
  scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("ECM Betweenness Centrality")

ggplot(m_networks, 
       aes(x = Tree_Type, y = ECM_degree, fill = Tree_Type)) +
  geom_boxplot() + theme_classic() + 
  scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("ECM Degree Centrality")

ggplot(m_networks, 
       aes(x = Tree_Type, y = plantpath_betweenness, fill = Tree_Type)) +
  geom_boxplot() + theme_classic() + 
  scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("Plant Pathogen Betweenness Centrality")

ggplot(m_networks, 
       aes(x = Tree_Type, y = plantpath_degree, fill = Tree_Type)) +
  geom_boxplot() + theme_classic() + 
  scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("Plant Pathogen Degree Centrality")

ggplot(m_networks, 
       aes(x = Tree_Type, y = animalpath_betweenness, fill = Tree_Type)) +
  geom_boxplot() + theme_classic() + 
  scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("Animal Pathogen Betweenness Centrality")

ggplot(m_networks, 
       aes(x = Tree_Type, y = animalpath_degree, fill = Tree_Type)) +
  geom_boxplot() + theme_classic() + 
  scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("Animal Pathogen Degree Centrality")

ggplot(m_networks, 
       aes(x = Tree_Type, y = sap_degree, fill = Tree_Type)) +
  geom_boxplot() + theme_classic() + 
  scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("Saprotroph Degree Centrality")

ggplot(m_networks, 
       aes(x = Tree_Type, y = sap_betweenness, fill = Tree_Type)) +
  geom_boxplot() + theme_classic() + 
  scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("Saprotroph Betweenness Centrality")

summary(aov(Density ~ Tree_Type, m_networks))
summary(aov(Complexity ~ Tree_Type, m_networks))
summary(aov(Transitivity ~ Tree_Type, m_networks))
summary(aov(K_assort ~ Tree_Type, m_networks))
summary(aov(ECM_betweenness ~ Tree_Type, m_networks))
summary(aov(ECM_degree ~ Tree_Type, m_networks))
summary(aov(plantpath_betweenness ~ Tree_Type, m_networks))
summary(aov(plantpath_degree ~ Tree_Type, m_networks))
summary(aov(animalpath_betweenness ~ Tree_Type, m_networks))
summary(aov(animalpath_degree ~ Tree_Type, m_networks))
summary(aov(sap_betweenness ~ Tree_Type, m_networks))
summary(aov(sap_degree ~ Tree_Type, m_networks))

TukeyHSD(aov(Complexity ~ Tree_Type, m_networks))
TukeyHSD(aov(ECM_degree ~ Tree_Type, m_networks))
```

```{r root networks}
root_networks <- network_metrics[which(network_metrics$Sample_Type == "Root"),]

ggplot(root_networks, aes(x = n_samples, y = Complexity)) + geom_point() + geom_smooth(method = "lm", aes(ymax = after_stat(y + se * sqrt(length(y))),                   ymin = after_stat(y - se * sqrt(length(y))))) +  

ggplot(root_networks, aes(x = Tree_Type, y = Density, fill = Tree_Type)) + geom_boxplot() + theme_classic() + scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("Average Number of Connections per Microbe")

ggplot(root_networks, aes(x = Tree_Type, y = Complexity, fill = Tree_Type)) + geom_boxplot() + theme_classic() + scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("Average Number of Connections per Microbe")

ggplot(root_networks, 
       aes(x = Tree_Type, y = Transitivity, fill = Tree_Type)) +
  geom_boxplot() + theme_classic() + 
  scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("Network Transitivity (Clustering Coefficient)")

ggplot(root_networks, 
       aes(x = Tree_Type, y = K_assort, fill = Tree_Type)) +
  geom_boxplot() + theme_classic() + 
  scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("Kingdom Assortativity")

ggplot(root_networks, 
       aes(x = Tree_Type, y = ECM_betweenness, fill = Tree_Type)) +
  geom_boxplot() + theme_classic() + 
  scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("ECM Betweenness Centrality")

ggplot(root_networks, 
       aes(x = Tree_Type, y = ECM_degree, fill = Tree_Type)) +
  geom_boxplot() + theme_classic() + 
  scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("ECM Degree Centrality")

ggplot(root_networks, 
       aes(x = Tree_Type, y = plantpath_betweenness, fill = Tree_Type)) +
  geom_boxplot() + theme_classic() + 
  scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("Plant Pathogen Betweenness Centrality")

ggplot(root_networks, 
       aes(x = Tree_Type, y = plantpath_degree, fill = Tree_Type)) +
  geom_boxplot() + theme_classic() + 
  scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("Plant Pathogen Degree Centrality")

ggplot(root_networks, 
       aes(x = Tree_Type, y = animalpath_betweenness, fill = Tree_Type)) +
  geom_boxplot() + theme_classic() + 
  scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("Animal Pathogen Betweenness Centrality")

ggplot(root_networks, 
       aes(x = Tree_Type, y = animalpath_degree, fill = Tree_Type)) +
  geom_boxplot() + theme_classic() + 
  scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("Animal Pathogen Degree Centrality")

ggplot(root_networks, 
       aes(x = Tree_Type, y = sap_degree, fill = Tree_Type)) +
  geom_boxplot() + theme_classic() + 
  scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("Saprotroph Degree Centrality")

ggplot(root_networks, 
       aes(x = Tree_Type, y = sap_betweenness, fill = Tree_Type)) +
  geom_boxplot() + theme_classic() + 
  scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("Saprotroph Betweenness Centrality")

summary(aov(Density ~ Tree_Type, root_networks))
summary(aov(Complexity ~ Tree_Type, root_networks))
summary(aov(Transitivity ~ Tree_Type, root_networks))
summary(aov(K_assort ~ Tree_Type, root_networks))
summary(aov(ECM_betweenness ~ Tree_Type, root_networks))
summary(aov(ECM_degree ~ Tree_Type, root_networks))
summary(aov(plantpath_betweenness ~ Tree_Type, root_networks))
summary(aov(plantpath_degree ~ Tree_Type, root_networks))
summary(aov(animalpath_betweenness ~ Tree_Type, root_networks))
summary(aov(animalpath_degree ~ Tree_Type, root_networks))
summary(aov(sap_betweenness ~ Tree_Type, root_networks))
summary(aov(sap_degree ~ Tree_Type, root_networks))
```

```{r leaf networks}
leaf_networks <- network_metrics[which(network_metrics$Sample_Type == "Leaf"),]
ggplot(leaf_networks, aes(x = n_samples, y = Complexity)) + geom_point() + geom_smooth(method = "lm", aes(ymax = after_stat(y + se * sqrt(length(y))),                   ymin = after_stat(y - se * sqrt(length(y))))) +  

ggplot(leaf_networks, aes(x = Tree_Type, y = Density, fill = Tree_Type)) + geom_boxplot() + theme_classic() + scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("Average Number of Connections per Microbe")

ggplot(leaf_networks, aes(x = Tree_Type, y = Complexity, fill = Tree_Type)) + geom_boxplot() + theme_classic() + scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("Average Number of Connections per Microbe")

ggplot(leaf_networks, 
       aes(x = Tree_Type, y = Transitivity, fill = Tree_Type)) +
  geom_boxplot() + theme_classic() + 
  scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("Network Transitivity (Clustering Coefficient)")

ggplot(leaf_networks, 
       aes(x = Tree_Type, y = K_assort, fill = Tree_Type)) +
  geom_boxplot() + theme_classic() + 
  scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("Kingdom Assortativity")

ggplot(leaf_networks, 
       aes(x = Tree_Type, y = epiphyte_betweenness, fill = Tree_Type)) +
  geom_boxplot() + theme_classic() + 
  scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("Epiphyte Betweenness Centrality")

ggplot(leaf_networks, 
       aes(x = Tree_Type, y = epiphyte_degree, fill = Tree_Type)) +
  geom_boxplot() + theme_classic() + 
  scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("Epiphyte Degree Centrality")

ggplot(leaf_networks, 
       aes(x = Tree_Type, y = plantpath_betweenness, fill = Tree_Type)) +
  geom_boxplot() + theme_classic() + 
  scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("Plant Pathogen Betweenness Centrality")

ggplot(leaf_networks, 
       aes(x = Tree_Type, y = plantpath_degree, fill = Tree_Type)) +
  geom_boxplot() + theme_classic() + 
  scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("Plant Pathogen Degree Centrality")

ggplot(leaf_networks, 
       aes(x = Tree_Type, y = animalpath_betweenness, fill = Tree_Type)) +
  geom_boxplot() + theme_classic() + 
  scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("Animal Pathogen Betweenness Centrality")

ggplot(leaf_networks, 
       aes(x = Tree_Type, y = animalpath_degree, fill = Tree_Type)) +
  geom_boxplot() + theme_classic() + 
  scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("Animal Pathogen Degree Centrality")

ggplot(leaf_networks, 
       aes(x = Tree_Type, y = endophyte_degree, fill = Tree_Type)) +
  geom_boxplot() + theme_classic() + 
  scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("Endophyte Degree Centrality")

ggplot(leaf_networks, 
       aes(x = Tree_Type, y = endophyte_betweenness, fill = Tree_Type)) +
  geom_boxplot() + theme_classic() + 
  scale_fill_manual(values = pal[c(1,3,5)]) + 
  theme(axis.title.x = element_text(size = 10), 
        axis.title.y = element_text(size = 10), 
        legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10), 
        plot.title = element_text(size = 10),
        legend.position = "none") + 
  xlab("Tree Type") + 
  ylab("Endophyte Betweenness Centrality")

summary(aov(Density ~ Tree_Type, leaf_networks))
summary(aov(Complexity ~ Tree_Type, leaf_networks))
summary(aov(Transitivity ~ Tree_Type, leaf_networks))
summary(aov(K_assort ~ Tree_Type, leaf_networks))
summary(aov(epiphyte_betweenness ~ Tree_Type, leaf_networks))
summary(aov(epiphyte_degree ~ Tree_Type, leaf_networks))
summary(aov(plantpath_betweenness ~ Tree_Type, leaf_networks))
summary(aov(plantpath_degree ~ Tree_Type, leaf_networks))
summary(aov(animalpath_betweenness ~ Tree_Type, leaf_networks))
summary(aov(animalpath_degree ~ Tree_Type, leaf_networks))
summary(aov(endophyte_betweenness ~ Tree_Type, leaf_networks))
summary(aov(endophyte_degree ~ Tree_Type, leaf_networks))

TukeyHSD(aov(plantpath_degree ~ Tree_Type, leaf_networks))
```